<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Karissa Barthelson" />

<meta name="date" content="2023-06-28" />

<title>analysis_final</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">2021_MPSIIIBvQ96-RNAseq-7dpfLarve</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="QC.html">QC</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">analysis_final</h1>
<h4 class="author">Karissa Barthelson</h4>
<h4 class="date">2023-06-28</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-09-02
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>2021_MPSIIIBvQ96-RNAseq-7dpfLarve/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211120code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211120)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211120code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211120)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomkarissab2021MPSIIIBvQ96RNAseq7dpfLarvetree9f57eae079b1a41bb0d82eb15f17819b7d9c58d9targetblank9f57eaea"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/karissa-b/2021_MPSIIIBvQ96-RNAseq-7dpfLarve/tree/9f57eae079b1a41bb0d82eb15f17819b7d9c58d9" target="_blank">9f57eae</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomkarissab2021MPSIIIBvQ96RNAseq7dpfLarvetree9f57eae079b1a41bb0d82eb15f17819b7d9c58d9targetblank9f57eaea" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/karissa-b/2021_MPSIIIBvQ96-RNAseq-7dpfLarve/tree/9f57eae079b1a41bb0d82eb15f17819b7d9c58d9" target="_blank">9f57eae</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rapp.history
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/
    Ignored:    data/.DS_Store
    Ignored:    data/R_objects/.DS_Store
    Ignored:    data/R_objects/larvae/.DS_Store
    Ignored:    data/adult_brain/.DS_Store
    Ignored:    data/adult_brain/05_featureCounts/.DS_Store
    Ignored:    data/gene_sets/.DS_Store
    Ignored:    data/larvae/.DS_Store
    Ignored:    data/larvae/fastqc_align/.DS_Store
    Ignored:    data/larvae/fastqc_align_dedup/.DS_Store
    Ignored:    data/larvae/fastqc_raw/.DS_Store
    Ignored:    data/larvae/fastqc_trim/.DS_Store
    Ignored:    data/larvae/featureCounts/.DS_Store
    Ignored:    data/larvae/meta/.DS_Store
    Ignored:    data/larvae/starAlignLog/.DS_Store
    Ignored:    output/.DS_Store
    Ignored:    output/plots/
    Ignored:    output/plots4pub/

Untracked files:
    Untracked:  analysis/6mBrain_fems.rmd
    Untracked:  analysis/analysis_final.rmd
    Untracked:  analysis/checkGenotypes.rmd
    Untracked:  analysis/nhi6mdata.rmd
    Untracked:  analysis/plots4pub2.rmd
    Untracked:  code/Snakefile
    Untracked:  data/Nhi_data/
    Untracked:  data/R_objects/adult_brain/celltypeNhi.rds
    Untracked:  data/R_objects/adult_brain/dgeNhi.rds
    Untracked:  data/R_objects/adult_brain/fryGoNhi.rds
    Untracked:  data/R_objects/adult_brain/hmp_ireNhi.rds
    Untracked:  data/R_objects/adult_brain/hmp_keggNhi.rds
    Untracked:  data/R_objects/adult_brain/logcpmNhi.rds
    Untracked:  data/R_objects/adult_brain/toptablescqnNhi.rds
    Untracked:  dre00190.pathview.multi.png
    Untracked:  dre00190.png
    Untracked:  dre00190.xml
    Untracked:  dre00511.pathview.multi.png
    Untracked:  dre00511.png
    Untracked:  dre00511.xml
    Untracked:  dre00531.png
    Untracked:  dre00531.xml
    Untracked:  dre03010.png
    Untracked:  dre03010.ribsome.multi.png
    Untracked:  dre03010.xml
    Untracked:  dre04142.pathview.multi.png
    Untracked:  dre04142.png
    Untracked:  dre04142.xml
    Untracked:  output/GEOcounts_adult.out
    Untracked:  output/plots4poster/
    Untracked:  output/plots4talk/
    Untracked:  output/spreadsheets/

Unstaged changes:
    Modified:   analysis/analysis.Rmd
    Modified:   analysis/analysis_adultbrain.rmd
    Modified:   code/plots4pub.Rmd
    Modified:   data/R_objects/adult_brain/celltype.rds
    Modified:   data/R_objects/adult_brain/dge.rds
    Modified:   data/R_objects/adult_brain/hmp_ire.rds
    Modified:   data/R_objects/adult_brain/hmp_kegg.rds
    Modified:   data/R_objects/adult_brain/logcpm.rds
    Modified:   data/R_objects/adult_brain/toptablescqn.rds
    Modified:   data/R_objects/larvae/celltype_larvae.rds
    Modified:   data/R_objects/larvae/dge.rds
    Modified:   data/R_objects/larvae/hmp_ire.rds
    Modified:   data/R_objects/larvae/hmp_kegg.rds
    Modified:   data/R_objects/larvae/logcpm.rds
    Modified:   data/R_objects/larvae/toptablescqn.rds
    Modified:   data/adult_brain/karissas_metadata.xlsx
    Modified:   dre04512.pathview.multi.png

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)
library(magrittr)
library(readxl)

# Bioc
library(ngsReports)
library(AnnotationHub)
library(msigdbr)
library(edgeR)
library(goseq)
library(fgsea)
library(cqn)
library(harmonicmeanp)
library(ssizeRNA)
library(clusterProfiler)
library(variancePartition)

# vis
library(pander)
library(UpSetR)
library(scales)
library(pheatmap)
library(ggpubr)
library(ggrepel)
library(ggfortify)
library(ggforce)
library(RColorBrewer)
library(colorspace)
library(pathview)

theme_set(theme_bw())</code></pre>
<pre class="r"><code>ah &lt;- AnnotationHub() %&gt;%
    subset(species == &quot;Danio rerio&quot;) %&gt;%
    subset(rdataclass == &quot;EnsDb&quot;)

ensDb &lt;- ah[[&quot;AH83189&quot;]] # for release 101, what the ge

grTrans &lt;- transcripts(ensDb)

trLengths &lt;- exonsBy(ensDb, &quot;tx&quot;) %&gt;%
    width() %&gt;%
    vapply(sum, integer(1))

mcols(grTrans)$length &lt;- trLengths[names(grTrans)]

gcGene &lt;- grTrans %&gt;%
  mcols() %&gt;%
  as.data.frame() %&gt;%
  dplyr::select(gene_id, tx_id, gc_content, length) %&gt;%
  as_tibble() %&gt;%
  group_by(gene_id) %&gt;%
  summarise(
    gc_content = sum(gc_content*length) / sum(length),
    length = ceiling(median(length))
  )

grGenes &lt;- genes(ensDb)

mcols(grGenes) %&lt;&gt;%
  as.data.frame() %&gt;%
  left_join(gcGene) %&gt;%
  as.data.frame() %&gt;%
  DataFrame()</code></pre>
<pre class="r"><code># read in metadata and cleanup column names and types. 
meta &lt;- list(larvae = read_excel(&quot;data/larvae/meta/naglu_v_Q96_larvae_metadata.xlsx&quot;, sheet = 3) %&gt;% 
  na.omit() %&gt;% 
  left_join(read_excel(&quot;data/larvae/meta/naglu_v_Q96_larvae_metadata.xlsx&quot;, sheet = 5) %&gt;% 
              mutate(temp1 = str_split(temp, pattern = &quot; &quot;)) %&gt;% 
              mutate(ULN = lapply(temp1, function(x){
                x %&gt;% 
                  .[1]
              }), 
              sample_name =  lapply(temp1, function(x){
                x %&gt;% 
                  .[2]
              }), 
              RIN = lapply(temp1, function(x){
                x %&gt;% 
                  .[4]
              })
              ) %&gt;% 
              unnest() %&gt;% 
              dplyr::select(ULN, sample_name, RIN) %&gt;% 
              na.omit() %&gt;% 
              unique
  ) %&gt;% 
  mutate(Genotype = case_when(
    `naglu genotype` == &quot;wt&quot; &amp; `psen1 genotype` == &quot;wt&quot; ~ &quot;wt&quot;,
    `naglu genotype` == &quot;A603fs/A603fs&quot; &amp; `psen1 genotype` == &quot;wt&quot; ~ &quot;MPS IIIB&quot;,
    `naglu genotype` == &quot;wt&quot; &amp; `psen1 genotype` == &quot;Q96_K97del/+&quot; ~ &quot;EOfAD-like&quot;,
  ) %&gt;% 
    factor(levels = c(&quot;wt&quot;, &quot;MPS IIIB&quot;, &quot;EOfAD-like&quot;)), 
  sample = paste0(Genotype, &quot;_&quot;, ULN), 
  RIN = as.numeric(RIN), 
  age = &quot;7 dpf&quot;
  ) %&gt;% 
    as_tibble(), 
  
  adult = read_excel(&quot;data/adult_brain/karissas_metadata.xlsx&quot;, sheet = &quot;onlyseq&quot;) %&gt;% 
  mutate(Genotype = case_when(
         `usable genotype?` == &quot;wt&quot; ~ &quot;wt&quot;, 
         `usable genotype?` == &quot;EOfAD&quot; ~ &quot;EOfAD-like&quot;, 
         `usable genotype?` == &quot;MPS-III&quot; ~ &quot;MPS IIIB&quot;
  ) %&gt;% 
           factor(levels = c(&quot;wt&quot;, &quot;MPS IIIB&quot;, &quot;EOfAD-like&quot;)), 
  tank = as.factor(tank), 
  sex  = as.factor(sex),
  sample = paste0(fish, &quot;_&quot;, Genotype), 
  age = &quot;6 m adult brain&quot;,
  RIN = as.numeric(`RIN/DIN`)
  ) %&gt;% 
    as_tibble()
)

# read in the outputs of featurecounts. 

featureCounts &lt;- list( 
  larvae = read_delim(&quot;data/larvae/featureCounts/counts.out&quot;, delim = &quot;\t&quot;, skip = 1) %&gt;%
    set_names(basename(names(.))) %&gt;% 
    set_names(names(.) %&gt;% str_remove(pattern = &quot;_S1_merged.Aligned.sortedByCoord.dedup.out.bam&quot;)) %&gt;% 
    as_tibble() %&gt;% 
    dplyr::select(-c(Chr, Start, End, Length, Strand)) %&gt;% 
    gather(key = &quot;ULN&quot;, value = &quot;counts&quot;, starts_with(&quot;21&quot;)) %&gt;% 
    left_join(meta$larvae) %&gt;% 
    dplyr::select(Geneid, counts, sample) %&gt;% 
    spread(key = &quot;sample&quot;, value = &quot;counts&quot;) %&gt;% 
    column_to_rownames(&quot;Geneid&quot;), 
  
  adult = read_delim(&quot;data/adult_brain/05_featureCounts/counts.out&quot;, delim = &quot;\t&quot;, skip = 1) %&gt;%
  set_names(basename(names(.))) %&gt;% 
  set_names(names(.) %&gt;% str_remove(pattern = &quot;_S[0-9]+_merged.Aligned.sortedByCoord.dedup.out.bam&quot;)) %&gt;% 
  as_tibble() %&gt;% 
  dplyr::select(-c(Chr, Start, End, Length, Strand)) %&gt;% 
  gather(key = &quot;ULN&quot;, value = &quot;counts&quot;, starts_with(&quot;22&quot;)) %&gt;% 
  left_join(meta$adult) %&gt;% 
  mutate(sample = paste0(fish, &quot;_&quot;, Genotype)) %&gt;% 
  dplyr::select(Geneid, counts, sample) %&gt;% 
  spread(key = &quot;sample&quot;, value = &quot;counts&quot;) %&gt;% 
  column_to_rownames(&quot;Geneid&quot;) %&gt;% 
  .[,1:24]
)
  
  # write out the cleaned up output for uploading to GEO
  
#write_csv(rownames_to_column(featureCounts$larvae, &quot;gene_id&quot;), &quot;output/GEOcounts.out&quot;)
  
#write_csv(rownames_to_column(featureCounts$adult, &quot;gene_id&quot;), &quot;output/GEOcounts_adult.out&quot;)</code></pre>
<div id="filter-lowly-expressed-genes" class="section level2">
<h2>filter lowly expressed genes</h2>
<p>Genes which are lowly expressed are considered uninformative for de analysis. Here, we set the threshold to be a min CPM of 0.5 (following the 10/min lib size rule proposed by gordon smyth). The effect of filtering is found below</p>
<pre class="r"><code>filteringPlots &lt;- lapply(featureCounts, function(x) {
  
a &lt;- x %&gt;% 
    cpm(log = TRUE) %&gt;%
    as.data.frame() %&gt;% 
  pivot_longer(
    cols = everything(),
    names_to = &quot;sample&quot;,
    values_to = &quot;logCPM&quot;
  ) %&gt;%
  split(f = .$sample) %&gt;%
  lapply(function(x){
    d &lt;- density(x$logCPM)
    tibble(
      sample = unique(x$sample),
      x = d$x,
      y = d$y
    )
  }) %&gt;%
  bind_rows() %&gt;%
  ggplot(aes(x, y, group = sample)) +
  geom_line() +
  labs(
    x = &quot;logCPM&quot;,
    y = &quot;Density&quot;
  )+
  ggtitle(&quot;Before filtering&quot;)


b &lt;- x %&gt;% 
  .[rowSums(cpm(.) &gt;= 0.5) &gt;= 8,] %&gt;% 
    cpm(log = TRUE) %&gt;%
    as.data.frame() %&gt;% 
  pivot_longer(
    cols = everything(),
    names_to = &quot;sample&quot;,
    values_to = &quot;logCPM&quot;
  ) %&gt;%
  split(f = .$sample) %&gt;%
  lapply(function(x){
    d &lt;- density(x$logCPM)
    tibble(
      sample = unique(x$sample),
      x = d$x,
      y = d$y
    )
  }) %&gt;%
  bind_rows() %&gt;%
  ggplot(aes(x, y, group = sample)) +
  geom_line() +
  labs(
    x = &quot;logCPM&quot;,
    y = &quot;Density&quot;,
  )+
  ggtitle(&quot;After filtering&quot;)
ggarrange(a, b)
})</code></pre>
<pre class="r"><code>filteringPlots$larvae +
  ggtitle(&quot;Larvae&quot;)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/analysis_final.rmd/unnamed-chunk-1-1.png" alt="Effect of filtering in larvae" width="75%" height="75%" />
<p class="caption">
Effect of filtering in larvae
</p>
</div>
<pre class="r"><code>filteringPlots$adult </code></pre>
<div class="figure" style="text-align: center">
<img src="figure/analysis_final.rmd/unnamed-chunk-2-1.png" alt="Effect of filtering in adult brain" width="75%" height="75%" />
<p class="caption">
Effect of filtering in adult brain
</p>
</div>
<pre class="r"><code>x.larvae &lt;- featureCounts$larvae %&gt;% 
  as.matrix() %&gt;% 
  .[rowSums(cpm(.) &gt;= 0.5) &gt;= 8,] %&gt;%
  DGEList(
    samples = tibble(sample = colnames(.)) %&gt;%
      left_join(meta$larvae),
    genes = grGenes[rownames(.)] %&gt;%
      as.data.frame() %&gt;%
      dplyr::select(
        chromosome = seqnames, start, end, 
        gene_id, gene_name, gene_biotype, description, entrezid
      ) %&gt;% 
      left_join(gcGene) %&gt;% 
      as_tibble()
  ) %&gt;%
  calcNormFactors()

x.adult &lt;- featureCounts$adult %&gt;% 
  as.matrix() %&gt;% 
 .[rowSums(cpm(.) &gt;= 0.5) &gt;= 8,] %&gt;%
  DGEList(
    samples = tibble(sample = colnames(.)) %&gt;%
      left_join(meta$adult),
    genes = grGenes[rownames(.)] %&gt;%
      as.data.frame() %&gt;%
      dplyr::select(
        chromosome = seqnames, start, end, 
        gene_id, gene_name, gene_biotype, description, entrezid
      ) %&gt;% 
      left_join(gcGene) %&gt;% 
      as_tibble()
  ) %&gt;%
  calcNormFactors()</code></pre>
</div>
<div id="pca" class="section level2">
<h2>PCA</h2>
<p>I first want to explore the overall similarity between samples by PCA. No distinct clusters of samples are observed by genotype in lavrae</p>
<pre class="r"><code># custom function to plot by genotype
plotPCA &lt;- function(dge, colourFactor, shapeFactor, colour) 
{dge %&gt;% 
    cpm(log=TRUE) %&gt;% 
    t() %&gt;% 
    prcomp() %&gt;% 
    autoplot(data = tibble(sample = rownames(.$x)) %&gt;%
               left_join(dge$samples),
             colour = colourFactor,
             shape = shapeFactor, 
             size = 6
    ) +
    theme(aspect.ratio = 1) +
    scale_color_manual(values = colour)
}

# use to make the plots
ggarrange(plotPCA(dge = x.larvae, 
                  colourFactor = &quot;Genotype&quot;, 
                  shapeFactor = 16, # circles
                  colour = viridis_pal(end = 0.95)(3)) + # yelllow hard to see
            ggtitle(&quot;Larvae&quot;), 
          
          plotPCA(dge = x.adult, 
                  colourFactor = &quot;Genotype&quot;, 
                  shapeFactor = &quot;sex&quot;,
                  colour = viridis_pal(end = 0.9)(3)) +
            ggtitle(&quot;Adult&quot;), 
          common.legend = T)</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-3-1.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
<p>There is a clear outlier sample in each experiment. Is it due to the RNA integrity number (RIN)? DOes not appear so.</p>
<pre class="r"><code>plotPCA.RIN &lt;- function(dge) 
{dge %&gt;% 
    cpm(log=TRUE) %&gt;% 
    t() %&gt;% 
    prcomp() %&gt;% 
    autoplot(data = tibble(sample = rownames(.$x)) %&gt;%
               left_join(dge$samples),
             colour = &quot;RIN&quot;,
             size = 6
    ) +
    theme_bw() + 
    theme(aspect.ratio = 1) +
    scale_color_viridis_c(option = &quot;inferno&quot;, end = 0.9)
}

ggarrange(plotPCA.RIN(dge = x.larvae) +
            ggtitle(&quot;Larvae&quot;), 
          
          plotPCA.RIN(dge = x.adult) +
            ggtitle(&quot;Adult&quot;), 
          common.legend = T)</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-4-1.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
<p>What about the initial amount of RNA that was extracted? Maybe for the larvae.</p>
<pre class="r"><code>ggarrange(
  x.larvae %&gt;%     
    cpm(log=TRUE) %&gt;% 
    t() %&gt;% 
    prcomp() %&gt;% 
    autoplot(data = tibble(sample = rownames(.$x)) %&gt;%
               left_join(x.larvae$samples),
             colour = &quot;conc.after.Dnase&quot;,
             size = 6
    ) +
    scale_color_viridis_c(option = &quot;inferno&quot;, end = 0.9) +
    ggtitle(&quot;larvae&quot;), 
  
  x.adult %&gt;% 
    cpm(log=TRUE) %&gt;% 
    t() %&gt;% 
    prcomp() %&gt;% 
    autoplot(data = tibble(sample = rownames(.$x)) %&gt;%
               left_join(x.adult$samples),
             colour = &quot;conc.ng.uL.2&quot;,
             size = 6
    ) +
    scale_color_viridis_c(option = &quot;inferno&quot;, end = 0.9) +
    ggtitle(&quot;adult&quot;)
) &amp;
  theme(legend.position = &quot;top&quot;)</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-5-1.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
</div>
<div id="check-genotypes" class="section level1">
<h1>Check genotypes</h1>
<p>This is in the <code>checkGenotypes.rmd</code> file. All fish carry the genotypes indicated from the PCR genotyping.</p>
<div id="gc-and-length-check" class="section level2">
<h2>%GC and Length check</h2>
<pre class="r"><code>designlist &lt;- list(larvae = model.matrix(~Genotype, data = x.larvae$samples) %&gt;% 
                     set_colnames(gsub(colnames(.), pattern = &quot;Genotype&quot;, replacement = &quot;&quot;)), 
                   adult = model.matrix(~Genotype + tank + sex, data = x.adult$samples %&gt;% 
                                          droplevels()) %&gt;% 
                   set_colnames(gsub(colnames(.), pattern = &quot;Genotype&quot;, replacement = &quot;&quot;))
)

fits_1 &lt;- list(larvae = x.larvae %&gt;% 
                 estimateDisp(designlist$larvae) %&gt;% 
                 glmFit(designlist$larvae), 
               
               adult = x.adult %&gt;% 
                 estimateDisp(designlist$adult) %&gt;% 
                 glmFit(designlist$adult)
)</code></pre>
<pre class="r"><code># Call the toptable
calltoptab &lt;- function(coefs, fit) {
  coefs %&gt;% 
    sapply(function(x){
      glmLRT(fit, coef = x) %&gt;%
        topTags(n = Inf) %&gt;%
        .[[&quot;table&quot;]] %&gt;%
        as_tibble() %&gt;%
        arrange(PValue) %&gt;%
        mutate(
          coef = x,
          DE = FDR &lt; 0.05
        ) %&gt;% 
        dplyr::select(
          gene_name, logFC, logCPM, PValue, FDR, DE, everything()  
        )}, simplify = FALSE)
}

toptables_1 &lt;- list(
  larvae =  calltoptab(coefs = designlist$larvae %&gt;% colnames %&gt;% .[2:3], 
                       fit = fits_1$larvae) %&gt;% 
  sapply(function(x) {mutate(x , age = &quot;larvae&quot;)}, simplify = F),
  
  adult = calltoptab(coefs = designlist$adult %&gt;% colnames %&gt;% .[2:3], 
                     fit = fits_1$adult) %&gt;% 
    sapply(function(x) {mutate(x , age = &quot;adult brain&quot;)}, simplify = F)
)</code></pre>
</div>
</div>
<div id="vis" class="section level1">
<h1>Vis</h1>
<p>These plots are zoomed in between -5 and 5 on the y-axis for vis purposes. the blue lines of best fit are a bit wonky, particulaly for EOfAD in adults. meaning there are some biases here for %GC and Length. Therefore, conditional quantile normailsation is required.</p>
<pre class="r"><code>plotgclen = function(toptable) {
  ggarrange(
  toptable %&gt;% 
    bind_rows() %&gt;% 
    mutate(rankstat = sign(logFC)*-log10(PValue)) %&gt;% 
    ggplot(aes(x = length, y = rankstat)) +
    geom_point(
      aes(colour = DE),
      alpha = 0.5
    ) +
    geom_smooth(se = FALSE, method = &quot;gam&quot;) +
    facet_grid(rows = vars(coef)) +
    theme(legend.position = &quot;none&quot;) +
    scale_color_manual(values = c(&quot;grey50&quot;, &quot;red&quot;)) +
    scale_x_log10()+
    labs(x = &quot;Average transcript length per gene&quot;,
         colour = &quot;Differentially expressed?&quot;,
         y = &quot;sign(logFC)*-log10(PValue)&quot;) +
    coord_cartesian(ylim = c(-5, 5)),
  
  toptable %&gt;% 
    bind_rows() %&gt;% 
  mutate(rankstat = sign(logFC)*-log10(PValue)) %&gt;% 
  ggplot(aes(x = gc_content, y = rankstat)) +
  geom_point(
    aes(colour = DE),
    alpha = 0.5
  ) +
  geom_smooth(se = FALSE, method = &quot;gam&quot;) +
  facet_grid(rows = vars(coef)) +
  theme_bw() +
  theme(legend.position = &quot;none&quot;) +
  scale_color_manual(values = c(&quot;grey50&quot;, &quot;red&quot;)) +
  scale_x_log10()+
  labs(x = &quot;Average transcript %GC per gene&quot;,
       colour = &quot;Differentially expressed?&quot;,
       y = &quot;sign(logFC)*-log10(PValue)&quot;) +
  coord_cartesian(ylim = c(-5, 5))
  )
  }</code></pre>
<pre class="r"><code>plotgclen(toptables_1$larvae)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/analysis_final.rmd/unnamed-chunk-9-1.png" alt="larvae" width="75%" height="75%" />
<p class="caption">
larvae
</p>
</div>
<pre class="r"><code>plotgclen(toptables_1$adult)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/analysis_final.rmd/unnamed-chunk-10-1.png" alt="adult" width="75%" height="75%" />
<p class="caption">
adult
</p>
</div>
</div>
<div id="cqn" class="section level1">
<h1>CQN</h1>
<p>A little bit of bias is obsevred for %GC content and gene length, so will apply the <code>cqn</code> correction to adjust for this.</p>
<pre class="r"><code>runCQN &lt;- function(dge) {
  dge %&gt;% 
    with(
      cqn(
        counts = counts,
        x = genes$gc_content,
        lengths = genes$length,
        sizeFactors = samples$lib.size
      )
    )
}

cqns &lt;- list(larvae = runCQN(dge = x.larvae), 
             adult = runCQN(dge = x.adult)
)

logCPMs &lt;-  list(larvae = cqns$larvae$y + cqns$larvae$offset, 
                 adult = cqns$adult$y + cqns$adult$offset
)</code></pre>
<div id="plot-out-the-model-fits" class="section level3">
<h3>plot out the model fits</h3>
<pre class="r"><code>par(mfrow = c(2, 2))
cqnplot(cqns$larvae, n = 1, xlab = &quot;GC Content&quot;)
cqnplot(cqns$larvae, n = 2, xlab = &quot;Length&quot;)
cqnplot(cqns$adult, n = 1, xlab = &quot;GC Content&quot;)
cqnplot(cqns$adult, n = 2, xlab = &quot;Length&quot;)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/analysis_final.rmd/unnamed-chunk-12-1.png" alt="*Model fits for GC content and gene length under the CQN model. Variability is clearly visible at either end*" width="75%" height="75%" />
<p class="caption">
<em>Model fits for GC content and gene length under the CQN model. Variability is clearly visible at either end</em>
</p>
</div>
<pre class="r"><code>par(mfrow = c(1, 1))</code></pre>
</div>
</div>
<div id="pca-after-cqn" class="section level1">
<h1>PCA after cqn</h1>
<p>PCA analysis was repeated. No significant improvement of clustering of samples by genotype is observed. But they appear a bit closer than before, indicating some variability was explained and removed.</p>
<pre class="r"><code>ggarrange(
  logCPMs$larvae %&gt;% 
    t() %&gt;%
    prcomp() %&gt;% 
    autoplot(data = tibble(sample = rownames(.$x)) %&gt;%
               left_join(x.larvae$samples),
             colour = &quot;Genotype&quot;, 
             
             size = 4
    ) +
    scale_color_discrete_qualitative(palette = &quot;Dark 3&quot;) +
    ggtitle(&quot;Larvae&quot;),
  
    logCPMs$adult %&gt;% 
    t() %&gt;%
    prcomp() %&gt;% 
    autoplot(data = tibble(sample = rownames(.$x)) %&gt;%
               left_join(x.adult$samples),
             colour = &quot;Genotype&quot;, 
             shape = &quot;sex&quot;,
             size = 4
    ) +
    scale_color_discrete_qualitative(palette = &quot;Dark 3&quot;) +
    ggtitle(&quot;Adult&quot;),
  common.legend = T, 
  labels = &quot;AUTO&quot;
)</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-13-1.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
<div id="de-with-cqn" class="section level2">
<h2>DE with cqn</h2>
<p>The DE analysis was then repeated, this time including the offset generated by <code>cqn</code> in the model.</p>
<pre class="r"><code># Add the offset to the dge object
x.larvae$offset &lt;- cqns$larvae$glm.offset
x.adult$offset &lt;- cqns$adult$glm.offset

fits_cqn &lt;- list(larvae = x.larvae %&gt;% 
                 estimateDisp(designlist$larvae) %&gt;% 
                 glmFit(designlist$larvae), 
               
               adult = x.adult %&gt;% 
                 estimateDisp(designlist$adult) %&gt;% 
                 glmFit(designlist$adult)
)

# Call the toptable
calltoptab &lt;- function(coefs, fit) {
  coefs %&gt;% 
    sapply(function(x){
      glmLRT(fit, coef = x) %&gt;%
        topTags(n = Inf) %&gt;%
        .[[&quot;table&quot;]] %&gt;%
        as_tibble() %&gt;%
        arrange(PValue) %&gt;%
        mutate(
          coef = x,
          DE = FDR &lt; 0.05
        ) %&gt;% 
        dplyr::select(
          gene_name, logFC, logCPM, PValue, FDR, DE, everything()  
        )}, simplify = FALSE)
}

toptables_cqn &lt;- list(
  larvae =  calltoptab(coefs = designlist$larvae %&gt;% colnames %&gt;% .[2:3], 
                       fit = fits_cqn$larvae) %&gt;% 
  sapply(function(x) {mutate(x , age = &quot;larvae&quot;)}, simplify = F),
  
  adult = calltoptab(coefs = designlist$adult %&gt;% colnames %&gt;% .[2:3], 
                     fit = fits_cqn$adult) %&gt;% 
    sapply(function(x) {mutate(x , age = &quot;adult brain&quot;)}, simplify = F)
)

toptables_cqn %&lt;&gt;% 
  lapply(function(x) {
    x %&gt;% 
      lapply(function(y) {
        y %&gt;% 
          mutate(age = factor(age, levels = c(&quot;larvae&quot;, &quot;adult brain&quot;)))
      })
  })</code></pre>
<div id="vis-1" class="section level3">
<h3>Vis</h3>
<div id="recheck-the-rank-stat" class="section level4">
<h4>recheck the rank-stat</h4>
<p>The bias is still not perfect, but has improved from before cqn.</p>
<pre class="r"><code>plotgclen(toptables_cqn$larvae)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/analysis_final.rmd/unnamed-chunk-15-1.png" alt="larvae" width="75%" height="75%" />
<p class="caption">
larvae
</p>
</div>
<pre class="r"><code>plotgclen(toptables_cqn$adult)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/analysis_final.rmd/unnamed-chunk-16-1.png" alt="adult" width="75%" height="75%" />
<p class="caption">
adult
</p>
</div>
</div>
<div id="number-of-degs" class="section level4">
<h4>number of DEGs</h4>
<p>At both ages, the number of DEGs is greater in the MPS IIIB larvae c.f. EOfAD-like. This is consistent with MPS III being a more severe neurodegenerative disease.</p>
<pre class="r"><code>toptables_cqn$larvae %&gt;% 
  bind_rows() %&gt;% 
  bind_rows(toptables_cqn$adult %&gt;% 
              bind_rows()) %&gt;% 
  
  mutate(Direction = case_when(
    sign(logFC) == &quot;1&quot; ~ &quot;up&quot;,
    sign(logFC) == &quot;-1&quot; ~ &quot;down&quot;
  )) %&gt;% 
  dplyr::filter(DE == TRUE) %&gt;%
  ggplot(aes(y = `coef`)) +
  geom_bar(stat = &quot;count&quot;,
           width = 0.5,
           aes(fill = Direction),
           position = &quot;dodge&quot;) +
  facet_wrap(~age) +
  theme(legend.position = &quot;bottom&quot;) +
  ggtitle(&quot;Number of DE genes in each comparison&quot;) +
  labs(x = &quot;Number of DE genes&quot;,
       y = &quot;Contrast&quot;,
       colour = &quot;Direction of change&quot;, 
       title = &quot;Number of DE genes in each comparison&quot;)</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-17-1.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
</div>
<div id="md-plots" class="section level4">
<h4>MD plots</h4>
<p>MD plots look normal. The blue smooth line is mostly along the y = 0 line.</p>
<pre class="r"><code>toptables_cqn$larvae %&gt;% 
  bind_rows() %&gt;% 
  bind_rows(toptables_cqn$adult %&gt;% 
              bind_rows()) %&gt;% 
  ggplot(aes(x = logCPM, y = logFC)) +
  geom_point(
    aes(colour = DE),
    alpha = 0.5) +
  geom_smooth(se = F) +
  facet_wrap(~age + coef) +
  scale_color_manual(values = c(&quot;grey50&quot;, &quot;red&quot;))</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-18-1.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
</div>
<div id="volcano-plots" class="section level4">
<h4>volcano plots</h4>
<p>The shapes of the volcano plots below are a bit weird. This is beavcuse there are some genes which have very low logFC and/or pvalues.</p>
<pre class="r"><code>toptables_cqn$larvae %&gt;% 
  bind_rows() %&gt;% 
  bind_rows(toptables_cqn$adult %&gt;% 
              bind_rows()) %&gt;% 
  ggplot(aes(x = logFC, y = -log10(PValue))) +
  geom_point(
    aes(colour = DE),
    alpha = 0.5) +
  facet_wrap(~age + coef) +
  scale_color_manual(values = c(&quot;grey50&quot;, &quot;red&quot;))</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-19-1.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
<p>The volcano plots below are zoomed in (not filtered).</p>
<pre class="r"><code>toptables_cqn$larvae %&gt;% 
  bind_rows() %&gt;% 
  bind_rows(toptables_cqn$adult %&gt;% 
              bind_rows()) %&gt;% 
  ggplot(aes(x = logFC, y = -log10(PValue))) +
  geom_point(
    aes(colour = DE),
    alpha = 0.5) +
  facet_wrap(~age + coef) +
  coord_cartesian(xlim = c(-5,5), 
                  ylim = c(0, 20)) +
  scale_color_manual(values = c(&quot;grey50&quot;, &quot;red&quot;))</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-20-1.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
<p>There is a weird line showing up in the EOfAD-like larvae. I’m not really sure what this is.</p>
<pre class="r"><code>toptables_cqn$larvae$`EOfAD-like` %&gt;% 
  ggplot(aes(x = logFC, y = -log10(PValue))) +
  geom_point(aes(colour = logFC   &gt; -2),
             alpha = 0.5) +
  geom_label_repel(aes(label = gene_name), 
                   fill = NA,
                   max.overlaps = 20,
                   data = . %&gt;% 
                     dplyr::filter(logFC &lt; -2)) +
  coord_cartesian(xlim = c(-7,5), 
                  ylim = c(0, 6)) +
  scale_color_manual(values = c(&quot;red&quot;, &quot;grey50&quot;))</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-21-1.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
<p>Maybe these genes are all coming from chr17. But this is not the case, seems to be a few different chromosomes there</p>
<pre class="r"><code>toptables_cqn$larvae$`EOfAD-like` %&gt;% 
  ggplot(aes(x = logFC, y = -log10(PValue))) +
  geom_point(aes(colour = chromosome),
             alpha = 0.5) +
  facet_wrap(~age + coef) +
  coord_cartesian(xlim = c(-7,5), 
                  ylim = c(0, 6)) </code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-22-1.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
<p>Also looked at %GC and length but this doesn’t seem to be explaining the weird pattern.</p>
<pre class="r"><code>ggarrange(toptables_cqn$larvae$`EOfAD-like` %&gt;% 
  ggplot(aes(x = logFC, y = -log10(PValue))) +
  geom_point(aes(colour = gc_content),
             alpha = 0.5) +
  facet_wrap(~age + coef) +
  coord_cartesian(xlim = c(-7,5), 
                  ylim = c(0, 6)) +
  scale_color_viridis_b(),

toptables_cqn$larvae$`EOfAD-like` %&gt;% 
  ggplot(aes(x = logFC, y = -log10(PValue))) +
  geom_point(aes(colour = length),
             alpha = 0.5) +
  facet_wrap(~age + coef) +
  coord_cartesian(xlim = c(-7,5), 
                  ylim = c(0, 6)) +
  scale_color_viridis_b()
)</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-23-1.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
<p>Perhaps they are a GO term. Will come back and check this</p>
</div>
</div>
</div>
<div id="gene-set-enrichment-analysis" class="section level2">
<h2>Gene set enrichment analysis</h2>
<p>I now will perform some gene set enrichment analyses on MSIGDB gene sets.</p>
<div id="gene-sets" class="section level3">
<h3>Gene sets</h3>
<p>I will download the KEGG and GO term gene sets from MSigDB. GO terms were filtered based whether they showed &lt; 3 steps back to the root node. The IRE genes are obtained from Hin et al. 2021 (DOI: 10.3233/JAD-210200). The cell type markers are from Farnsworth et al. for larvae and Jiang et al. 2021 (<a href="https://doi.org/10.3389/fcell.2021.743421" class="uri">https://doi.org/10.3389/fcell.2021.743421</a>) for adult brain.</p>
<pre class="r"><code>KEGG.notfiltered &lt;- 
  msigdbr(&quot;Danio rerio&quot;, category = &quot;C2&quot;, subcategory = &quot;CP:KEGG&quot;) %&gt;% 
  distinct(gs_name, ensembl_gene, .keep_all = TRUE) %&gt;%
  split(f = .$gs_name) %&gt;%
  lapply(extract2, &quot;ensembl_gene&quot;)

KEGG.adult &lt;- KEGG.notfiltered %&gt;% 
  lapply(function(x) {
    x %&gt;% 
      as_tibble() %&gt;% 
      set_colnames(&quot;gene_id&quot;) %&gt;% 
      dplyr::filter(gene_id %in% rownames(x.adult)) %&gt;% 
      .$gene_id
  })

KEGG.larvae &lt;- KEGG.notfiltered %&gt;% 
  lapply(function(x) {
    x %&gt;% 
      as_tibble() %&gt;% 
      set_colnames(&quot;gene_id&quot;) %&gt;% 
      dplyr::filter(gene_id %in% rownames(x.larvae)) %&gt;% 
      .$gene_id
  })</code></pre>
<pre class="r"><code># GO Terms
goSummaries &lt;- url(&quot;https://uofabioinformaticshub.github.io/summaries2GO/data/goSummaries.RDS&quot;) %&gt;%
  readRDS() %&gt;%
  mutate(
    Term = Term(id),
    gs_name = Term %&gt;% str_to_upper() %&gt;% str_replace_all(&quot;[ -]&quot;, &quot;_&quot;),
    gs_name = paste0(&quot;GO_&quot;, gs_name)
    )

minPath &lt;- 3

GO.notfiltered &lt;-
  msigdbr(&quot;Danio rerio&quot;, category = &quot;C5&quot;) %&gt;%
  dplyr::filter(grepl(gs_name, pattern = &quot;^GO&quot;)) %&gt;%
  mutate(gs_name = str_replace(gs_name, pattern = &quot;GOBP_&quot;, replacement = &quot;GO_&quot;)) %&gt;%
  mutate(gs_name = str_replace(gs_name, pattern = &quot;GOMF_&quot;, replacement = &quot;GO_&quot;)) %&gt;%
  mutate(gs_name = str_replace(gs_name, pattern = &quot;GOCC_&quot;, replacement = &quot;GO_&quot;)) %&gt;%
  left_join(goSummaries) %&gt;%
  dplyr::filter(shortest_path &gt;= minPath) %&gt;%
  distinct(gs_name, ensembl_gene, .keep_all = TRUE) %&gt;%
  dplyr::select(gs_name, ensembl_gene) %&gt;% 
  na.omit %&gt;% 
  split(f = .$gs_name) %&gt;%
  lapply(extract2, &quot;ensembl_gene&quot;)

GO.larvae &lt;- GO.notfiltered %&gt;% 
  lapply(function(x) {
    x %&gt;% 
      as_tibble() %&gt;% 
      set_colnames(&quot;gene_id&quot;) %&gt;% 
      dplyr::filter(gene_id %in% rownames(x.larvae)) %&gt;% 
      .$gene_id
  })

GO.adult &lt;- GO.notfiltered %&gt;% 
  lapply(function(x) {
    x %&gt;% 
      as_tibble() %&gt;% 
      set_colnames(&quot;gene_id&quot;) %&gt;% 
      dplyr::filter(gene_id %in% rownames(x.adult)) %&gt;% 
      .$gene_id
  })</code></pre>
<pre class="r"><code>ireGenes.notfiltered &lt;- readRDS(&quot;data/gene_sets/zebrafishireGenes.rds&quot;) %&gt;%
  unlist() %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;ire&quot;) %&gt;%
  as_tibble() %&gt;%
  mutate(ire = str_extract(ire, pattern = &quot;ire[:digit:]_[:alpha:]+&quot;)) %&gt;%
  set_colnames(c(&quot;ire&quot;, &quot;gene_id&quot;)) %&gt;%
  split(f = .$ire) %&gt;%
  lapply(magrittr::extract2,&quot;gene_id&quot;)

ire.adult &lt;- ireGenes.notfiltered %&gt;% 
  lapply(function(x) {
    x %&gt;% 
      as_tibble() %&gt;% 
      set_colnames(&quot;gene_id&quot;) %&gt;% 
      dplyr::filter(gene_id %in% rownames(x.adult)) %&gt;% 
      .$gene_id
  })

ire.larvae &lt;- ireGenes.notfiltered %&gt;% 
  lapply(function(x) {
    x %&gt;% 
      as_tibble() %&gt;% 
      set_colnames(&quot;gene_id&quot;) %&gt;% 
      dplyr::filter(gene_id %in% rownames(x.larvae)) %&gt;% 
      .$gene_id
  })</code></pre>
<pre class="r"><code># Cell type markers for zerbafish larvae
cell_type_markerslarvae &lt;- read_excel(&quot;data/gene_sets/1-s2.0-S0012160619304919-mmc7.xlsx&quot;) %&gt;%
  dplyr::filter(grepl(`Day of Origin`, pattern = 5 )) %&gt;%  # restrict to only 5dpf
  dplyr::select(Tissue, 9:24) %&gt;%
  split(f = .$Tissue) %&gt;%
  lapply(function(y) {
    y %&gt;%
      dplyr::select(-Tissue) %&gt;%
      gather %&gt;%
      dplyr::select(value) %&gt;%
      set_colnames(&quot;gene_name&quot;) %&gt;%
      left_join(x.larvae$genes) %&gt;%
      na.omit %&gt;%
      .$gene_id
  })


# cell type markers for adult zebrafish brain
cell_type_markers.adultbrain &lt;- 
  read_xlsx(&quot;data/gene_sets/Suppdata2_jiangetal_2021_fcelldev.xlsx&quot;, sheet = &quot;Brain&quot;) %&gt;% 
  dplyr::rename(&quot;gene_name&quot; = gene) %&gt;% 
  left_join(x.adult$genes) %&gt;% 
  dplyr::filter(gene_id %in% rownames(x.adult)) %&gt;% 
  split(f = .$`cell type`) %&gt;% 
  lapply(extract2, &quot;gene_id&quot;)</code></pre>
</div>
</div>
<div id="check-whether-the-weird-tail-in-the-volcano-plot-is-a-go-term" class="section level2">
<h2>check whether the weird tail in the volcano plot is a GO term</h2>
<p>I iused enrichr to see whether the 16 genes in the EOfAD-like larvae with a logFC &lt; -2 are significantly over-represented with any GO terms. Enrichr uses a hypergeometric test which is not the best, but is sufficient for this case. 5 genes have the GO term <code>Response to virus</code>. So this is probably not the source of the strange pattern . I would be expecting all 15 or so to be there. I wonder if this is just an artifact.</p>
<pre class="r"><code>overrep &lt;- enricher(
  gene = toptables_cqn$larvae$`EOfAD-like` %&gt;% 
    dplyr::filter(logFC &lt; -2) %&gt;% 
    .$gene_id,
  universe = toptables_cqn$larvae$`EOfAD-like`$gene_id,
  minGSSize = 5,
  maxGSSize = 500,
  qvalueCutoff = 0.2,
  TERM2GENE = msigdbr(&quot;Danio rerio&quot;, category = &quot;C5&quot;) %&gt;%
    dplyr::filter(grepl(gs_name, pattern = &quot;^GO&quot;)) %&gt;%
    mutate(gs_name = str_replace(gs_name, pattern = &quot;GOBP_&quot;, replacement = &quot;GO_&quot;)) %&gt;%
    mutate(gs_name = str_replace(gs_name, pattern = &quot;GOMF_&quot;, replacement = &quot;GO_&quot;)) %&gt;%
    mutate(gs_name = str_replace(gs_name, pattern = &quot;GOCC_&quot;, replacement = &quot;GO_&quot;)) %&gt;%
    left_join(goSummaries) %&gt;%
    dplyr::filter(shortest_path &gt;= minPath) %&gt;%
    distinct(gs_name, ensembl_gene, .keep_all = TRUE) %&gt;%
    dplyr::select(gs_name, ensembl_gene) %&gt;% 
    dplyr::filter(ensembl_gene %in% toptables_cqn$larvae$`EOfAD-like`$gene_id) %&gt;% 
    na.omit
)

dotplot(overrep)</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-28-1.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
<p>I then plotted out the volcano plot again, this time colouring each gene by whether it is in the “Response to virus” gene set. The genes which have this GO term are scattered throughout all genes and are not particularly condensed in the line.</p>
<pre class="r"><code>toptables_cqn$larvae$`EOfAD-like` %&gt;% 
  ggplot(aes(x = logFC, y = -log10(PValue))) +
  geom_point(aes(colour = gene_id %in% GO.notfiltered$GO_RESPONSE_TO_VIRUS),
             alpha = 0.5) +
  # geom_label_repel(aes(label = gene_name), 
  #                  fill = NA,
  #                  max.overlaps = 20,
  #                  data = . %&gt;% 
  #                    dplyr::filter(logFC &lt; -2)) +
  coord_cartesian(xlim = c(-7,5), 
                  ylim = c(0, 6)) +
  scale_color_manual(values = c( &quot;grey50&quot;, &quot;red&quot;))</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-29-1.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="over-representation-analysis" class="section level1">
<h1>Over-representation analysis</h1>
<div id="goseq" class="section level2">
<h2>goseq</h2>
<p>I will test for over-representation of GO terms within the DE genes in each comparison. I will do this using goseq as it allows you to include a co-variate which is nice. Since there seems to still be a little bit of bias remaining for %GC content even after cqn, this is what I will speify as the covariate.</p>
<p>There were no DEGs in the EOfAD like mutants in the larvae, so no GO testing will be performed for this comparison.</p>
<p>The top 10 GO terms over-represented in the DEGs in MPS-IIIB are plotted below.</p>
<pre class="r"><code>pwfs &lt;- list(larvae.MPSIIIB = toptables_cqn$larvae$`MPS IIIB` %&gt;%
               with(
                 nullp(
                   DEgenes = structure(DE, names = gene_id),
                   bias.data = gc_content,
                 )
               ), 
             adult.MPSIIIB = toptables_cqn$adult$`MPS IIIB` %&gt;%
               with(
                 nullp(
                   DEgenes = structure(DE, names = gene_id),
                   bias.data = gc_content,
                 )
               ), 
             adult.EOfADlike = toptables_cqn$adult$`EOfAD-like` %&gt;%
               with(
                 nullp(
                   DEgenes = structure(DE, names = gene_id),
                   bias.data = gc_content,
                 )
               )
)</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-30-1.png" width="75%" height="75%" style="display: block; margin: auto;" /><img src="figure/analysis_final.rmd/unnamed-chunk-30-2.png" width="75%" height="75%" style="display: block; margin: auto;" /><img src="figure/analysis_final.rmd/unnamed-chunk-30-3.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
<pre class="r"><code>rungoseq = function(pwf, go) {
  
  goseq(pwf = pwf, 
        gene2cat = go) %&gt;% 
    
    as_tibble() %&gt;%
    dplyr::filter(numDEInCat &gt; 0) %&gt;%
    mutate(FDR = p.adjust(over_represented_pvalue, method = &quot;fdr&quot;)) %&gt;%
    dplyr::select(-under_represented_pvalue)
  
}

goseq &lt;- list(
  larvae.MPSIIIB = rungoseq(pwf = pwfs$larvae.MPSIIIB, 
                            go = GO.larvae), 
  adult.MPSIIIB = rungoseq(pwf = pwfs$adult.MPSIIIB, 
                            go = GO.adult), 
  adult.EOfADlike = rungoseq(pwf = pwfs$adult.EOfADlike, 
                            go = GO.adult)
)
  

goseq$adult.MPSIIIB %&gt;% 
  dplyr::slice(1:60) %&gt;%
  mutate(propDE = numDEInCat/numInCat) %&gt;%
  ggplot(aes(x = -log10(over_represented_pvalue), y = reorder(category, -over_represented_pvalue))) +
  geom_col(aes(fill = propDE, alpha = FDR &lt; 0.05)) +
  scale_fill_viridis_c() +
  scale_alpha_manual(values = c(0.4, 1)) +
  labs(y = &quot;GO Term&quot;,
       fill = &quot;Proportion of \nDE genes in \nGO term&quot;)</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-30-4.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
<pre class="r"><code>goseq$adult.EOfADlike %&gt;% 
  dplyr::slice(1:10) %&gt;%
  mutate(propDE = numDEInCat/numInCat) %&gt;%
  ggplot(aes(x = -log10(over_represented_pvalue), y = reorder(category, -over_represented_pvalue))) +
  geom_col(aes(fill = propDE, alpha = FDR &lt; 0.05)) +
  scale_fill_viridis_c() +
  scale_alpha_manual(values = c(0.4, 1)) +
  labs(y = &quot;GO Term&quot;,
       fill = &quot;Proportion of \nDE genes in \nGO term&quot;)</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-30-5.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
<pre class="r"><code>goseq$larvae.MPSIIIB %&gt;% 
  dplyr::slice(1:10) %&gt;%
  mutate(propDE = numDEInCat/numInCat) %&gt;%
  ggplot(aes(x = -log10(over_represented_pvalue), y = reorder(category, -over_represented_pvalue))) +
  geom_col(aes(fill = propDE, alpha = FDR &lt; 0.05)) +
  scale_fill_viridis_c() +
  scale_alpha_manual(values = c(0.4, 1)) +
  labs(y = &quot;GO Term&quot;,
       fill = &quot;Proportion of \nDE genes in \nGO term&quot;)</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-30-6.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
</div>
<div id="ranked-list-with-ensemble-approach" class="section level2">
<h2>Ranked list with ensemble approach</h2>
<p>Over-represenation analysis using goseq relies on a hard threshold for gene to be called DE. This can lose information as there are genes which are almost DE (i.e. FDRp = 0.06), but are not interpreted in this way.</p>
<p>Another method to test for enrichment of gene sets is to use ROAST. Roast uses a rotation-based approach to generate a null distribution of gene set enrichment scores by randomly permuting the samples or the gene labels in the data. Roast then calculates a gene set enrichment score for each gene set in the data, based on the correlation between the gene expression values and the sample labels. The gene set enrichment scores are compared to the null distribution to calculate the statistical significance of the enrichment. In this way, it does not requre a list of DEGs as input.</p>
<p>Here, I will perform the fast approx of ROAST: fry, from the limma package on the KEGG gene sets. Note I will filter the KEGG gene sets to only retain genes considered detectable in each experiment.</p>
<pre class="r"><code>designlist$adult %&gt;% colnames() %&gt;% .[2:3] %&gt;%
  sapply(function(y) {
  logCPMs$adult %&gt;%
      fry(
        index = c(KEGG.notfiltered, cell_type_markers.adultbrain),
        design = designlist$adult,
        contrast = y,
        sort = &quot;mixed&quot;
      ) %&gt;%
      rownames_to_column(&quot;pathway&quot;) %&gt;%
      as_tibble() %&gt;%
      mutate(coef = y)
  }, simplify = FALSE)</code></pre>
<pre><code>$`MPS IIIB`
# A tibble: 209 × 8
   pathway         NGenes Direction  PValue     FDR PValue.Mixed FDR.Mixed coef 
   &lt;chr&gt;            &lt;int&gt; &lt;chr&gt;       &lt;dbl&gt;   &lt;dbl&gt;        &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;
 1 KEGG_LYSOSOME      116 Up        1.25e-5 1.30e-3     8.27e-15  1.73e-12 MPS …
 2 Macrophage_grn…    137 Up        7.01e-4 1.83e-2     9.70e-14  1.01e-11 MPS …
 3 KEGG_AMINO_SUG…     39 Up        2.45e-2 2.13e-1     5.22e-12  3.64e-10 MPS …
 4 KEGG_OTHER_GLY…     14 Up        1.37e-6 2.87e-4     4.29e-10  2.24e- 8 MPS …
 5 KEGG_GLYCOSPHI…     16 Up        3.17e-1 7.21e-1     5.40e- 9  2.26e- 7 MPS …
 6 Neural stem ce…    121 Down      2.22e-4 1.17e-2     1.68e- 8  5.86e- 7 MPS …
 7 KEGG_GLYCOSAMI…     20 Up        2.59e-4 1.17e-2     4.65e- 8  1.39e- 6 MPS …
 8 Microglia_apoc…     89 Down      9.78e-3 1.14e-1     9.52e- 8  2.49e- 6 MPS …
 9 Oligodendrocyt…     51 Down      4.96e-4 1.48e-2     1.71e- 7  3.96e- 6 MPS …
10 KEGG_SNARE_INT…     36 Down      3.09e-1 7.11e-1     1.95e- 6  4.08e- 5 MPS …
# ℹ 199 more rows

$`EOfAD-like`
# A tibble: 209 × 8
   pathway            NGenes Direction PValue   FDR PValue.Mixed FDR.Mixed coef 
   &lt;chr&gt;               &lt;int&gt; &lt;chr&gt;      &lt;dbl&gt; &lt;dbl&gt;        &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;
 1 KEGG_LYSOSOME         116 Up        0.0185 0.507       0.0107     0.847 EOfA…
 2 KEGG_STARCH_AND_S…     26 Up        0.142  0.631       0.0124     0.847 EOfA…
 3 KEGG_GLYCOSAMINOG…     20 Up        0.0149 0.507       0.0175     0.847 EOfA…
 4 KEGG_HYPERTROPHIC…     73 Down      0.0296 0.507       0.0407     0.847 EOfA…
 5 KEGG_COMPLEMENT_A…     32 Down      0.278  0.691       0.0443     0.847 EOfA…
 6 KEGG_ARACHIDONIC_…     28 Up        0.792  0.925       0.0519     0.847 EOfA…
 7 KEGG_DILATED_CARD…     80 Down      0.0293 0.507       0.0610     0.847 EOfA…
 8 KEGG_BIOSYNTHESIS…     17 Up        0.748  0.925       0.0648     0.847 EOfA…
 9 KEGG_ALANINE_ASPA…     29 Up        0.534  0.846       0.0689     0.847 EOfA…
10 KEGG_GLUTATHIONE_…     35 Up        0.0748 0.625       0.0766     0.847 EOfA…
# ℹ 199 more rows</code></pre>
</div>
<div id="harmonic-mean-p" class="section level2">
<h2>Harmonic mean p</h2>
<p>Since I’m not detecting any significant gene sets in the EOfAD-like mutants, I need to try another method. I will use the harmonic mean p val method which combines indivisual p vals from 3 gene set enrichmen analysis methods: fry, camera and gsea.</p>
<pre class="r"><code># Make a custom function for calculating the HMP.
runHMP = function(gene.set.obj,
                  design,
                  logCPM.df,
                  toptable) {
fry &lt;-
  design %&gt;% colnames() %&gt;% .[2:3] %&gt;%
  sapply(function(y) {
  logCPM.df %&gt;%
      fry(
        index = gene.set.obj,
        design = design,
        contrast = y,
        sort = &quot;mixed&quot;
      ) %&gt;%
      rownames_to_column(&quot;pathway&quot;) %&gt;%
      as_tibble() %&gt;%
      mutate(coef = y)
  }, simplify = FALSE)

camera &lt;-
  design %&gt;% colnames() %&gt;% .[2:3] %&gt;%
  sapply(function(y) {
    logCPM.df %&gt;%
      camera(
        index = gene.set.obj,
        design = design,
        contrast = y
      ) %&gt;%
      rownames_to_column(&quot;pathway&quot;) %&gt;%
      as_tibble() %&gt;%
      mutate(coef = y)
  }, simplify = FALSE)

ranks &lt;-
   sapply(toptable, function(y) {
     y %&gt;%
       mutate(rankstat = sign(logFC) * log10(1/PValue)) %&gt;%
       arrange(rankstat) %&gt;%
       dplyr::select(c(&quot;gene_id&quot;, &quot;rankstat&quot;)) %&gt;% #only want the Pvalue with sign
       with(structure(rankstat, names = gene_id)) %&gt;%
       rev() # reverse so the start of the list is upregulated genes
   }, simplify = FALSE)

# Run fgsea

# set a seed for a reproducible result
set.seed(33)
fgsea &lt;- ranks %&gt;%
  sapply(function(x){
    fgseaMultilevel(stats = x, pathways = gene.set.obj) %&gt;%
      as_tibble() %&gt;%
      dplyr::rename(FDR = padj) %&gt;%
      mutate(padj = p.adjust(pval, &quot;bonferroni&quot;)) %&gt;%
      dplyr::select(pathway, pval, FDR, padj, everything()) %&gt;%
      arrange(pval) %&gt;%
      mutate(sig = padj &lt; 0.05)
  }, simplify = F)

fgsea$`MPS IIIB` %&lt;&gt;%
  mutate(coef = &quot;MPS IIIB&quot;)
fgsea$`EOfAD-like` %&lt;&gt;%
  mutate(coef = &quot;EOfAD-like&quot;)

fry %&gt;%
  bind_rows() %&gt;%
  dplyr::select(pathway, PValue.Mixed, coef) %&gt;%
  dplyr::rename(fry_p = PValue.Mixed) %&gt;%
  left_join(camera %&gt;%
              bind_rows() %&gt;%
              dplyr::select(pathway, PValue, coef),
            by = c(&quot;pathway&quot;, &quot;coef&quot;)) %&gt;%
  dplyr::rename(camera_p = PValue) %&gt;%
  left_join(fgsea %&gt;%
              bind_rows() %&gt;%
              dplyr::select(pathway, pval, coef),
            by = c(&quot;pathway&quot;, &quot;coef&quot;)) %&gt;%
  dplyr::rename(fgsea_p = pval) %&gt;%
  bind_rows() %&gt;%
  nest(p = one_of(c(&quot;fry_p&quot;, &quot;camera_p&quot;, &quot;fgsea_p&quot;))) %&gt;%
 mutate(harmonic_p = vapply(p, function(x){
        x &lt;- unlist(x)
        x &lt;- x[!is.na(x)]
        p.hmp(x, L = 4)
      }, numeric(1))
 ) %&gt;%
  unnest() %&gt;%
  mutate(harmonic_p_FDR = p.adjust(harmonic_p, &quot;fdr&quot;),
         sig = harmonic_p_FDR &lt; 0.05) %&gt;%
  arrange(harmonic_p)

}</code></pre>
<div id="kegg" class="section level3">
<h3>KEGG</h3>
<div id="larvae" class="section level4">
<h4>larvae</h4>
<pre class="r"><code>HMP.KEGG.larvae &lt;- runHMP(gene.set.obj = KEGG.larvae,
                   design = designlist$larvae,
                   logCPM.df = logCPMs$larvae,
                   toptable = toptables_cqn$larvae)

sig.paths.larvae &lt;- HMP.KEGG.larvae %&gt;%
  dplyr::filter(harmonic_p_FDR &lt; 0.05) %&gt;%
  .$pathway

HMP.KEGG.larvae %&gt;%
  dplyr::filter(pathway %in% sig.paths.larvae) %&gt;%
  ggplot(aes(x = coef, y = reorder(pathway, -harmonic_p))) +
  geom_tile(aes(fill = -log10(harmonic_p),
                alpha = sig)) +
  geom_label(aes(label = signif(harmonic_p_FDR, digits = 2)),
             fill = NA) +
  scale_fill_viridis_c()</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-33-1.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
</div>
<div id="adult" class="section level4">
<h4>adult</h4>
<pre class="r"><code>HMP.KEGG.adult &lt;- runHMP(gene.set.obj = KEGG.adult,
                   design = designlist$adult,
                   logCPM.df = logCPMs$adult,
                   toptable = toptables_cqn$adult)

sig.paths.adult &lt;- HMP.KEGG.adult %&gt;%
  dplyr::filter(harmonic_p_FDR &lt; 0.05) %&gt;%
  .$pathway

HMP.KEGG.adult %&gt;%
  dplyr::filter(pathway %in% sig.paths.adult) %&gt;%
  ggplot(aes(x = coef, y = reorder(pathway, -harmonic_p))) +
  geom_tile(aes(fill = -log10(harmonic_p),
                alpha = sig)) +
  geom_label(aes(label = signif(harmonic_p_FDR, digits = 2)),
             fill = NA) +
  scale_fill_viridis_c()</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-34-1.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="cell-type-prop-check" class="section level3">
<h3>Cell type prop check</h3>
<p>Since this experiment was performed on whole larvae or adult brain, we need to have a look at whether changes to cell type proportions could explain cause some artefactual changes to gene expression. We have done this previously by assessing the expression of marker genes of specific cell types. So this is what I will do here. I have taken the lists of genes from day 5 only from the zebrafish single cell atlas (Farnsworth et al. 2020) which contain at least 10 genes. This leaves 24 gene sets (cell types) to test.</p>
<div id="larvae-1" class="section level4">
<h4>larvae</h4>
<pre class="r"><code>fry.cell.larvae &lt;- designlist$larvae %&gt;% colnames() %&gt;% .[2:3] %&gt;%
  sapply(function(y) {
    logCPMs$larvae %&gt;%
      fry(
        index = cell_type_markerslarvae,
        design = designlist$larvae,
        contrast = y,
        sort = &quot;directional&quot;
      ) %&gt;%
      rownames_to_column(&quot;pathway&quot;) %&gt;%
      as_tibble() %&gt;%
      mutate(coef = y) %&gt;%
      dplyr::select(1:5, coef)
  }, simplify = FALSE)</code></pre>
<p>No significantly altered cell type gene sets are detected in the larvae. The top 10 most significantly altered gene sets are shown in the plot below. The FDR-adjusted p-values are shown in the graph labels</p>
<pre class="r"><code>fry.cell.larvae %&gt;% 
  bind_rows() %&gt;% 
  arrange(PValue) %&gt;% 
  head(10) %&gt;% 
  ggplot(aes(x = -log10(PValue), y = pathway, fill = coef)) +
  geom_col(position = &quot;dodge&quot;)+
  geom_label(aes(label = signif(FDR, digits = 2)))</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-36-1.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
</div>
<div id="adult-1" class="section level4">
<h4>adult</h4>
<pre class="r"><code>fry.cell.adult &lt;- designlist$adult %&gt;% colnames() %&gt;% .[2:3] %&gt;%
  sapply(function(y) {
    logCPMs$adult %&gt;%
      fry(
        index = cell_type_markers.adultbrain,
        design = designlist$adult,
        contrast = y,
        sort = &quot;directional&quot;
      ) %&gt;%
      rownames_to_column(&quot;pathway&quot;) %&gt;%
      as_tibble() %&gt;%
      mutate(coef = y) %&gt;%
      dplyr::select(1:5, coef)
  }, simplify = FALSE)</code></pre>
<p>In the adults, I used an adult single cell dataset of Jiang et al which shows the genes expressed in 23 different cell types in adult zebrafish brain. 10 gene sets were significantly altered in the MPS IIIB brains and none in the EOfAD brains.</p>
<pre class="r"><code>fry.cell.adult %&gt;% 
  bind_rows() %&gt;% 
  arrange(PValue) %&gt;% 
  ggplot(aes(x = coef, y = pathway)) +
  geom_tile(aes(fill = -log10(PValue))) +
  geom_label(aes(label = signif(FDR, digits = 2), 
                 fill = -log10(PValue))) +
  geom_label(label = &quot;**&quot;, # add labels for sig. altered gene sets
             data = . %&gt;% 
               dplyr::filter(FDR &lt; 0.05), 
             position = position_nudge(x = .2)) +
  scale_fill_viridis_c()</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-38-1.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="ire-gene-sets" class="section level3">
<h3>IRE gene sets</h3>
<p>These gene sets come from Hin et al. </p>
<div id="larvae-2" class="section level4">
<h4>larvae</h4>
<p>No staistical significance in the IRE genes in 7 day old larve.</p>
<pre class="r"><code>HMP.ire.larvae &lt;- runHMP(gene.set.obj = ire.larvae,
                   design = designlist$larvae,
                   logCPM.df = logCPMs$larvae,
                   toptable = toptables_cqn$larvae)


HMP.ire.larvae %&gt;%
  ggplot(aes(x = coef, y = reorder(pathway, -harmonic_p))) +
  geom_tile(aes(fill = -log10(harmonic_p),
                alpha = sig)) +
  geom_label(aes(label = signif(harmonic_p_FDR, digits = 2)),
             fill = NA) +
  scale_fill_viridis_c()</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-39-1.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
</div>
<div id="adult-brain" class="section level4">
<h4>adult brain</h4>
<p>No significantly altered gene sets in the adults either.</p>
<pre class="r"><code>HMP.ire.adult &lt;- runHMP(gene.set.obj = ire.adult,
                   design = designlist$adult,
                   logCPM.df = logCPMs$adult,
                   toptable = toptables_cqn$adult)


HMP.ire.adult %&gt;%
  ggplot(aes(x = coef, y = reorder(pathway, -harmonic_p))) +
  geom_tile(aes(fill = -log10(harmonic_p),
                alpha = sig)) +
  geom_label(aes(label = signif(harmonic_p_FDR, digits = 2)),
             fill = NA) +
  scale_fill_viridis_c()</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-40-1.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
</div>
<div id="post-hoc-power-calc" class="section level1">
<h1>Post-hoc power calc</h1>
<div id="larvae-3" class="section level2">
<h2>larvae</h2>
<pre class="r"><code>set.seed(2016)

mu.larvae &lt;- x.larvae$counts %&gt;%
  as.data.frame() %&gt;%
  .[grepl(colnames(x.larvae$counts), pattern = &quot;wt&quot;)] %&gt;%
  rowMeans()

disp.larvae &lt;-  x.larvae %&gt;% estimateDisp(designlist$larvae) %&gt;% . $tagwise.dispersion

fc &lt;- function(y){exp(rnorm(y, log(1), 0.5*log(2)))}

power.larvae &lt;- ssizeRNA_vary(nGenes = dim(x.larvae)[1], # Num detectable genes
              pi0 = 0.9, # Proportion of non-DE genes
              m = 30, # pseudo sample size
              mu = mu.larvae, # Average read count for each gene in control group. Calculated from this current dataset)
              disp = disp.larvae, # Dispersion parameter for each gene
              fc = fc, # Fold change per gene
              fdr = 0.05, # FDR level to control
              #power = 0.7, # power level
              maxN = 40,
              replace = T
                )</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-41-1.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
<pre class="r"><code>power.larvae$power %&gt;%
  set_colnames(c(&quot;n&quot;, &quot;power&quot;)) %&gt;%
  ggplot(aes(x = n, y = power)) +
  geom_point() +
  geom_line() +
  coord_cartesian(xlim = c(0,30)) +
  geom_vline(xintercept = 8)</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-41-2.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
</div>
<div id="adult-2" class="section level2">
<h2>adult</h2>
<pre class="r"><code>set.seed(2016)

mu.adult &lt;- x.adult$counts %&gt;%
  as.data.frame() %&gt;%
  .[grepl(colnames(x.adult$counts), pattern = &quot;wt&quot;)] %&gt;%
  rowMeans()

disp.adult &lt;-  x.adult %&gt;% estimateDisp(designlist$adult) %&gt;% . $tagwise.dispersion

fc &lt;- function(y){exp(rnorm(y, log(1), 0.5*log(2)))}

power.adult &lt;- ssizeRNA_vary(nGenes = dim(x.adult)[1], # Num detectable genes
              pi0 = 0.9, # Proportion of non-DE genes
              m = 30, # pseudo sample size
              mu = mu.adult, # Average read count for each gene in control group. Calculated from this current dataset)
              disp = disp.adult, # Dispersion parameter for each gene
              fc = fc, # Fold change per gene
              fdr = 0.05, # FDR level to control
              #power = 0.7, # power level
              maxN = 40,
              replace = T
                )</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-42-1.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
<pre class="r"><code>power.adult$power %&gt;%
  set_colnames(c(&quot;n&quot;, &quot;power&quot;)) %&gt;%
  as_tibble() %&gt;% 
  mutate(age = &quot;adult&quot;) %&gt;% 
  bind_rows(power.larvae$power %&gt;% 
              set_colnames(c(&quot;n&quot;, &quot;power&quot;)) %&gt;%
              as_tibble() %&gt;% 
              mutate(age = &quot;larvae&quot;)
            ) %&gt;% 
  ggplot(aes(x = n, y = power, colour = age)) +
  geom_point() +
  geom_line() +
  coord_cartesian(xlim = c(0,30)) +
  geom_vline(xintercept = 8)</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-42-2.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="compare-with-dong-et-al.-data" class="section level1">
<h1>Compare with Dong et al. data</h1>
<p>Yang had previously analysed the EOfAD-like mutation in pools of larvae. Here, I want to see how similar my results are.</p>
<p>Obtained his github repo from here, which has the data <a href="https://github.com/yangdongau/20190717_Lardelli_RNASeq_Larvae" class="uri">https://github.com/yangdongau/20190717_Lardelli_RNASeq_Larvae</a></p>
<pre class="r"><code># read in his SDGE list objkect
x.Dong &lt;- read_rds(&quot;data/DongEtAlData/dgeList.rds&quot;)

# convert to be compatible with my analysios
x.Dong$samples %&lt;&gt;%
  mutate(Genotype = case_when(
    Genotype == &quot;Q96K97del/+&quot; ~ &quot;EOfAD-like&quot;,
    Genotype == &quot;WT&quot; ~ &quot;wt&quot;
  ) %&gt;%
    factor(levels = c(&quot;wt&quot;, &quot;EOfAD-like&quot;)))

larvaeGenos &lt;- x.Dong$samples %&gt;%
  dplyr::select(Genotype) %&gt;%
  mutate(exp = &quot;pools&quot;) %&gt;%
  bind_rows(x.larvae$samples %&gt;% 
              dplyr::select(Genotype) %&gt;% 
              mutate(exp = &quot;individual&quot;)) %&gt;%
  rownames_to_column(&quot;sample&quot;)</code></pre>
<div id="pca-1" class="section level3">
<h3>PCA</h3>
<p>First, I will do a PCA on both datasets to see how simialr they are and whether the samples cluster by genotype or experiment. The samples seperate very much by experiment and this explains almost all of the variation in the dataset…</p>
<pre class="r"><code># PCA
cpm(x.larvae, log = T) %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;gene_id&quot;) %&gt;%
  left_join(x.Dong$counts %&gt;% 
              cpm(log=T) %&gt;% 
              as.data.frame %&gt;% 
              rownames_to_column(&quot;gene_id&quot;)) %&gt;%
  column_to_rownames(&quot;gene_id&quot;) %&gt;%
  dplyr::select(!starts_with(&quot;MPS IIIB&quot;)) %&gt;%
  na.omit %&gt;%
  t %&gt;%
  prcomp() %&gt;%
  autoplot(data = tibble(sample = rownames(.$x)) %&gt;%
             left_join(larvaeGenos),
           colour = &quot;Genotype&quot;,
           size = 4
  ) +
  geom_mark_ellipse(aes(labels = exp)) +
  scale_color_discrete_qualitative(palette = &quot;Dark 3&quot;) +
  theme(aspect.ratio = 1)</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-44-1.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
</div>
<div id="compare-fc-values-in-the-ecm-gene-set" class="section level3">
<h3>Compare FC values in the ECM gene set</h3>
<pre class="r"><code># suss out ECM
toptable_dongetal &lt;-
  read_csv(&quot;data/DongEtAlData/topTable.csv&quot;) %&gt;%
  dplyr::rename( gene_id = ensembl_gene_id,
                 gene_name = external_gene_name)

toptable_dongetal %&gt;%
  dplyr::filter(gene_id %in% KEGG.notfiltered$KEGG_ECM_RECEPTOR_INTERACTION) %&gt;%
  dplyr::select(gene_name, logFC.pools = logFC) %&gt;%
  left_join(toptables_cqn$larvae$`EOfAD-like` %&gt;% dplyr::select(gene_name, logFC.individuals = logFC)) %&gt;%
  dplyr::distinct(gene_name, .keep_all = T) %&gt;%
  column_to_rownames(&quot;gene_name&quot;) %&gt;%
  pheatmap(color = colorRampPalette(rev(brewer.pal(n = 5,
                                               name = &quot;RdBu&quot;)))(200),
           breaks = seq(-0.6, 0.6, by = 0.6*2/200),
           main =&quot;KEGG_ECM_RECEPTOR_INTERACTION&quot;)</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-45-1.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
<pre class="r"><code>toptable_dongetal %&gt;%
  dplyr::filter(gene_id %in% KEGG.notfiltered$KEGG_ECM_RECEPTOR_INTERACTION) %&gt;%
  dplyr::select(gene_name, logFC.pools = logFC) %&gt;%
  left_join(toptables_cqn$adult$`EOfAD-like` %&gt;% 
              dplyr::select(gene_name, logFC.individuals = logFC)) %&gt;%
  dplyr::distinct(gene_name, .keep_all = T) %&gt;%
  column_to_rownames(&quot;gene_name&quot;) %&gt;%
  ggscatter(x = &quot;logFC.individuals&quot;, y = &quot;logFC.pools&quot;,
   color = &quot;black&quot;, shape = 21, size = 3, # Points color, shape and size
   add = &quot;reg.line&quot;,  # Add regressin line
   add.params = list(color = &quot;blue&quot;, fill = &quot;lightgray&quot;), # Customize reg. line
   conf.int = TRUE, # Add confidence interval
   cor.coef = TRUE, # Add correlation coefficient. see ?stat_cor
   #cor.coeff.args = list(method = &quot;pearson&quot;, label.x = 3, label.sep = &quot;\n&quot;)
   ) +
  geom_abline(slope = 1) +
  labs(title = &quot;Pearson corr of genes in KEGG_ECM_RECEPTOR_INTERACTION&quot;)</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-45-2.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pwf.dong &lt;- toptable_dongetal %&gt;%
      with(
        nullp(
          DEgenes = structure(DE, names = gene_id),
          bias.data = aveLength
        )
      )</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-45-3.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
<pre class="r"><code>goseq_dong &lt;- goseq(pwf.dong, gene2cat = GO.notfiltered) %&gt;%
  as_tibble() %&gt;%
  #dplyr::filter(numDEInCat &gt; 0) %&gt;%
  mutate(FDR = p.adjust(over_represented_pvalue, method = &quot;fdr&quot;)) %&gt;%
  dplyr::select(-under_represented_pvalue)

goseq_dong %&gt;% dplyr::filter(category == &quot;GO_PROTON_TRANSPORTING_V_TYPE_ATPASE_COMPLEX&quot;)</code></pre>
<pre><code># A tibble: 1 × 5
  category                      over_represented_pva…¹ numDEInCat numInCat   FDR
  &lt;chr&gt;                                          &lt;dbl&gt;      &lt;int&gt;    &lt;int&gt; &lt;dbl&gt;
1 GO_PROTON_TRANSPORTING_V_TYP…                      1          0       25     1
# ℹ abbreviated name: ¹​over_represented_pvalue</code></pre>
<pre class="r"><code>goseq_dong %&gt;%
  dplyr::slice(1:40) %&gt;%
  mutate(propDE = numDEInCat/numInCat) %&gt;%
  ggplot(aes(x = -log10(over_represented_pvalue), y = reorder(category, -over_represented_pvalue))) +
  geom_col(aes(fill = propDE, alpha = FDR &lt; 0.05)) +
  scale_fill_viridis_c() +
  scale_alpha_manual(values = c(0.4, 1)) +
  labs(y = &quot;GO Term&quot;,
       fill = &quot;Proportion of \nDE genes in \nGO term&quot;)</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-45-4.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
</div>
<div id="hmp-with-dong-et-al." class="section level3">
<h3>HMP with dong et al.</h3>
<pre class="r"><code>design.dong &lt;- model.matrix(~Genotype + W_1, x.Dong$samples) %&gt;%
  set_colnames(str_replace(colnames(.), pattern = &quot;Genotype.+&quot;, &quot;EOfAD-like&quot;))

kegg.dong &lt;- KEGG.notfiltered %&gt;% 
  lapply(function(x) {
    x %&gt;% 
      as_tibble() %&gt;% 
      set_colnames(&quot;gene_id&quot;) %&gt;% 
      dplyr::filter(gene_id %in% rownames(x.Dong)) %&gt;% 
      .$gene_id
  })

fry &lt;-
  cpm(x.Dong, log = T) %&gt;%
      fry(
        index = kegg.dong,
        design = design.dong,
        contrast = &quot;EOfAD-like&quot;,
        sort = &quot;mixed&quot;
      ) %&gt;%
      rownames_to_column(&quot;pathway&quot;) %&gt;%
      as_tibble() %&gt;%
      mutate(coef = &quot;EOfAD-like&quot;,
             sig = FDR.Mixed &lt; 0.05)

camera &lt;-
  cpm(x.Dong, log = T) %&gt;%
  camera(
    index = kegg.dong,
    design = design.dong,
    contrast = &quot;EOfAD-like&quot;
  ) %&gt;%
      rownames_to_column(&quot;pathway&quot;) %&gt;%
      as_tibble() %&gt;%
      mutate(coef = &quot;EOfAD-like&quot;)

ranks &lt;-
  toptable_dongetal %&gt;%
  mutate(rankstat = sign(logFC) * log10(1/PValue)) %&gt;%
  arrange(rankstat) %&gt;%
  dplyr::select(c(&quot;gene_id&quot;, &quot;rankstat&quot;)) %&gt;% #only want the Pvalue with sign
  with(structure(rankstat, names = gene_id)) %&gt;%
  rev() # reverse so the start of the list is upregulated genes

# Run fgsea

# set a seed for a reproducible result
set.seed(33)
fgsea &lt;- ranks %&gt;%
  fgseaMultilevel(pathways = kegg.dong) %&gt;%
  as_tibble() %&gt;%
  dplyr::rename(FDR = padj) %&gt;%
  dplyr::select(pathway, pval, FDR, everything()) %&gt;%
  arrange(pval) %&gt;%
  mutate(sig = FDR &lt; 0.05,
         coef =&quot;EOfAD-like&quot;)

hmp.dong &lt;- fry %&gt;%
  dplyr::select(pathway, PValue.Mixed, coef) %&gt;%
  dplyr::rename(fry_p = PValue.Mixed) %&gt;%
  left_join(camera %&gt;%
              dplyr::select(pathway, PValue, coef),
            by = c(&quot;pathway&quot;, &quot;coef&quot;)) %&gt;%
  dplyr::rename(camera_p = PValue) %&gt;%
  left_join(fgsea %&gt;%
              bind_rows() %&gt;%
              dplyr::select(pathway, pval, coef),
            by = c(&quot;pathway&quot;, &quot;coef&quot;)) %&gt;%
  dplyr::rename(fgsea_p = pval) %&gt;%
  bind_rows() %&gt;%
  nest(p = one_of(c(&quot;fry_p&quot;, &quot;camera_p&quot;, &quot;fgsea_p&quot;))) %&gt;%
 mutate(harmonic_p = vapply(p, function(x){
        x &lt;- unlist(x)
        x &lt;- x[!is.na(x)]
        p.hmp(x, L = 4)
      }, numeric(1))
 ) %&gt;%
  unnest() %&gt;%
  mutate(harmonic_p_FDR = p.adjust(harmonic_p, &quot;fdr&quot;),
         sig = harmonic_p_FDR &lt; 0.05) %&gt;%
  arrange(harmonic_p)

hmp.kegg.q96 &lt;- hmp.dong %&gt;%
  mutate(exp = &quot;pools&quot;) %&gt;%
  bind_rows( HMP.KEGG.larvae %&gt;%
              mutate(exp = &quot;individuals&quot;)) %&gt;%
  dplyr::filter(coef == &quot;EOfAD-like&quot;)

sig.q96 &lt;- hmp.kegg.q96 %&gt;%
  dplyr::filter(sig == T) %&gt;%
  .$pathway %&gt;%
  unique

hmp.dong %&gt;%
  dplyr::filter(pathway %in% sig.q96) %&gt;% 
  ggplot(aes(x = -log10(harmonic_p), y = reorder(pathway, -harmonic_p))) +
  geom_col(aes(fill = -log10(harmonic_p))) +
  scale_fill_viridis_c() +
  theme_classic()</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-46-1.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
<!-- ### does Genotype drive the Dong ECM? -->
<!-- ```{r} -->
<!-- form <- ~ (1|Genotype) + (1|pair) -->
<!-- geneExpr <- -->
<!--   x.Dong$counts %>% -->
<!--      as.data.frame() %>% -->
<!--      rownames_to_column("gene_id") %>% -->
<!--      dplyr::filter(gene_id %in% kegg.dong$KEGG_ECM_RECEPTOR_INTERACTION) %>% -->
<!--      column_to_rownames("gene_id") -->
<!-- n <- parallel::detectCores()/2 # experiment! -->
<!-- cl <- parallel::makeCluster(n) -->
<!-- doParallel::registerDoParallel(cl) -->
<!-- # this takes forever to run. -->
<!-- #varpar <- fitExtractVarPartModel(geneExpr, form, x.Dong$samples, -->
<!--      #                            BPPARAM=SnowParam(1)) # turn off parallel as my laptop sucks -->
<!-- #saveRDS(varpar, "data/dongvarpar.rds") -->
<!-- varpar <- readRDS("data/dongvarpar.rds") -->
<!-- varpar %>% -->
<!--   .[kegg.dong$KEGG_ECM_RECEPTOR_INTERACTION,] %>% -->
<!--   rownames_to_column("gene_id") %>% -->
<!--   na.omit %>% -->
<!--   arrange(Genotype) %>% -->
<!--   gather(key = "var", value = "prop", c(Genotype, pair, Residuals)) %>% -->
<!--   ggplot(aes(x = var, y = prop)) + -->
<!--   geom_violin() + -->
<!--   geom_boxplot(width = 0.1) -->
<!-- ``` -->
</div>
<div id="power-calc-dong" class="section level3">
<h3>power calc Dong</h3>
<pre class="r"><code>set.seed(2016)

mu2 &lt;-
  x.Dong$counts %&gt;%
  as.data.frame() %&gt;%
  .[,(x.Dong$samples %&gt;% dplyr::filter(Genotype == &quot;wt&quot;) %&gt;% .$sample)] %&gt;%
  rowMeans()

disp2 &lt;-  x.Dong %&gt;% estimateDisp() %&gt;% . $tagwise.dispersion

fc &lt;- function(y){exp(rnorm(y, log(1), 0.5*log(1.5)))}

power2 &lt;- ssizeRNA_vary(nGenes = dim(x.Dong)[1], # Num detectable genes
              pi0 = 0.8, # Proportion of non-DE genes
              m = 30, # pseudo sample size
              mu = mu2, # Average read count for each gene in control group. Calculated from this current dataset)
              disp = disp2, # Dispersion parameter for each gene
              fc = fc, # Fold change per gene
              fdr = 0.05, # FDR level to control
              #power = 0.7, # power level
              maxN = 40,
              replace = T
                )</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-47-1.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
<pre class="r"><code>power.adult$power %&gt;%
  set_colnames(c(&quot;n&quot;, &quot;power&quot;)) %&gt;%
  as_tibble() %&gt;% 
  mutate(age = &quot;adult&quot;) %&gt;% 
  bind_rows(power.larvae$power %&gt;% 
              set_colnames(c(&quot;n&quot;, &quot;power&quot;)) %&gt;%
              as_tibble() %&gt;% 
              mutate(age = &quot;larvae&quot;)
            ) %&gt;% 
  bind_rows(power2$power %&gt;%
              set_colnames(c(&quot;n&quot;, &quot;power&quot;)) %&gt;% 
              as_tibble() %&gt;% 
              mutate(age = &quot;larvae (pools)&quot;)) %&gt;% 
  ggplot(aes(x = n, y = power, colour = age)) +
  geom_point() +
  geom_line() +
  coord_cartesian(xlim = c(0,30)) +
  geom_vline(xintercept = 8)</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-47-2.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="compare-with-hin-et.-al." class="section level1">
<h1>compare with Hin et. al.</h1>
<p>Nhi also looked at the psen1 Q96_K97del mutation in 6m adult brain. I reprocessed her data in the nhi6mdata doc.</p>
<pre class="r"><code># read in the outputs from the analysis
x.nhi &lt;- read_rds(&quot;data/R_objects/adult_brain/dgeNhi.rds&quot;)
logCPM.nhi &lt;- read_rds(&quot;data/R_objects/adult_brain/logcpmNhi.rds&quot;)
toptables_cqn.nhi &lt;- read_rds(&quot;data/R_objects/adult_brain/toptablescqnNhi.rds&quot;)
hmp.KEGG.nhi &lt;- read_rds(&quot;data/R_objects/adult_brain/hmp_keggNhi.rds&quot;)
hmp.ire.nhi &lt;- read_rds(&quot;data/R_objects/adult_brain/hmp_ireNhi.rds&quot;)
celltype.nhi &lt;- read_rds(&quot;data/R_objects/adult_brain/celltypeNhi.rds&quot;) 

# convert to be compatible with my analysios
x.nhi$samples %&lt;&gt;% 
  mutate(Genotype = case_when(
    genotype == &quot;Q96_K97del/+&quot; ~ &quot;EOfAD-like&quot;,
    genotype == &quot;wt&quot; ~ &quot;wt&quot;
  ) %&gt;%
    factor(levels = c(&quot;wt&quot;, &quot;EOfAD-like&quot;))) %&gt;% 
  dplyr::select(-genotype)

adult.genotype.df &lt;- x.nhi$samples %&gt;%
  dplyr::select(Genotype) %&gt;%
  mutate(sex = &quot;F&quot;,
         exp = &quot;exp 2&quot;) %&gt;%
  bind_rows(x.adult$samples %&gt;% 
              dplyr::select(Genotype, sex) %&gt;% 
              mutate(exp = &quot;exp 1&quot;,)) %&gt;%
  rownames_to_column(&quot;sample&quot;)</code></pre>
<div id="pca-2" class="section level3">
<h3>PCA</h3>
<pre class="r"><code># PCA
logCPMs$adult %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;gene_id&quot;) %&gt;%
  left_join(logCPM.nhi %&gt;% 
              as.data.frame %&gt;% 
              rownames_to_column(&quot;gene_id&quot;)) %&gt;%
  column_to_rownames(&quot;gene_id&quot;) %&gt;%
  dplyr::select(!contains(&quot;MPS&quot;)) %&gt;% 
  na.omit %&gt;%
  t %&gt;%
  prcomp() %&gt;%
  autoplot(data = tibble(sample = rownames(.$x)) %&gt;% 
             left_join(adult.genotype.df),
           colour = &quot;Genotype&quot;,
           shape = &quot;sex&quot;,
           size = 4
  ) +
  geom_mark_ellipse(aes(labels = exp)) +
  scale_color_discrete_qualitative(palette = &quot;Dark 3&quot;) +
  theme(aspect.ratio = 1)</code></pre>
<p><img src="figure/analysis_final.rmd/unnamed-chunk-50-1.png" width="75%" height="75%" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.3 (2023-03-15)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] stats4    splines   stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ensembldb_2.22.0         AnnotationFilter_1.22.0  GenomicFeatures_1.50.4  
 [4] AnnotationDbi_1.60.2     Biobase_2.58.0           GenomicRanges_1.50.2    
 [7] GenomeInfoDb_1.34.9      IRanges_2.32.0           S4Vectors_0.36.2        
[10] pathview_1.38.0          colorspace_2.1-0         RColorBrewer_1.1-3      
[13] ggforce_0.4.1            ggfortify_0.4.16         ggrepel_0.9.3           
[16] ggpubr_0.6.0             pheatmap_1.0.12          scales_1.2.1            
[19] UpSetR_1.4.0             pander_0.6.5             variancePartition_1.28.9
[22] BiocParallel_1.32.6      clusterProfiler_4.6.2    ssizeRNA_1.3.2          
[25] harmonicmeanp_3.0        FMStable_0.1-4           cqn_1.44.0              
[28] quantreg_5.95            SparseM_1.81             preprocessCore_1.60.2   
[31] nor1mix_1.3-0            mclust_6.0.0             fgsea_1.24.0            
[34] goseq_1.50.0             geneLenDataBase_1.34.0   BiasedUrn_2.0.9         
[37] edgeR_3.40.2             limma_3.54.2             msigdbr_7.5.1           
[40] AnnotationHub_3.6.0      BiocFileCache_2.6.1      dbplyr_2.3.2            
[43] ngsReports_2.0.3         patchwork_1.1.2          BiocGenerics_0.44.0     
[46] readxl_1.4.2             magrittr_2.0.3           lubridate_1.9.2         
[49] forcats_1.0.0            stringr_1.5.0            dplyr_1.1.2             
[52] purrr_1.0.1              readr_2.1.4              tidyr_1.3.0             
[55] tibble_3.2.1             ggplot2_3.4.2            tidyverse_2.0.0         

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3                rtracklayer_1.58.0           
  [3] clusterGeneration_1.3.7       bit64_4.0.5                  
  [5] knitr_1.42                    DelayedArray_0.24.0          
  [7] data.table_1.14.8             KEGGREST_1.38.0              
  [9] RCurl_1.98-1.12               doParallel_1.0.17            
 [11] generics_0.1.3                RhpcBLASctl_0.23-42          
 [13] cowplot_1.1.1                 RSQLite_2.3.1                
 [15] shadowtext_0.1.2              bit_4.0.5                    
 [17] tzdb_0.3.0                    enrichplot_1.18.4            
 [19] xml2_1.3.4                    httpuv_1.6.10                
 [21] SummarizedExperiment_1.28.0   viridis_0.6.3                
 [23] xfun_0.39                     hms_1.1.3                    
 [25] jquerylib_0.1.4               babelgene_22.9               
 [27] evaluate_0.21                 promises_1.2.0.1             
 [29] fansi_1.0.4                   restfulr_0.0.15              
 [31] progress_1.2.2                caTools_1.18.2               
 [33] Rgraphviz_2.42.0              igraph_1.4.2                 
 [35] DBI_1.1.3                     htmlwidgets_1.6.2            
 [37] ellipsis_0.3.2                backports_1.4.1              
 [39] aod_1.3.2                     biomaRt_2.54.1               
 [41] MatrixGenerics_1.10.0         vctrs_0.6.2                  
 [43] abind_1.4-5                   cachem_1.0.8                 
 [45] withr_2.5.0                   HDO.db_0.99.1                
 [47] vroom_1.6.3                   GenomicAlignments_1.34.1     
 [49] treeio_1.22.0                 prettyunits_1.1.1            
 [51] DOSE_3.24.2                   ape_5.7-1                    
 [53] lazyeval_0.2.2                crayon_1.5.2                 
 [55] labeling_0.4.2                pkgconfig_2.0.3              
 [57] tweenr_2.0.2                  ProtGenerics_1.30.0          
 [59] nlme_3.1-162                  rlang_1.1.1                  
 [61] lifecycle_1.0.3               MatrixModels_0.5-1           
 [63] downloader_0.4                filelock_1.0.2               
 [65] cellranger_1.1.0              rprojroot_2.0.3              
 [67] polyclip_1.10-4               matrixStats_0.63.0           
 [69] graph_1.76.0                  Matrix_1.5-4                 
 [71] aplot_0.1.10                  carData_3.0-5                
 [73] boot_1.3-28.1                 zoo_1.8-12                   
 [75] png_0.1-8                     viridisLite_0.4.2            
 [77] rjson_0.2.21                  bitops_1.0-7                 
 [79] gson_0.1.0                    KernSmooth_2.23-21           
 [81] Biostrings_2.66.0             blob_1.2.4                   
 [83] workflowr_1.7.0               qvalue_2.30.0                
 [85] rstatix_0.7.2                 remaCor_0.0.11               
 [87] gridGraphics_0.5-1            ggsignif_0.6.4               
 [89] memoise_2.0.1                 plyr_1.8.8                   
 [91] gplots_3.1.3                  zlibbioc_1.44.0              
 [93] compiler_4.2.3                scatterpie_0.1.9             
 [95] BiocIO_1.8.0                  KEGGgraph_1.58.3             
 [97] lme4_1.1-33                   Rsamtools_2.14.0             
 [99] cli_3.6.1                     XVector_0.38.0               
[101] MASS_7.3-60                   mgcv_1.8-42                  
[103] tidyselect_1.2.0              stringi_1.7.12               
[105] highr_0.10                    yaml_2.3.7                   
[107] GOSemSim_2.24.0               locfit_1.5-9.7               
[109] grid_4.2.3                    sass_0.4.6                   
[111] fastmatch_1.1-3               tools_4.2.3                  
[113] timechange_0.2.0              parallel_4.2.3               
[115] rstudioapi_0.14               foreach_1.5.2                
[117] git2r_0.32.0                  gridExtra_2.3                
[119] farver_2.1.1                  ggraph_2.1.0                 
[121] digest_0.6.31                 BiocManager_1.30.20          
[123] shiny_1.7.4                   Rcpp_1.0.10                  
[125] car_3.1-2                     broom_1.0.4                  
[127] BiocVersion_3.16.0            later_1.3.1                  
[129] org.Hs.eg.db_3.16.0           httr_1.4.6                   
[131] ggdendro_0.1.23               Rdpack_2.4                   
[133] XML_3.99-0.14                 fs_1.6.2                     
[135] ssize.fdr_1.3                 statmod_1.5.0                
[137] yulab.utils_0.0.6             tidytree_0.4.2               
[139] graphlayouts_1.0.0            ggplotify_0.1.0              
[141] systemfonts_1.0.4             plotly_4.10.1                
[143] xtable_1.8-4                  jsonlite_1.8.4               
[145] nloptr_2.0.3                  ggtree_3.6.2                 
[147] tidygraph_1.2.3               ggfun_0.0.9                  
[149] RUnit_0.4.32                  R6_2.5.1                     
[151] pillar_1.9.0                  htmltools_0.5.5              
[153] mime_0.12                     glue_1.6.2                   
[155] fastmap_1.1.1                 minqa_1.2.5                  
[157] DT_0.27                       interactiveDisplayBase_1.36.0
[159] codetools_0.2-19              mvtnorm_1.1-3                
[161] utf8_1.2.3                    lattice_0.21-8               
[163] bslib_0.4.2                   pbkrtest_0.5.2               
[165] curl_5.0.0                    gtools_3.9.4                 
[167] GO.db_3.16.0                  survival_3.5-5               
[169] rmarkdown_2.21                munsell_0.5.0                
[171] GenomeInfoDbData_1.2.9        iterators_1.0.14             
[173] reshape2_1.4.4                gtable_0.3.3                 
[175] rbibutils_2.2.13             </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
