---
title: "analysis_adultbrain"
author: "Karissa Barthelson"
date: "2023-03-01"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---
## Introduction
```{r setup, include=FALSE}
knitr::opts_chunk$set(
  autodep = TRUE,
  echo = TRUE,
  warning = FALSE,
  message = FALSE,
  fig.align = "center", 
  out.width ="75%", 
  out.height = "75%"
)
```

```{r loadLibs}
library(tidyverse)
library(magrittr)
library(readxl)
library(ngsReports)
library(AnnotationHub)
library(pander)
library(scales)
library(pheatmap)
library(ggpubr)
library(msigdbr)
library(scales)
library(ggrepel)
library(ggfortify)
library(RColorBrewer)
library(UpSetR)
library(edgeR)
library(goseq)
library(fgsea)
library(cqn)
library(kableExtra)

library(ssizeRNA)

theme_set(theme_bw())
```

```{r anno}
ah <- AnnotationHub() %>%
	subset(species == "Danio rerio") %>%
	subset(rdataclass == "EnsDb")

ensDb <- ah[["AH83189"]] # for release 101, latest version and the alignment
grTrans <- transcripts(ensDb)
trLengths <- exonsBy(ensDb, "tx") %>%
	width() %>%
	vapply(sum, integer(1))
mcols(grTrans)$length <- trLengths[names(grTrans)]
gcGene <- grTrans %>%
  mcols() %>%
  as.data.frame() %>%
  dplyr::select(gene_id, tx_id, gc_content, length) %>%
  as_tibble() %>%
  group_by(gene_id) %>%
  summarise(
    gc_content = sum(gc_content*length) / sum(length),
    length = ceiling(median(length))
  )
grGenes <- genes(ensDb)
mcols(grGenes) %<>%
  as.data.frame() %>%
  left_join(gcGene) %>%
  as.data.frame() %>%
  DataFrame()
```

```{r meta}
meta <- read_excel("data/adult_brain/karissas_metadata.xlsx", sheet = "onlyseq") %>% 
  mutate(Genotype = `usable genotype?` %>% 
           factor(levels = c("wt", "MPS-III", "EOfAD")), 
         Tank = as.factor(tank)
  ) %>% 
  mutate(sample = paste0(fish, "_", Genotype)) 
```

<!-- First, I will assess the quality of the RNA-seq data for the *psen1* Q96_K97del/+ vs *naglu* A603fs/A603fs experiment on zebrafish adult brains at 7 days post fertilisation (dpf).  -->

<!-- Total RNA was purified from the head end of the individual larvae, while the tail end was used for gDNA extraction and PCR genotyping. The total RNA was DNase treated (to remove any genomic DNA which was carried over from the RNA extraction), then delivered to SAGC for polyA+ library preparation and sequencing using the MGI DNBSEQ technology.  -->

<!-- The sequencing was performed over four lanes which were subsequently merged. This was done using the `merge.sh` script shown below.  -->

<!-- <!-- # ```{r meregFilesBashScript} --> -->
<!-- <!-- # ## Insert the merge files script here --> -->
<!-- <!-- # readLines("code/mergeFiles.sh") %>%  --> -->
<!-- <!-- #   cat(sep = "\n") --> -->
<!-- <!-- # ``` --> -->

<!-- ## fastqc: raw data -->

<!-- Here, I will use the `ngsReports` package to combine and visualise the fastqc results.  -->
<!-- ```{r fastqcRawObejct} -->
<!-- fastqc_raw <- list.files( -->
<!--   path = "data/fastqc_raw", -->
<!--   pattern = "zip",  -->
<!--   recursive = TRUE, -->
<!--   full.names = TRUE) %>%  -->
<!--   FastqcDataList() -->
<!-- ``` -->

<!-- The total number of reads ranged between `r range(readTotals(fastqc_raw)$Total_Sequences) %>% comma %>% pander` reads. Note that the number of reads in the `R1` file indeed equals to the number of reads in the `R2` file.  -->

<!-- ```{r} -->
<!-- readTotals(fastqc_raw) %>%  -->
<!--   mutate(Read = case_when( -->
<!--     grepl(Filename, pattern = "_R1") ~ "R1",  -->
<!--     grepl(Filename, pattern = "_R2") ~ "R2" -->
<!--   ),  -->
<!--   ULN = str_remove(Filename, "_S[0-9]+_merged.+") -->
<!--   ) %>%  -->
<!--   left_join(meta) %>%  -->
<!--   ggplot(aes(x = ULN, y = Total_Sequences, fill = Read)) +  -->
<!--            geom_col(position = "dodge") + -->
<!--   coord_flip() + -->
<!--   scale_fill_viridis_d(end = 0.8) + -->
<!--   facet_wrap(~Genotype, scales = "free_y", ncol = 1, strip.position = "right") -->
<!-- ``` -->

<!-- The base quality of all the reads also looked good. However, something a bit strange is going on in the sample `21-015556` file R2 file. Inspecting the boxplot for this file and it looks OK to me. -->

<!-- ```{r} -->
<!-- plotBaseQuals(fastqc_raw) -->
<!-- ``` -->

<!-- ### GC Content -->

<!-- All samples have similar GC content. No issues are present.  -->

<!-- ```{r} -->
<!-- plotGcContent( -->
<!--   x = fastqc_raw,  -->
<!--   plotType = "line", -->
<!--   gcType = "Transcriptome",  -->
<!--   species = "Drerio",  -->
<!--   usePlotly = F -->
<!-- ) + -->
<!--   theme(legend.position = "none") -->
<!-- ``` -->

<!-- ### Over-repreented seq -->

<!-- No over-represented sequences are present in this dataset.  -->

<!-- ```{r} -->
<!-- getModule(fastqc_raw, "Overrep")  -->
<!-- ``` -->


<!-- ## trimmed data fastQC -->

<!-- The raw fastq. files were then processed with `fastp`. In this step, the adaptor sequeces were trimmed from the reads. Then all length and quality filters were left as default values. Less than 1% of the reads was discarded, and no observed changes are apparent in the %GC in the reads.  -->

<!-- ```{r} -->
<!-- fastqc_trim <- list.files(path = "data/fastqc_trim", -->
<!--   pattern = "zip",  -->
<!--   recursive = TRUE, -->
<!--   full.names = TRUE) %>%  -->
<!--   FastqcDataList() -->
<!-- ``` -->

<!-- ```{r} -->
<!-- trimStats <- readTotals(fastqc_raw) %>% -->
<!--   dplyr::rename(Raw = Total_Sequences) %>% -->
<!--   left_join(readTotals(fastqc_trim), by = "Filename") %>% -->
<!--   dplyr::rename(Trimmed = Total_Sequences) %>% -->
<!--   mutate( -->
<!--     Discarded = 1 - Trimmed / Raw, -->
<!--     Retained = Trimmed / Raw -->
<!--   ) -->

<!-- trimStats %>%  -->
<!--   mutate(ULN = str_remove(Filename, "_S[0-9]+_merged.+") -->
<!--   ) %>%  -->
<!--   left_join(meta) %>%  -->
<!--   unique() %>%  -->
<!--   ggplot(aes(y = ULN)) + -->
<!--   geom_col(aes(x = Discarded*100)) + -->
<!--   facet_wrap(~Genotype, scales = "free_y", ncol = 1, strip.position = "right") + -->
<!--   labs(x = "Percentage reads discarded by fastp") -->
<!-- ``` -->

<!-- ```{r} -->
<!-- plotBaseQuals(fastqc_trim) -->
<!-- ``` -->


<!-- ```{r} -->
<!-- ggarrange( -->
<!--   plotGcContent( -->
<!--     x = fastqc_raw,  -->
<!--     plotType = "line", -->
<!--     gcType = "Transcriptome",  -->
<!--     species = "Drerio" -->
<!--   ) + -->
<!--     theme(legend.position = "none") + -->
<!--     ggtitle("Before fastp"),  -->
<!--   plotGcContent( -->
<!--   x = fastqc_trim,  -->
<!--   plotType = "line", -->
<!--   gcType = "Transcriptome",  -->
<!--   species = "Drerio" -->
<!-- ) + -->
<!--   theme(legend.position = "none")+ -->
<!--   ggtitle("After fastp") -->
<!-- )  -->

<!-- ``` -->

<!-- ## Aligned QC -->
<!-- The reads were aligned to the GRCz11 genome. The majority of reads were aligned uniquely. S -->
<!-- ```{r} -->
<!-- fastqc_align <- list.files( -->
<!--   path = "data/fastqc_align", -->
<!--   pattern = "zip",  -->
<!--   recursive = TRUE, -->
<!--   full.names = TRUE) %>%  -->
<!--   FastqcDataList() -->
<!-- ``` -->

<!-- ```{r} -->
<!-- list.files("data/starlog", full.names = TRUE) %>%  -->
<!--   .[grepl(x = ., pattern = "Log.final.out")] %>%  -->
<!--   ngsReports::plotAlignmentSummary(type = "star") + -->
<!--   scale_fill_viridis_d(end = 0.8) + -->
<!--   theme(legend.position = "right") + -->
<!--   ggtitle("Summary of alignment (STAR)",  -->
<!--           subtitle = "In all samples, the majority of reads mapped uniquely to the zebrafish genome.") -->
<!-- ``` -->

<!-- ```{r} -->
<!-- plotBaseQuals(fastqc_align) -->
<!-- ``` -->


<!-- ```{r} -->
<!-- plotGcContent(x = fastqc_align,  -->
<!--     plotType = "line", -->
<!--     gcType = "Transcriptome",  -->
<!--     species = "Drerio" -->
<!--   ) + -->
<!--   theme(legend.position = "none")  -->
<!-- ``` -->

<!-- ## Dedup align QC -->

<!-- This dataset was processed with UMIs, which allow PCR duplicates to be removed. I did this using `umi-tools`. After de-duplciation ** reads were retained.  -->

<!-- ```{r} -->
<!-- fastqc_align_dedup <- list.files( -->
<!--   path = "data/fastqc_dedup", -->
<!--   pattern = "zip",  -->
<!--   recursive = TRUE, -->
<!--   full.names = TRUE) %>%  -->
<!--   FastqcDataList() -->
<!-- ``` -->

<!-- ```{r} -->
<!-- readTotals(fastqc_align) %>%  -->
<!--   mutate(align = "raw") %>%  -->
<!--   bind_rows(readTotals(fastqc_align_dedup) %>%  -->
<!--               mutate(align = "dedup")) %>%  -->
<!--   mutate(ULN = str_remove(Filename, "_S[0-9]+_merged.+")) %>%  -->
<!--   left_join(meta) %>%  -->
<!--   ggplot(aes(x = ULN, y = Total_Sequences, fill = align)) +  -->
<!--            geom_col(position = "dodge") + -->
<!--   coord_flip() + -->
<!--   scale_fill_viridis_d(end = 0.8) + -->
<!--   scale_y_continuous(labels = comma) + -->
<!--   facet_wrap(~Genotype, scales = "free_y", ncol = 1, strip.position = "right") -->
<!-- ``` -->

<!-- ## FeatureCounts summary -->

<!-- ```{r} -->
<!-- FC_summary <- read.delim("data/05_featureCounts/counts.out.summary") -->

<!-- colnames(FC_summary) %<>%  -->
<!--   str_remove(pattern = "_S[0-9]+_merged.Aligned.sortedByCoord.dedup.out.bam") %>%  -->
<!--   str_remove(pattern = "X04_dedup.bam.") %>%  -->
<!--   str_replace(pattern = "\\.", replacement = "\\-") -->

<!-- FC_summary %>%  -->
<!--  gather(key = "ULN", value = "NumReads", starts_with("22")) %>%  -->
<!--   left_join(meta) %>%  -->
<!--   as_tibble() %>%  -->
<!--   dplyr::filter(NumReads > 0) %>%    -->
<!--   ggplot(aes(y = ULN, x = NumReads, fill = Status)) + -->
<!--   geom_col() + -->
<!--   scale_fill_viridis_d(end = 0.8) + -->
<!--   scale_x_continuous(labels = comma) + -->
<!--   facet_wrap(~Genotype, scales = "free_y", ncol = 1, strip.position = "right") -->
<!-- ``` -->

# DGE

```{r}
featureCounts <- 
  read_delim("data/adult_brain/05_featureCounts/counts.out", delim = "\t", skip = 1) %>%
  set_names(basename(names(.))) %>% 
  set_names(names(.) %>% str_remove(pattern = "_S[0-9]+_merged.Aligned.sortedByCoord.dedup.out.bam")) %>% 
  as_tibble() %>% 
  dplyr::select(-c(Chr, Start, End, Length, Strand)) %>% 
  gather(key = "ULN", value = "counts", starts_with("22")) %>% 
  left_join(meta) %>% 
  mutate(sample = paste0(fish, "_", Genotype)) %>% 
  dplyr::select(Geneid, counts, sample) %>% 
  spread(key = "sample", value = "counts") %>% 
  column_to_rownames("Geneid") %>% 
  .[,1:24]
```


## filter lowly expressed genes


```{r filt}
a <- featureCounts %>% 
    cpm(log = TRUE) %>%
    as.data.frame() %>% 
  pivot_longer(
    cols = everything(),
    names_to = "sample",
    values_to = "logCPM"
  ) %>%
  split(f = .$sample) %>%
  lapply(function(x){
    d <- density(x$logCPM)
    tibble(
      sample = unique(x$sample),
      x = d$x,
      y = d$y
    )
  }) %>%
  bind_rows() %>%
  left_join(meta) %>% 
  ggplot(aes(x, y, colour = Genotype, group = sample)) +
  geom_line() +
  labs(
    x = "logCPM",
    y = "Density",
    colour = "Genotype"
  )+
  ggtitle("Before filtering") +
  theme_bw()
b <- featureCounts %>% 
  .[rowSums(cpm(.) >= 0.5) >= 8,] %>% 
    cpm(log = TRUE) %>%
    as.data.frame() %>% 
  pivot_longer(
    cols = everything(),
    names_to = "sample",
    values_to = "logCPM"
  ) %>%
  split(f = .$sample) %>%
  lapply(function(x){
    d <- density(x$logCPM)
    tibble(
      sample = unique(x$sample),
      x = d$x,
      y = d$y
    )
  }) %>%
  bind_rows() %>%
  left_join(meta) %>% 
  ggplot(aes(x, y, colour = Genotype, group = sample)) +
  geom_line() +
  labs(
    x = "logCPM",
    y = "Density",
    colour = "Genotype"
  )+
  ggtitle("After filtering")

ggarrange(a, b, common.legend = TRUE)
```

```{r dge}
x <- featureCounts %>% 
  as.matrix() %>% 
 .[rowSums(cpm(.) >= 0.5) >= 8,] %>%
  DGEList(
    samples = tibble(sample = colnames(.)) %>%
      left_join(meta),
    genes = grGenes[rownames(.)] %>%
      as.data.frame() %>%
      dplyr::select(
        chromosome = seqnames, start, end, 
        gene_id, gene_name, gene_biotype, description, entrezid
      ) %>% 
      left_join(gcGene) %>% 
      as_tibble()
  ) %>%
  calcNormFactors()
```

## PCA

```{r}
x$counts %>% 
  cpm(log=TRUE) %>% 
  t() %>% 
  prcomp() %>% 
  autoplot(data = tibble(sample = rownames(.$x)) %>%
      left_join(x$samples),
    colour = "Genotype", 
    size = 6
  ) +
  theme(aspect.ratio = 1) +
  labs(colour = "Genotype") 

x$counts %>% 
  cpm(log=TRUE) %>% 
  t() %>% 
  prcomp() %>% 
  autoplot(data = tibble(sample = rownames(.$x)) %>%
      left_join(x$samples),
    colour = "sex", 
    size = 6
  ) +
  theme(aspect.ratio = 1) +
  labs(colour = "Sex") 

x$counts %>% 
  cpm(log=TRUE) %>% 
  t() %>% 
  prcomp() %>% 
  autoplot(data = tibble(sample = rownames(.$x)) %>%
      left_join(x$samples),
    colour = "Tank", 
    size = 6
  ) +
  theme(aspect.ratio = 1) +
  labs(colour = "Tank") 

  #ggsave("output/plots/PCAblue.png", width = 10, height = 10, units = "cm", dpi = 300, scale = 1.25)
```

# GC Length check
```{r}
design <- model.matrix(~Genotype + Tank, data = x$samples) %>% 
  set_colnames(gsub(colnames(.), pattern = "Genotype", replacement = ""))

fit_1 <- x %>% 
  estimateDisp(design) %>% 
  glmFit(design)

toptable_1 <- design %>% colnames() %>% .[2:3] %>% 
  sapply(function(x){
    glmLRT(fit_1, coef = x) %>%
      topTags(n = Inf) %>%
      .[["table"]] %>%
      as_tibble() %>%
      arrange(PValue) %>%
       mutate(
        coef = x,
        DE = FDR < 0.05
      ) %>% 
      dplyr::select(
        gene_name, logFC, logCPM, PValue, FDR, DE, everything()  
      )
      
  }, simplify = FALSE)

toptable_1 %>%
  bind_rows() %>% 
  ggplot(aes(y = -log10(PValue), x = logFC, colour = DE)) +
  geom_point(
    alpha = 0.5, size = 1.25
  ) +
  facet_wrap(~coef) +
  # geom_label_repel(
  #   aes(label = gene_name),
  #   data = .  %>% dplyr::filter(FDR < 0.05),
  #   show.legend = FALSE
  #   ) +
  coord_cartesian(xlim = c(-2.5,2.5), 
                  ylim = c(0, 20) )+
  theme(legend.position = "none") +
  scale_color_manual(values = c("grey50", "red")) 

toptable_1 %>%
  bind_rows() %>% 
  ggplot(aes(x = logCPM, y = logFC,  colour = DE)) +
  geom_point(
    alpha = 0.5
  ) +
  facet_wrap(~coef) +
  # geom_label_repel(
  #   aes(label = gene_name), 
  #   data = .  %>% dplyr::filter(FDR < 0.05), 
  #   show.legend = FALSE
  #   ) +
  theme(legend.position = "none") +
  scale_color_manual(values = c("grey50", "red"))


```

### Vis
```{r}
ggarrange(
toptable_1 %>% 
  bind_rows() %>% 
  mutate(rankstat = sign(logFC)*-log10(PValue)) %>% 
  ggplot(aes(x = length, y = rankstat)) +
  geom_point(
    aes(colour = DE),
    alpha = 0.5
  ) +
  geom_smooth(se = FALSE, method = "gam") +
  facet_grid(rows = vars(coef)) +
  theme_bw() +
  theme(legend.position = "none") +
  scale_color_manual(values = c("grey50", "red")) +
  scale_x_log10()+
  labs(x = "Average transcript length per gene",
       colour = "Differentially expressed?",
       y = "sign(logFC)*-log10(PValue)") +
  coord_cartesian(ylim = c(-10, 10)),

toptable_1 %>% 
  bind_rows() %>% 
  mutate(rankstat = sign(logFC)*-log10(PValue)) %>% 
  ggplot(aes(x = gc_content, y = rankstat)) +
  geom_point(
    aes(colour = DE),
    alpha = 0.5
  ) +
  geom_smooth(se = FALSE, method = "gam") +
  facet_grid(rows = vars(coef)) +
  theme_bw() +
  theme(legend.position = "none") +
  scale_color_manual(values = c("grey50", "red")) +
  coord_cartesian(ylim = c(-10,10)) + 
  labs(x = "Weighted average GC content (%) per gene", 
       colour = "Differentially expressed?", 
       y = "sign(logFC)*-log10(PValue)"),
common.legend = TRUE, 
labels = "AUTO"
) 
```

# CQN

Some 
```{r}
cqn <- 
  x %>% 
  with(
    cqn(
      counts = counts,
      x = genes$gc_content,
      lengths = genes$length,
      sizeFactors = samples$lib.size
    )
  )
# cre
logCPM <- cqn$y + cqn$offset
```

### plot out the model fits
```{r, fig.cap ="*Model fits for GC content and gene length under the CQN model. Variability is clearly visible at either end*"}
par(mfrow = c(1, 2))
cqnplot(cqn, n = 1, xlab = "GC Content")
cqnplot(cqn, n = 2, xlab = "Length")
par(mfrow = c(1, 1))
```

### PCA after cqn
PCA analysis was repeated. Similar clustering of samples was observed, with MPS-IIIA samples appearing to form a more distinct cluster from the rest of the samples. The sgsh/+ samples appear much more variable. 

```{r}
ggarrange(
  cpm(x, log = T) %>%
    t() %>%
    prcomp() %>% 
    autoplot(data = tibble(sample = rownames(.$x)) %>%
               left_join(x$samples),
             colour = "Genotype", 
             shape = "sex", 
             size = 4
    ) +
    scale_color_viridis_d(option = "turbo", end = 0.8) +
    theme(aspect.ratio = 1) +
    ggtitle("Before CQN"),
  
  logCPM %>% 
    t() %>% 
    prcomp() %>%
    autoplot(data = tibble(sample = rownames(.$x)) %>%
               left_join(x$samples),
             colour = "Genotype", 
             size = 4
    ) +
    scale_color_viridis_d(option = "turbo", end = 0.8) +
    theme(aspect.ratio = 1) +
    ggtitle("After CQN"),
  common.legend = T, 
  labels = "AUTO"
) 
#ggsave("plotsforpub/PCA_cqn.png", width = 18, height = 8, units = "cm", scale = 1.4)
```

## DE with cqn

The DE analysis was then repeated, this time including the offset generated by `cqn` in the model. 
```{r}
# Add the offset to the dge object 
x$offset <- cqn$glm.offset

# fit the glm
fit_cqn <- x %>% 
  estimateDisp(design) %>% 
  glmFit(design)

toptables_cqn <- design %>% colnames() %>% .[2:3] %>% 
  sapply(function(x){
    glmLRT(fit_cqn, coef = x) %>%
      topTags(n = Inf) %>%
      .[["table"]] %>%
      as_tibble() %>%
      arrange(PValue) %>%
       mutate(
        coef = x,
        DE = FDR < 0.05
      ) %>% 
      dplyr::select(
        gene_name, logFC, logCPM, PValue, FDR, DE, everything()  
      )
      
  }, simplify = FALSE)
```

## Visualisation
### Number of DE genes
```{r}
toptables_cqn %>% 
  lapply(function(x){
    x %>%
      mutate(Direction = case_when(
        sign(logFC) == "1" ~ "up",
        sign(logFC) == "-1" ~ "down"
      ))
  }) %>%
  bind_rows() %>%
  dplyr::filter(DE == TRUE) %>%
  ggplot(aes(y = `coef`)) +
  geom_bar(stat = "count",
           width = 0.5,
           aes(fill = Direction),
           position = "dodge") +
  theme(legend.position = "bottom") +
  ggtitle("Number of DE genes in each comparison") +
  labs(x = "Number of DE genes",
       y = "Contrast",
       colour = "Direction of change", 
       title = "Number of DE genes in each comparison")
```

### Volcano plot
```{r}
toptables_cqn %>%
  bind_rows() %>% 
  ggplot(aes(y = -log10(PValue), x = logFC, colour = DE)) +
  geom_point(
    alpha = 0.5, size = 1.25
  ) +
  facet_wrap(~coef) +
  # geom_label_repel(
  #   aes(label = gene_name),
  #   data = .  %>% dplyr::filter(FDR < 0.05),
  #   show.legend = FALSE
  #   ) +
  coord_cartesian(xlim = c(-2.5,2.5), 
                  ylim = c(0, 20) )+
  theme(legend.position = "none") +
  scale_color_manual(values = c("grey50", "red")) 
  #ggsave("output/plots/volvano.png", width = 18, height = 10, units = "cm", dpi = 300, scale = 1.5)
```

### MD plot
```{r}
toptables_cqn %>%
  bind_rows() %>% 
  ggplot(aes(x = logCPM, y = logFC)) +
  geom_point(
    aes(colour = DE),
    alpha = 0.5
  ) +
  facet_wrap(~coef, ncol = 1) +
  theme(legend.position = "none") +
  scale_color_manual(values = c("grey50", "red")) +
  geom_smooth(se = F)
```

## Recheck the %GC and length bias

The %GC and length bias data was checked again. Some gene length bias still remains. But this seems to be driven only at the ends, where there is less genes. So this should be ok.
```{r}
ggarrange(
toptables_cqn %>% 
  bind_rows() %>% 
  mutate(rankstat = sign(logFC)*-log10(PValue)) %>% 
  ggplot(aes(x = length, y = rankstat)) +
  geom_point(
    aes(colour = DE),
    alpha = 0.5
  ) +
  geom_smooth(se = FALSE, method = "gam") +
  facet_grid(rows = vars(coef)) +
  theme(legend.position = "none") +
  scale_color_manual(values = c("grey50", "red")) +
  scale_x_log10()+
  labs(x = "Average transcript length per gene",
       colour = "Differentially expressed?",
       y = "sign(logFC)*-log10(PValue)") +
  coord_cartesian(ylim = c(-10, 10)),

toptables_cqn %>% 
  bind_rows() %>% 
  mutate(rankstat = sign(logFC)*-log10(PValue)) %>% 
  ggplot(aes(x = gc_content, y = rankstat)) +
  geom_point(
    aes(colour = DE),
    alpha = 0.5
  ) +
  geom_smooth(se = FALSE, method = "gam") +
  facet_grid(rows = vars(coef)) +
  theme(legend.position = "none") +
  scale_color_manual(values = c("grey50", "red")) +
  coord_cartesian(ylim = c(-10,10)) + 
  labs(x = "Weighted average GC content (%) per gene", 
       colour = "Differentially expressed?", 
       y = "sign(logFC)*-log10(PValue)"),
common.legend = TRUE, 
labels = "AUTO"
) 
```

## Gene set enrichment analysis

I now am going to test the KEGG, gene ontology (GO), iron-responsive element (IRE), cell type markers and chromosomal location gene sets to determine whether any of them display changes to gene expression as a group. The gene sets were obtained from MSIGDB using the msigdbr package. 

### Gene sets

The KEGG and GO terms were obtained from msigdb using msigdbr. These gene sets were filtered to only contain the genes above the detectable threshold in this experiment. Also, any gene sets with less than 5 genes were omitted, as these probably arent very biologically meaninigful. The GO terms were also restricted to those with 3 or more steps back to the ontology root terms. 

The iron-responsive element (IRE) genes were obtained from Hin et al. 2021 (DOI: 10.3233/JAD-210200). The zebrafish IRE-containing genes were used, and were also filtered to only retain the detectable genes. 

The chromosome gene sets were generated from the DGE object. 

The cell type marker gene sets were obtained from the single-cell RNA-seq data of Jiang et al. 2021 (https://doi.org/10.3389/fcell.2021.743421)/

```{r}

# Obtain the KEGG gene sets. 
KEGG <- msigdbr("Danio rerio", category = "C2", subcategory = "CP:KEGG") %>% 
  left_join(x$genes, by = c("gene_symbol" = "gene_name")) %>% 
  dplyr::filter(gene_id %in% rownames(x)) %>% 
  distinct(gs_name, gene_id, .keep_all = TRUE) %>% 
  split(f = .$gs_name) %>%
  lapply(extract2, "gene_id")

# calculate the number of genes per gene set
sizes <- KEGG %>% 
  lapply(length) %>% 
  unlist %>% 
  as.data.frame() %>% 
  set_colnames( "n_genes") %>% 
  rownames_to_column("gs")


# retain gene sets with at least 5 genes in it
KEGG <- KEGG[sizes %>% dplyr::filter(n_genes > 5) %>% .$gs]

# GO Terms summaues that The UofA biohub made. 
goSummaries <- url("https://uofabioinformaticshub.github.io/summaries2GO/data/goSummaries.RDS") %>%
  readRDS() %>%
  mutate(
    Term = Term(id),
    gs_name = Term %>% str_to_upper() %>% str_replace_all("[ -]", "_"),
    gs_name = paste0("GO_", gs_name)
    )

minPath <- 3
# obtain the GO terms 
GO <- 
  msigdbr("Danio rerio", category = "C5") %>% 
  dplyr::filter(grepl(gs_name, pattern = "^GO")) %>% 
  left_join(x$genes, by = c("gene_symbol" = "gene_name")) %>% 
  dplyr::filter(gene_id %in% rownames(x)) %>% 
  mutate(gs_name = str_replace(gs_name, pattern = "GOBP_", replacement = "GO_")) %>% 
  mutate(gs_name = str_replace(gs_name, pattern = "GOMF_", replacement = "GO_")) %>%     
  mutate(gs_name = str_replace(gs_name, pattern = "GOCC_", replacement = "GO_")) %>%     
  left_join(goSummaries) %>% 
  dplyr::filter(shortest_path >= minPath) %>%
  distinct(gs_name, gene_id, .keep_all = TRUE) %>% 
  split(f = .$gs_name) %>%
  lapply(extract2, "gene_id")

# GOdf <-  msigdbr("Danio rerio", category = "C5") %>% 
#   dplyr::filter(grepl(gs_name, pattern = "^GO")) %>% 
#   left_join(x.ab$genes, by = c("gene_symbol" = "gene_name")) %>% 
#   dplyr::filter(gene_id %in% rownames(x)) %>% 
#   mutate(gs_name = str_replace(gs_name, pattern = "GOBP_", replacement = "GO_")) %>% 
#   mutate(gs_name = str_replace(gs_name, pattern = "GOMF_", replacement = "GO_")) %>%     
#   mutate(gs_name = str_replace(gs_name, pattern = "GOCC_", replacement = "GO_")) %>%     
#   left_join(goSummaries) %>% 
#   dplyr::filter(shortest_path >= minPath) %>%
#   distinct(gs_name, gene_id, .keep_all = TRUE) %>% 
#   dplyr::select(gs_name, gene_id)

sizes_GO <- GO %>%
  lapply(length) %>%
  unlist %>%
  as.data.frame() %>%
  set_colnames( "n_genes") %>%
  rownames_to_column("gs")


# retain gene sets with at least 5 genes in it
GO <- GO[sizes_GO %>% dplyr::filter(n_genes > 5) %>% .$gs]

ireGenes <- readRDS("data/gene_sets/zebrafishireGenes.rds") %>% 
  unlist() %>% 
  as.data.frame() %>% 
  rownames_to_column("ire") %>% 
  as_tibble() %>% 
  mutate(ire = str_extract(ire, pattern = "ire[:digit:]_[:alpha:]+")) %>% 
  set_colnames(c("ire", "gene_id")) %>% 
  dplyr::filter(gene_id %in% rownames(x)) %>% 
  split(f = .$ire) %>% 
  lapply(magrittr::extract2,"gene_id") 

chr <- x$genes %>% 
  dplyr::select(gene_id, chromosome) %>% 
  split(f = .$chromosome) %>%
  lapply(extract2, "gene_id") %>% 
  .[c("MT", seq(1:25))] # omit the ALT chromosmes and scaffolds


cell_type_markers <- 
  read_xlsx("data/gene_sets/Suppdata2_jiangetal_2021_fcelldev.xlsx", sheet = "Brain") %>% 
  dplyr::rename("gene_name" = gene) %>% 
  left_join(x$genes) %>% 
  dplyr::filter(gene_id %in% rownames(x)) %>% 
  split(f = .$`cell type`) %>% 
  lapply(extract2, "gene_id")
  

cell_type_markers %<>% 
  lapply(function(y) {
    y[y %in% rownames(x)]
  })

```

## GOSEQ 

I first will look at enrichment of GO terms within the DE genes using goseq. I like goseq as it lets you include a covariate. Since there was still a bit of bias left after cqn for gene length, i will use this as the covariate.  
```{r}
pwf <- toptables_cqn$`MPS-III` %>%
      with(
        nullp(
          DEgenes = structure(DE, names = gene_id), 
          bias.data = length
        )
      )

goseq.mps <- goseq(pwf, gene2cat = GO) %>% 
  as_tibble() %>% 
  dplyr::filter(numDEInCat > 0) %>% 
  mutate(FDR = p.adjust(over_represented_pvalue, method = "fdr")) %>% 
  dplyr::select(-under_represented_pvalue)
  
goseq.mps %>% 
  dplyr::filter(FDR <0.05) %>% 
  kable(caption = "Sign over-represented GO terms in MPS-IIIB DE genes") %>% 
  kable_styling(full_width = FALSE)

```

For the DE genes in the OEFAD brains, lots of GO terms are observed. But they generally have just 1 or 2 DE genes in the significant GO terms. So I dont think these are particularly meaningful. 

```{r}
pwf2 <- toptables_cqn$EOfAD %>%
      with(
        nullp(
          DEgenes = structure(DE, names = gene_id), 
          bias.data = length
        )
      )
goseq.eofad <- goseq(pwf2, gene2cat = GO) %>% 
  as_tibble() %>% 
  dplyr::filter(numDEInCat > 0) %>% 
  mutate(FDR = p.adjust(over_represented_pvalue, method = "fdr")) %>% 
  dplyr::select(-under_represented_pvalue)
  
goseq.eofad %>% 
  dplyr::slice(1:10)%>% 
  kable(caption = "Top 10 over-represented GO terms inEOfAD DE genes.") %>% 
  kable_styling(full_width = FALSE)
```


## FRY

Overrepresenation analysis using goseq relies on a hard threshold for gene to be called DE. This can lose information as there are genes which are almost DE (i.e. FDRp = 0.06), but are not interpreted in this way. 

Another method to test for enrichment of gene sets is to use ROAST. Roast uses a rotation-based approach to generate a null distribution of gene set enrichment scores by randomly permuting the samples or the gene labels in the data. Roast then calculates a gene set enrichment score for each gene set in the data, based on the correlation between the gene expression values and the sample labels. The gene set enrichment scores are compared to the null distribution to calculate the statistical significance of the enrichment. In this way, it does not requre a list of DEGs as input. 

Here, I will perform the fast approx of ROAST: fry, frmo the limma package on the KEGG gene sets. The top 10 gene sets in each mutant zebrafish are shown below. 

### KEGG 
```{r}
fryKEGG <- 
  design %>% colnames() %>% .[2:3] %>% 
  sapply(function(y) {
    logCPM %>% 
      fry(
        index = KEGG,
        design = design, 
        contrast = y, 
        sort = "mixed"
      ) %>% 
      rownames_to_column("pathway") %>% 
      as_tibble() %>% 
      mutate(coef = y)
  }, simplify = FALSE)

fryKEGG$`MPS-III` %>% 
  dplyr::slice(1:10) %>% 
  mutate(pathway = str_remove(pathway, pattern = "KEGG_")) %>% 
  ggplot(aes(x = -log10(PValue.Mixed), y = reorder(pathway, -FDR.Mixed))) +
  geom_col(aes(fill = FDR.Mixed < 0.05)) +
  scale_fill_manual(values = c("grey50", 'red')) +
  theme_pubr() +
  labs(fill = "FDR < 0.05", 
       x = "-log10(PValue)", 
       y = "")

fryKEGG$EOfAD %>% 
  dplyr::slice(1:10) %>% 
  mutate(pathway = str_remove(pathway, pattern = "KEGG_")) %>% 
  ggplot(aes(x = -log10(PValue.Mixed), y = reorder(pathway, -FDR.Mixed))) +
  geom_col(aes(fill = FDR.Mixed < 0.05)) +
  scale_fill_manual(values = c("grey50", 'red')) +
  theme_pubr() +
  labs(fill = "FDR < 0.05", 
       x = "-log10(PValue)", 
       y = "")  


```


### GO
```{r}
fry_GO <-  design %>% colnames() %>% .[2:3] %>% 
  sapply(function(y) {
    logCPM %>% 
      fry(
        index = GO,
        design = design, 
        contrast = y, 
        sort = "mixed"
      ) %>% 
      rownames_to_column("pathway") %>% 
      as_tibble() %>% 
      mutate(coef = y, 
             sig = FDR.Mixed < 0.05)
  }, simplify = FALSE)
```

```{r}
toptables_cqn %>% 
  bind_rows() %>% 
  dplyr::filter(gene_id %in% GO$GO_REGULATION_OF_MICROGLIAL_CELL_ACTIVATION) %>% 
  dplyr::select(gene_name, logFC, coef) %>% 
  dplyr::distinct(gene_name, coef, .keep_all = T) %>% 
  spread(key = "coef", value = "logFC") %>% 
  column_to_rownames("gene_name") %>% 
  t() %>% 
  pheatmap(color = colorRampPalette(rev(brewer.pal(n = 5,
                                               name = "RdBu")))(100), 
           main = "GO_REGULATION_OF_MICROGLIAL_CELL_ACTIVATION: FDR = 6.44e-5", 
           breaks = c(seq(min(.), 0, length.out=ceiling(100/2) + 1), 
              seq(max(.)/100, max(.), length.out=50)), 
           cellheight = 30, cellwidth = 12,angle_col = 45,
           treeheight_row = 0, treeheight_col = 0
           )


toptables_cqn %>% 
  bind_rows() %>% 
    dplyr::filter(gene_id %in% cell_type_markers$`Oligodendrocyte-enriched`) %>% 
  dplyr::select(gene_name, logFC, coef) %>% 
  dplyr::distinct(gene_name, coef, .keep_all = T) %>% 
  spread(key = "coef", value = "logFC") %>% 
  column_to_rownames("gene_name") %>% 
  t() %>% 
  pheatmap(color = colorRampPalette(rev(brewer.pal(n = 5,
                                               name = "RdBu")))(100), 
           breaks = c(seq(min(.), 0, length.out=ceiling(100/2) + 1), 
              seq(max(.)/100, max(.), length.out=50)), 
           cellheight = 30, cellwidth = 12,angle_col = 45,
           treeheight_row = 0, treeheight_col = 0
           )
```

### IRE

```{r}
fryIRE <- design %>% colnames() %>% .[2:3] %>% 
  sapply(function(y) {
    logCPM %>% 
      fry(
        index = ireGenes,
        design = design, 
        contrast = y, 
        sort = "directional"
      ) %>% 
      rownames_to_column("pathway") %>% 
      as_tibble() %>% 
      mutate(coef = y)
  }, simplify = FALSE)

fryIRE %>% 
  bind_rows() %>% 
  dplyr::select(pathway, coef, pvalue = PValue.Mixed, FDR = FDR.Mixed) %>% 
  ggplot(aes(y = pathway, x = -log10(pvalue))) + 
  geom_col(aes(fill = coef, alpha = ), 
           position = "dodge") +
  scale_fill_viridis_d(end = 0.75) +
  geom_vline(xintercept = 1.88) +
  annotate(geom = "label", x = 1.8, y = "ire5_HQ", label = "FDR = 0.05")
  #ggsave("output/plots/AB_ire.png", width = 20, height = 10, units = "cm", dpi = 400, scale = 2)


toptables_cqn %>% 
  bind_rows() %>% 
    dplyr::filter(gene_id %in% ireGenes$ire3_all) %>% 
  dplyr::select(gene_name, logFC, coef) %>% 
  dplyr::distinct(gene_name, coef, .keep_all = T) %>% 
  spread(key = "coef", value = "logFC") %>% 
  column_to_rownames("gene_name") %>% 
  t() %>% 
  pheatmap(color = colorRampPalette(rev(brewer.pal(n = 5,
                                               name = "RdBu")))(100), 
           main = "3' IRE genes: FDR = 0.066", 
           breaks = c(seq(min(.), 0, length.out=ceiling(100/2) + 1), 
              seq(max(.)/100, max(.), length.out=50)), 
           cellheight = 30, 
           cellwidth = 1,
           angle_col = 45,
           treeheight_row = 0, 
           treeheight_col = 0, 
           show_colnames = F
           )
```

### Cell type prop check

Since this experiment was performed on whole larvae, we need to have a look at whether changes to cell type proportions could explain cause some artefactual changes to gene expression. We have done this previously by assessing the expression of marker genes of specific cell types. So this is what I will do here. I have taken the lists of genes frmo day 5 only from the zebrafish single cell atlas (Farnsworth et al. 2020) which contain at least 10 genes. This leaves  `r length(cell_type_markers)` gene sets (cell types) to test.

```{r}
celltype <- design %>% colnames() %>% .[2:3] %>% 
  sapply(function(y) {
    logCPM %>% 
      fry(
        index = cell_type_markers,
        design = design, 
        contrast = y, 
        sort = "directional"
      ) %>% 
      rownames_to_column("pathway") %>% 
      as_tibble() %>% 
      mutate(coef = y) %>% 
      dplyr::select(1:5, coef)
  }, simplify = FALSE)

celltype %>% 
  bind_rows() %>% 
  ggplot(aes(y = pathway, x = -log10(PValue))) + 
  geom_col(aes(fill = coef), 
           position = "dodge") +
  scale_fill_viridis_d(end = 0.75) +
  geom_vline(xintercept = 1.7) +
  annotate(geom = "label", x = 1.2, y = 4.4, label = "FDR = 0.05") 
```


### Chrosomosomes gene sets

```{r}
fry_chr <- design %>% colnames() %>% .[2:3] %>% 
  sapply(function(y) {
    logCPM %>% 
      fry(
        index = c(chr),
        design = design, 
        contrast = y, 
        sort = "mixed"
      ) %>% 
      rownames_to_column("pathway") %>% 
      as_tibble() %>% 
      mutate(coef = y)
  }, simplify = FALSE)


fry_chr %>% 
  bind_rows() %>% 
  mutate(pathway = factor(pathway, levels = c("MT", seq(1:25)))) %>% 
  ggplot(aes(y = pathway, x = -log10(PValue.Mixed), fill = coef)) +
  geom_col(position = "dodge")
  
```

An enrichment of DE genes is observed on the same chromosome as the mutation. This is obsevred in the manhat plots below. 

```{r}
manhat_input <- 
	toptables_cqn %>% 
  bind_rows() %>% 
	left_join(x$genes %>% 
			  	as_tibble() %>% 
			  	dplyr::select(start, end, chromosome, gene_id) %>% 
			  	group_by(gene_id) %>% 
			  	mutate(mid = mean(c(start, end)))
	) %>% 
  dplyr::filter(chromosome %in% seq(1:25)) %>% 
  group_by(gene_id) %>% 
  mutate(mid = mean(c(start, end))) %>% 
  ungroup %>% 
  mutate(chromosome = factor(chromosome, levels = seq(1:25)))  %>% 
  group_by(chromosome) %>% 
  summarise(chrLen = max(mid)) %>% 
  mutate(chrSt = cumsum(chrLen)-chrLen) %>% 
	dplyr::select(-chrLen) %>% 
  left_join(toptable_1 %>% 
              bind_rows() %>% 
              left_join(x$genes %>% 
                          as_tibble() %>% 
                          dplyr::select(start, end, chromosome, gene_id) %>% 
                          group_by(gene_id) %>% 
                          mutate(mid = mean(c(start, end)))
              )
  ) %>% 
  group_by(gene_id) %>% 
  mutate(mid = mean(c(start, end))) %>% 
  ungroup %>% 
  dplyr::arrange(mid, chromosome) %>% 
  mutate(midCum = chrSt + mid)

axis <- manhat_input %>%
  group_by(chromosome) %>%
  dplyr::arrange(chromosome) %>%
  summarize(center = (max(midCum) + min(midCum)) / 2) %>%
  mutate(
    seqnames = as.numeric(chromosome),
    colour = rep(c("grey40", "black"), length.out = 25)
  )

ggplot(manhat_input, aes(x = midCum, y = -log10(PValue))) +
  geom_point(aes(color=chromosome), alpha = 0.5, size = 1) +
  geom_point(alpha = 0.5, size = 1, colour = "red", 
             data = . %>% 
               dplyr::filter(coef == "MPS-III" & chromosome == 24)) +
  geom_point(alpha = 0.5, size = 1, colour = "red", 
             data = . %>% 
               dplyr::filter(coef ==  "EOfAD" & chromosome == 17)) +
  scale_color_manual(values = axis$colour) +
  scale_x_continuous(label = axis$chromosome, breaks = axis$center) +
  scale_y_continuous(limits =c(0, 10)) + # zoom in
  facet_wrap(~coef, ncol = 1) +
  labs(x = "Chromosome", y = expression(paste(-log[10], "(p)"))) +
  theme( 
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank()
  ) 
```

## Exploration with HMP

Im not really detecting anything using fry in the EOfAD mutants. This is consistent with all my previous analyses. I will use the harmonic mean p val method which combines individual p vals from 3 gene set enrichment analysis methods: fry, camera and gsea. 

```{r}
# Make a custom function for calculating the HMP.  
runHMP = function(gene.set.obj, 
                  design,
                  logCPM.df, 
                  toptable) {
fry <- 
  design %>% colnames() %>% .[2:3] %>% 
  sapply(function(y) {
  logCPM.df %>% 
      fry(
        index = gene.set.obj,
        design = design, 
        contrast = y, 
        sort = "mixed"
      ) %>% 
      rownames_to_column("pathway") %>% 
      as_tibble() %>% 
      mutate(coef = y)
  }, simplify = FALSE)

camera <- 
  design %>% colnames() %>% .[2:3] %>% 
  sapply(function(y) {
    logCPM.df %>% 
      camera(
        index = gene.set.obj,
        design = design, 
        contrast = y
      ) %>% 
      rownames_to_column("pathway") %>% 
      as_tibble() %>% 
      mutate(coef = y)
  }, simplify = FALSE)

ranks <- 
   sapply(toptable, function(y) {
     y %>% 
       mutate(rankstat = sign(logFC) * log10(1/PValue)) %>% 
       arrange(rankstat) %>% 
       dplyr::select(c("gene_id", "rankstat")) %>% #only want the Pvalue with sign
       with(structure(rankstat, names = gene_id)) %>% 
       rev() # reverse so the start of the list is upregulated genes
   }, simplify = FALSE)

# Run fgsea

# set a seed for a reproducible result
set.seed(33)
fgsea <- ranks %>%
  sapply(function(x){
    fgseaMultilevel(stats = x, pathways = gene.set.obj) %>%
      as_tibble() %>%
      dplyr::rename(FDR = padj) %>%
      mutate(padj = p.adjust(pval, "bonferroni")) %>%
      dplyr::select(pathway, pval, FDR, padj, everything()) %>%
      arrange(pval) %>%
      mutate(sig = padj < 0.05)
  }, simplify = F)

fgsea$`MPS-III` %<>% 
  mutate(coef = "MPS-III")
fgsea$`EOfAD` %<>% 
  mutate(coef = "EOfAD")

fry %>% 
  bind_rows() %>% 
  dplyr::select(pathway, PValue.Mixed, coef) %>% 
  dplyr::rename(fry_p = PValue.Mixed) %>% 
  left_join(camera %>% 
              bind_rows() %>% 
              dplyr::select(pathway, PValue, coef), 
            by = c("pathway", "coef")) %>% 
  dplyr::rename(camera_p = PValue) %>% 
  left_join(fgsea %>% 
              bind_rows() %>% 
              dplyr::select(pathway, pval, coef), 
            by = c("pathway", "coef")) %>% 
  dplyr::rename(fgsea_p = pval) %>% 
  bind_rows() %>% 
  nest(p = one_of(c("fry_p", "camera_p", "fgsea_p"))) %>% 
 mutate(harmonic_p = vapply(p, function(x){
        x <- unlist(x)
        x <- x[!is.na(x)]
        p.hmp(x, L = 4)
      }, numeric(1))
 ) %>% 
  unnest() %>% 
  mutate(harmonic_p_FDR = p.adjust(harmonic_p, "fdr"),
         sig = harmonic_p_FDR < 0.05) %>% 
  arrange(harmonic_p) 

}
```

```{r}
hmp.fry <- runHMP(gene.set.obj = KEGG, 
       design = design, 
       logCPM.df = logCPM, 
       toptable = toptables_cqn)

sigpaths <- hmp.fry %>% 
  dplyr::filter(sig == TRUE) %>% 
  .$pathway

sig_in <- hmp.fry %>% 
  dplyr::select(pathway, coef, sig) %>% 
  spread(key = coef, value = sig ) %>% 
  mutate(sig_in = case_when(
    EOfAD == FALSE & `MPS-III` == TRUE ~ "mpsiiionly",
    EOfAD == FALSE & `MPS-III` == FALSE ~ "neither",
    EOfAD == TRUE & `MPS-III` == TRUE ~ "both",
    EOfAD == TRUE & `MPS-III` == F ~ "fas only"
  )) %>% 
  dplyr::select(pathway, sig_in)

hmp.fry %>% 
  dplyr::filter(pathway %in% sigpaths) %>% 
  left_join(sig_in) %>% 
  ggplot(aes(y = pathway, x = -log10(harmonic_p))) +
  geom_col(aes(fill = coef,alpha = sig), 
           position = "dodge") +
  geom_vline(xintercept = -log10(3.0000e-03)) + 
  scale_alpha_manual(values = c(0.5, 1))+
  facet_wrap(~sig_in, scales = "free_y", ncol = 1, strip.position = "right")
```


```{r}
toptables_cqn %>% 
  bind_rows() %>% 
    dplyr::filter(gene_id %in% KEGG$KEGG_RIBOSOME) %>% 
  dplyr::select(gene_name, logFC, coef) %>% 
  dplyr::distinct(gene_name, coef, .keep_all = T) %>% 
  spread(key = "coef", value = "logFC") %>% 
  column_to_rownames("gene_name") %>% 
  t() %>% 
  pheatmap(color = colorRampPalette(rev(brewer.pal(n = 5,
                                               name = "RdBu")))(100), 
           main = "KEGG_RIBOSOME", 
           breaks = c(seq(min(.), 0, length.out=ceiling(100/2) + 1), 
              seq(max(.)/100, max(.), length.out=50)), 
           cellheight = 30, 
           cellwidth = 3,
           angle_col = 45,
           treeheight_row = 0, 
           treeheight_col = 0, 
           show_colnames = F
           )

toptables_cqn %>% 
  bind_rows() %>% 
    dplyr::filter(gene_id %in% KEGG$KEGG_RIBOSOME) %>% 
  dplyr::select(gene_name, logFC, coef) %>% 
  dplyr::distinct(gene_name, coef, .keep_all = T) %>% 
  spread(key = "coef", value = "logFC") %>% 
  column_to_rownames("gene_name") %>% 
  ggscatter(x = "EOfAD", y = "MPS-III", 
          add = "reg.line", 
          conf.int = TRUE, 
          cor.coef = TRUE, 
          cor.method = "spearman",
  )

toptables_cqn %>% 
  bind_rows() %>% 
    dplyr::filter(gene_id %in% KEGG$KEGG_OXIDATIVE_PHOSPHORYLATION) %>% 
  dplyr::select(gene_name, logFC, coef) %>% 
  dplyr::distinct(gene_name, coef, .keep_all = T) %>% 
  spread(key = "coef", value = "logFC") %>% 
  column_to_rownames("gene_name") %>% 
  ggscatter(x = "EOfAD", y = "MPS-III", 
          add = "reg.line", 
          conf.int = TRUE, 
          cor.coef = TRUE, 
          cor.method = "spearman",
  )

toptables_cqn %>% 
  bind_rows() %>% 
    dplyr::filter(gene_id %in% KEGG$KEGG_PARKINSONS_DISEASE) %>% 
  dplyr::select(gene_name, logFC, coef) %>% 
  dplyr::distinct(gene_name, coef, .keep_all = T) %>% 
  spread(key = "coef", value = "logFC") %>% 
  column_to_rownames("gene_name") %>% 
  ggscatter(x = "EOfAD", y = "MPS-III", 
          add = "reg.line", 
          conf.int = TRUE, 
          cor.coef = TRUE, 
          cor.method = "spearman",
  )

toptables_cqn %>% 
  bind_rows() %>% 
    dplyr::filter(gene_id %in% KEGG$KEGG_OXIDATIVE_PHOSPHORYLATION) %>% 
  dplyr::select(gene_name, logFC, coef) %>% 
  dplyr::distinct(gene_name, coef, .keep_all = T) %>% 
  spread(key = "coef", value = "logFC") %>% 
  column_to_rownames("gene_name") %>% 
  ggscatter(x = "EOfAD", y = "MPS-III", 
          add = "reg.line", 
          conf.int = TRUE, 
          cor.coef = TRUE, 
          cor.method = "spearman",
  )
```

## HMP IRE

```{r}
hmp.ire <- runHMP(gene.set.obj = ireGenes, 
       design = design, 
       logCPM.df = logCPM, 
       toptable = toptables_cqn)

hmp.ire %>% 
  ggplot(aes(y = pathway, x = -log10(harmonic_p))) +
  geom_col(aes(fill = coef,
               alpha = sig), 
           position = "dodge") +
  geom_vline(xintercept = -log10(3.0000e-03)) + 
  scale_alpha_manual(values = c(0.5, 1))

```

# Post-hoc power calc
```{r}
set.seed(2016)

mu <- x$counts %>%
  as.data.frame() %>%
  .[grepl(colnames(x$counts), pattern = "wt")] %>%
  rowMeans()

disp <-  x %>% estimateDisp() %>% . $tagwise.dispersion

fc <- function(y){exp(rnorm(y, log(1), 0.5*log(2)))}

power <- ssizeRNA_vary(nGenes = dim(x)[1], # Num detectable genes
              pi0 = 0.9, # Proportion of non-DE genes
              m = 40, # pseudo sample size
              mu = mu, # Average read count for each gene in control group. Calculated from this current dataset)
              disp = disp, # Dispersion parameter for each gene
              fc = fc, # Fold change per gene
              fdr = 0.05, # FDR level to control
              power = 0.7, # power level
              maxN = 40,
              replace = T
                )

power$power %>% 
  as_tibble() %>% 
  set_colnames(c("n", "Power")) %>% 
  ggplot(aes(x = n, y = Power)) +
  geom_point() + 
  geom_line() +
  geom_vline(xintercept = 8, colour = "red") +
  annotate(geom = "label", label = "n = 8, ~53% power", 
           x = 15, y = 0.4, colour = "red")
```

## Data export
```{r}
x %>% saveRDS("data/R_objects/adult_brain/dge.rds")
logCPM %>% saveRDS("data/R_objects/adult_brain/logcpm.rds")
toptables_cqn %>% saveRDS("data/R_objects/adult_brain/toptablescqn.rds")
hmp.fry %>% saveRDS("data/R_objects/adult_brain/hmp_kegg.rds")
hmp.ire %>% saveRDS("data/R_objects/adult_brain/hmp_ire.rds")
```




