---
title: "analysis_adultbrain"
author: "Karissa Barthelson"
date: "2023-03-01"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  autodep = TRUE,
  echo = TRUE,
  warning = FALSE,
  message = FALSE,
  fig.align = "center", 
  out.width ="75%", 
  out.height = "75%"
)
```

```{r loadLibs}
library(tidyverse)
library(magrittr)
library(readxl)
library(ngsReports)
library(AnnotationHub)
library(pander)
library(scales)
library(pheatmap)
library(ggpubr)
library(msigdbr)
library(scales)
library(ggrepel)
library(ggfortify)
library(RColorBrewer)
library(ComplexHeatmap)
library(tidyHeatmap)
library(circlize)
library(UpSetR)
library(edgeR)
library(goseq)
library(fgsea)
library(cqn)
library(kableExtra)
library(harmonicmeanp)
library(ssizeRNA)

theme_set(
  theme_bw() +
    theme(
      plot.title = element_text(hjust = 0.5)
    ))
```

# Read in data 


```{r anno}
ah <- AnnotationHub() %>%
	subset(species == "Danio rerio") %>%
	subset(rdataclass == "EnsDb")

ensDb <- ah[["AH83189"]] # for release 101, latest version and the alignment
grTrans <- transcripts(ensDb)
trLengths <- exonsBy(ensDb, "tx") %>%
	width() %>%
	vapply(sum, integer(1))
mcols(grTrans)$length <- trLengths[names(grTrans)]
gcGene <- grTrans %>%
  mcols() %>%
  as.data.frame() %>%
  dplyr::select(gene_id, tx_id, gc_content, length) %>%
  as_tibble() %>%
  group_by(gene_id) %>%
  summarise(
    gc_content = sum(gc_content*length) / sum(length),
    length = ceiling(median(length))
  )
grGenes <- genes(ensDb)
mcols(grGenes) %<>%
  as.data.frame() %>%
  left_join(gcGene) %>%
  as.data.frame() %>%
  DataFrame()
```

```{r meta}
# readin meta and tidy up columns
meta <- read_excel("data/adult_brain/karissas_metadata.xlsx", sheet = "onlyseq") %>% 
  mutate(genotype = case_when(
    `usable genotype?` == "wt" ~ "wt", 
    `usable genotype?` == "EOfAD" ~ "EOfADlike", 
    `usable genotype?` == "MPS-III" ~ "MPSIIIB"
  ) %>% 
    factor(levels = c("wt",  "EOfADlike", "MPSIIIB")), 
  tank = as.factor(tank), 
  sex  = as.factor(sex),
  sample = paste0(fish, "_", genotype), 
  group = paste0(genotype, "_", sex) %>% as.factor(),
  age = "6 m adult brain",
  RIN = as.numeric(`RIN/DIN`)
  ) %>% 
  dplyr::select(
    # reorder the metadata nicely
    fish,genotype,sex,group,batch,tank,RIN,everything()
    ) %>% 
  as_tibble()
```

```{r}
# read in counts and cleanup
featureCounts <- 
  read_delim("data/adult_brain/05_featureCounts/counts.out", delim = "\t", skip = 1) %>%
  set_names(basename(names(.))) %>% 
  set_names(names(.) %>% str_remove(pattern = "_S[0-9]+_merged.Aligned.sortedByCoord.dedup.out.bam")) %>% 
  as_tibble() %>% 
  dplyr::select(-c(Chr, Start, End, Length, Strand)) %>% 
  gather(key = "ULN", value = "counts", starts_with("22")) %>% 
  left_join(meta) %>% 
  mutate(sample = paste0(fish, "_", genotype)) %>% 
  dplyr::select(Geneid, counts, sample) %>% 
  spread(key = "sample", value = "counts") %>% 
  column_to_rownames("Geneid") %>% 
  .[,1:24]
```


# filter lowly expressed genes

Genes which are lowly expressed are considered uninformative for de analysis. Here, we set the threshold to be a min CPM of 0.5 (following the 10/min lib size rule proposed by gordon smyth). The effect of filtering is found below.

DGEList objects were created, which brings together the sample metadata and counts in each experiment. TMM normalisation to the counts is also applied when making these objects. 

```{r filt}
a <- featureCounts %>% 
    cpm(log = TRUE) %>%
    as.data.frame() %>% 
  pivot_longer(
    cols = everything(),
    names_to = "sample",
    values_to = "logCPM"
  ) %>%
  split(f = .$sample) %>%
  lapply(function(x){
    d <- density(x$logCPM)
    tibble(
      sample = unique(x$sample),
      x = d$x,
      y = d$y
    )
  }) %>%
  bind_rows() %>%
  left_join(meta) %>% 
  ggplot(aes(x, y, colour = genotype, group = sample)) +
  geom_line() +
  labs(
    x = "logCPM",
    y = "Density",
    colour = "genotype"
  )+
  ggtitle("Before filtering") +
  theme_bw()
b <- featureCounts %>% 
  .[rowSums(cpm(.) >= 0.5) >= 8,] %>% 
    cpm(log = TRUE) %>%
    as.data.frame() %>% 
  pivot_longer(
    cols = everything(),
    names_to = "sample",
    values_to = "logCPM"
  ) %>%
  split(f = .$sample) %>%
  lapply(function(x){
    d <- density(x$logCPM)
    tibble(
      sample = unique(x$sample),
      x = d$x,
      y = d$y
    )
  }) %>%
  bind_rows() %>%
  left_join(meta) %>% 
  ggplot(aes(x, y, colour = genotype, group = sample)) +
  geom_line() +
  labs(
    x = "logCPM",
    y = "Density",
    colour = "genotype"
  )+
  ggtitle("After filtering")

ggarrange(a, b, common.legend = TRUE)
```

```{r dge}
x <- featureCounts %>% 
  as.matrix() %>% 
 .[rowSums(cpm(.) >= 0.5) >= 8,] %>%
  DGEList(
    samples = tibble(sample = colnames(.)) %>%
      left_join(meta),
    genes = grGenes[rownames(.)] %>%
      as.data.frame() %>%
      dplyr::select(
        chromosome = seqnames, start, end, 
        gene_id, gene_name, gene_biotype, description, entrezid
      ) %>% 
      left_join(gcGene) %>% 
      as_tibble()
  ) %>%
  calcNormFactors()
```

## PCA
I first want to explore the overall similarity between samples by PCA. No distinct clusters of samples are observed. However, there might be some subtle seperatation of the MPS IIIB mutants. 
```{r}
x %>% 
  cpm(log=TRUE) %>% 
  t() %>% 
  prcomp() %>% 
  autoplot(data = tibble(sample = rownames(.$x)) %>%
      left_join(x$samples),
    colour = "genotype", 
    shape = "sex",
    size = 6
  ) +
  theme(aspect.ratio = 1) +
  labs(colour = "genotype") 

x$counts %>% 
  cpm(log=TRUE) %>% 
  t() %>% 
  prcomp() %>% 
  autoplot(data = tibble(sample = rownames(.$x)) %>%
      left_join(x$samples),
    colour = "sex", 
    size = 6
  ) +
  theme(aspect.ratio = 1) +
  labs(colour = "Sex") 

x$counts %>% 
  cpm(log=TRUE) %>% 
  t() %>% 
  prcomp() %>% 
  autoplot(data = tibble(sample = rownames(.$x)) %>%
      left_join(x$samples),
    colour = "tank", 
    size = 6
  ) +
  theme(aspect.ratio = 1) +
  labs(colour = "Tank") 
```

### within sexes
Since it is a bit hard to see whats going on within sexes, I will next perform PCAs on each sex seperately

```{r}
fem.samples <- x$samples %>% 
  dplyr::filter(sex == "F") %>% 
  rownames()

male.samples <- x$samples %>% 
  dplyr::filter(sex == "M") %>% 
  rownames()

x %>% 
  cpm(log = TRUE) %>% 
  .[,fem.samples] %>% 
  t %>% 
  prcomp() %>% 
  autoplot(data = tibble(sample = rownames(.$x)) %>%
             left_join(x$samples),
           colour = "genotype",
           shape = "sex", 
           size = 6
  ) +
  theme(aspect.ratio = 1) +
  labs(
    title = "Females"
  )

x %>% 
  cpm(log = TRUE) %>% 
  .[,male.samples] %>% 
  t %>% 
  prcomp() %>% 
  autoplot(data = tibble(sample = rownames(.$x)) %>%
             left_join(x$samples),
           colour = "genotype",
           shape = "sex", 
           size = 6
  ) +
  theme(aspect.ratio = 1) +
  labs(
    title = "Males"
  )
```


# Initial DE analysis to check for %GC and length bias

For this analysis, I will keep the model matrix simple and look at only effect of genotype. This should be sufficient to look for bias due to GC and length, ans so whether CQN is required. 

```{r}
design.simple <- model.matrix(~genotype, data = x$samples) %>% 
  set_colnames(gsub(colnames(.), pattern = "genotype", replacement = ""))

fit_1 <- x %>% 
  estimateDisp(design.simple) %>% 
  glmFit(design.simple)

toptable_1 <- design.simple %>% colnames() %>% .[2:3] %>%
  sapply(function(x){
    glmLRT(fit_1, coef = x) %>%
      topTags(n = Inf) %>%
      .[["table"]] %>%
      as_tibble() %>%
      arrange(PValue) %>%
       mutate(
        coef = x,
        DE = FDR < 0.05
      ) %>% 
      dplyr::select(
        gene_name, logFC, logCPM, PValue, FDR, DE, everything()  
      )
      
  }, simplify = FALSE)

toptable_1 %>%
  bind_rows() %>% 
  ggplot(aes(y = -log10(PValue), x = logFC, colour = DE)) +
  geom_point(
    alpha = 0.5, size = 1.25
  ) +
  facet_wrap(~coef) +
  # geom_label_repel(
  #   aes(label = gene_name),
  #   data = .  %>% dplyr::filter(FDR < 0.05),
  #   show.legend = FALSE
  #   ) +
  coord_cartesian(xlim = c(-2.5,2.5), 
                  ylim = c(0, 20) )+
  theme(legend.position = "none") +
  scale_color_manual(values = c("grey50", "red")) 

toptable_1 %>%
  bind_rows() %>% 
  ggplot(aes(x = logCPM, y = logFC,  colour = DE)) +
  geom_point(
    alpha = 0.5
  ) +
  
  facet_wrap(~coef) +
  # geom_label_repel(
  #   aes(label = gene_name), 
  #   data = .  %>% dplyr::filter(FDR < 0.05), 
  #   show.legend = FALSE
  #   ) +
  theme(legend.position = "none") +
  scale_color_manual(values = c("grey50", "red"))


```

### Vis
```{r}
ggarrange(
toptable_1 %>% 
  bind_rows() %>% 
  mutate(rankstat = sign(logFC)*-log10(PValue)) %>% 
  ggplot(aes(x = length, y = rankstat)) +
  geom_point(
    aes(colour = DE),
    alpha = 0.5
  ) +
  geom_smooth(se = FALSE, method = "gam") +
  facet_grid(rows = vars(coef)) +
  theme_bw() +
  theme(legend.position = "none") +
  scale_color_manual(values = c("grey50", "red")) +
  scale_x_log10()+
  labs(x = "Average transcript length per gene",
       colour = "Differentially expressed?",
       y = "sign(logFC)*-log10(PValue)") +
  coord_cartesian(ylim = c(-10, 10)),

toptable_1 %>% 
  bind_rows() %>% 
  mutate(rankstat = sign(logFC)*-log10(PValue)) %>% 
  ggplot(aes(x = gc_content, y = rankstat)) +
  geom_point(
    aes(colour = DE),
    alpha = 0.5
  ) +
  geom_smooth(se = FALSE, method = "gam") +
  facet_grid(rows = vars(coef)) +
  theme_bw() +
  theme(legend.position = "none") +
  scale_color_manual(values = c("grey50", "red")) +
  coord_cartesian(ylim = c(-10,10)) + 
  labs(x = "Weighted average GC content (%) per gene", 
       colour = "Differentially expressed?", 
       y = "sign(logFC)*-log10(PValue)"),
common.legend = TRUE, 
labels = "AUTO"
) 
```

# CQN

Some bias is observed, and so CQN will be applied. 

```{r}
cqn <- 
  x %>% 
  with(
    cqn(
      counts = counts,
      x = genes$gc_content,
      lengths = genes$length,
      sizeFactors = samples$lib.size
    )
  )
# 
logCPM <- cqn$y + cqn$offset
```

### plot out the model fits
```{r, fig.cap ="*Model fits for GC content and gene length under the CQN model. Variability is clearly visible at either end*"}
par(mfrow = c(1, 2))
cqnplot(cqn, n = 1, xlab = "GC Content")
cqnplot(cqn, n = 2, xlab = "Length")
par(mfrow = c(1, 1))
```

### PCA after cqn
PCA analysis was repeated. Similar clustering of samples was observed, with MPS-IIIA samples appearing to form a more distinct cluster from the rest of the samples. The sgsh/+ samples appear much more variable. 

```{r}
genocols = c("#000000", "#DBDB2F", "#377EB8")

ggarrange(
  cpm(x, log = T) %>%
    t() %>%
    prcomp() %>% 
    autoplot(data = tibble(sample = rownames(.$x)) %>%
               left_join(x$samples),
             colour = "genotype", 
             shape = "sex", 
             size = 4
    ) +
    scale_colour_manual(values = genocols) +
    theme(aspect.ratio = 1) +
    ggtitle("Before CQN"),
  
  logCPM %>% 
    t() %>% 
    prcomp() %>%
    autoplot(data = tibble(sample = rownames(.$x)) %>%
               left_join(x$samples),
             colour = "genotype", 
             size = 4
    ) +
    scale_colour_manual(values = genocols) +
    theme(aspect.ratio = 1) +
    ggtitle("After CQN"),
  common.legend = T, 
  labels = "AUTO"
) 
#ggsave("plotsforpub/PCA_cqn.png", width = 18, height = 8, units = "cm", scale = 1.4)
```

## DE with cqn

The DE analysis was then repeated, this time including the offset generated by `cqn` in the model. 

Will also create a new design and contrasts matrices to be able to determine the DEGs within sexes. 

```{r}
# Add the offset to the dge object 
x$offset <- cqn$glm.offset

# new desing matrix
design.bygroup <- 
  model.matrix(~0 + group, data = x$samples %>% 
                 droplevels()) %>% 
   set_colnames(gsub(colnames(.), pattern = "group", replacement = ""))


contrasts <- makeContrasts( 
 
  MPSIIIB_M - wt_M , # effect of MPS IIIB in males
  MPSIIIB_F - wt_F, # effect of MPS IIIB in females
  
  EOfADlike_M - wt_M, # effect of EOfAD in males
  EOfADlike_F - wt_F, # effect of EOfAD in females

  levels = design.bygroup
  )

# fit the glm
fit_cqn_simple <- x %>% 
  estimateDisp(design.simple) %>% 
  glmFit(design.simple)

fit_cqn_bysex <- x %>% 
  estimateDisp(design.bygroup) %>% 
  glmFit(design.bygroup)

# call toptables 
toptables_cqn_simple <- design.simple %>% colnames() %>% .[2:3] %>% 
  sapply(function(x){
    glmLRT(fit_cqn_simple, coef = x) %>%
      topTags(n = Inf) %>%
      .[["table"]] %>%
      as_tibble() %>%
      arrange(PValue) %>%
       mutate(
        coef = x,
        DE = FDR < 0.05
      ) %>% 
      dplyr::select(
        gene_name, logFC, logCPM, PValue, FDR, DE, everything()  
      )
      
  }, simplify = FALSE)

toptables_cqn_bygroup <- 
  colnames(contrasts) %>% 
  sapply(function(x) {
    glmLRT(fit_cqn_bysex, contrast = contrasts[,x]) %>% 
      topTags(n = Inf)  %>% 
      .[["table"]] %>%
      as_tibble() %>%
      arrange(PValue) %>%
      mutate(
        coef = x,
        coef = str_remove(coef, pattern = " - .+"),
        DE = FDR < 0.05
      ) %>% 
      dplyr::select(
        gene_name, logFC, logCPM, PValue, FDR, DE, everything()
      )
        }, simplify = FALSE)

# tidy up the names of this. 
names(toptables_cqn_bygroup) %<>% 
  str_remove(pattern = " - .+")

```

## Visualisation

### tables of n DE
```{r}
toptables_cqn_bygroup %>% 
  lapply(function(y) {
    y %>% 
      mutate(
      direction = sign(logFC), .after = logFC
      ) %>% 
      dplyr::filter(DE == TRUE) %>% 
      group_by(direction) %>% 
      summarise(n = n())
  })

toptables_cqn_bygroup %>% lapply(dplyr::filter, DE == T)


toptables_cqn_simple %>% 
  lapply(function(y) {
    y %>% 
      mutate(
      direction = sign(logFC), .after = logFC
      ) %>% 
      dplyr::filter(DE == TRUE) %>% 
      group_by(direction) %>% 
      summarise(n = n())
  })
```


### Volcano plot
```{r}
toptables_cqn_simple %>%
  bind_rows() %>% 
  ggplot(aes(y = -log10(PValue), x = logFC, colour = DE)) +
  geom_point(
    alpha = 0.5, size = 1.25
  ) +
  facet_wrap(~coef) +
  # geom_label_repel(
  #   aes(label = gene_name),
  #   data = .  %>% dplyr::filter(FDR < 0.05),
  #   show.legend = FALSE
  #   ) +
  coord_cartesian(xlim = c(-2.5,2.5), 
                  ylim = c(0, 20) )+
  theme(legend.position = "none") +
  scale_color_manual(values = c("grey50", "red")) 

toptables_cqn_bygroup %>%
  bind_rows() %>% 
  ggplot(aes(y = -log10(PValue), x = logFC, colour = DE)) +
  geom_point(
    alpha = 0.5, size = 1.25
  ) +
  facet_wrap(~coef) +
  # geom_label_repel(
  #   aes(label = gene_name),
  #   data = .  %>% dplyr::filter(FDR < 0.05),
  #   show.legend = FALSE
  #   ) +
  coord_cartesian(xlim = c(-2.5,2.5), 
                  ylim = c(0, 20) )+
  theme(legend.position = "none") +
  scale_color_manual(values = c("grey50", "red")) 
```

### MD plot
```{r}
toptables_cqn_bygroup %>%
  bind_rows() %>% 
  ggplot(aes(x = logCPM, y = logFC)) +
  geom_point(
    aes(colour = DE),
    alpha = 0.5
  ) +
  facet_wrap(~coef, ncol = 1) +
  theme(legend.position = "none") +
  scale_color_manual(values = c("grey50", "red")) +
  geom_smooth(se = F)
```


# Gene set enrichment analysis

I now am going to test the KEGG, gene ontology (GO), iron-responsive element (IRE), cell type markers and chromosomal location gene sets to determine whether any of them display changes to gene expression as a group. The gene sets were obtained from MSIGDB using the msigdbr package. 

### Gene sets

The KEGG and GO terms were obtained from msigdb using msigdbr. These gene sets were filtered to only contain the genes above the detectable threshold in this experiment. Also, any gene sets with less than 5 genes were omitted, as these probably arent very biologically meaninigful. The GO terms were also restricted to those with 3 or more steps back to the ontology root terms. 

The iron-responsive element (IRE) genes were obtained from Hin et al. 2021 (DOI: 10.3233/JAD-210200). The zebrafish IRE-containing genes were used, and were also filtered to only retain the detectable genes. 

The chromosome gene sets were generated from the DGE object. 

The cell type marker gene sets were obtained from the single-cell RNA-seq data of Jiang et al. 2021 (https://doi.org/10.3389/fcell.2021.743421)/

```{r}

# Obtain the KEGG gene sets. 
KEGG <- msigdbr("Danio rerio", category = "C2", subcategory = "CP:KEGG") %>% 
  left_join(x$genes, by = c("gene_symbol" = "gene_name")) %>% 
  dplyr::filter(gene_id %in% rownames(x)) %>% 
  distinct(gs_name, gene_id, .keep_all = TRUE) %>% 
  split(f = .$gs_name) %>%
  lapply(extract2, "gene_id")

# calculate the number of genes per gene set
sizes <- KEGG %>% 
  lapply(length) %>% 
  unlist %>% 
  as.data.frame() %>% 
  set_colnames( "n_genes") %>% 
  rownames_to_column("gs")


# retain gene sets with at least 3 genes in it
KEGG <- KEGG[sizes %>% dplyr::filter(n_genes > 3) %>% .$gs]

# GO Terms summaries that The UofA biohub made. 
goSummaries <- url("https://uofabioinformaticshub.github.io/summaries2GO/data/goSummaries.RDS") %>%
  readRDS() %>%
  mutate(
    Term = Term(id),
    gs_name = Term %>% str_to_upper() %>% str_replace_all("[ -]", "_"),
    gs_name = paste0("GO_", gs_name)
    )

minPath <- 3

# obtain the GO terms 
GO <- 
  msigdbr("Danio rerio", category = "C5") %>% 
  dplyr::filter(grepl(gs_name, pattern = "^GO")) %>% 
  left_join(x$genes, by = c("gene_symbol" = "gene_name")) %>% 
  dplyr::filter(gene_id %in% rownames(x)) %>% 
  mutate(gs_name = str_replace(gs_name, pattern = "GOBP_", replacement = "GO_")) %>% 
  mutate(gs_name = str_replace(gs_name, pattern = "GOMF_", replacement = "GO_")) %>%     
  mutate(gs_name = str_replace(gs_name, pattern = "GOCC_", replacement = "GO_")) %>%     
  left_join(goSummaries) %>% 
  dplyr::filter(shortest_path >= minPath) %>%
  distinct(gs_name, gene_id, .keep_all = TRUE) %>% 
  split(f = .$gs_name) %>%
  lapply(extract2, "gene_id")

GOdf <-  msigdbr("Danio rerio", category = "C5") %>%
  dplyr::filter(grepl(gs_name, pattern = "^GO")) %>%
  left_join(x$genes, by = c("gene_symbol" = "gene_name")) %>%
  dplyr::filter(gene_id %in% rownames(x)) %>%
  mutate(
    gs_full_name = gs_name, 
    gs_name = str_replace(gs_name, pattern = "GOBP_", replacement = "GO_"), 
    gs_name = str_replace(gs_name, pattern = "GOMF_", replacement = "GO_"), 
    gs_name = str_replace(gs_name, pattern = "GOCC_", replacement = "GO_")
  ) %>%
  left_join(goSummaries) %>%
  dplyr::filter(shortest_path >= minPath) %>%
  distinct(gs_name, gene_id, .keep_all = TRUE) 

sizes_GO <- GO %>%
  lapply(length) %>%
  unlist %>%
  as.data.frame() %>%
  set_colnames( "n_genes") %>%
  rownames_to_column("gs")


# retain gene sets with at least 3 genes in it
GO <- GO[sizes_GO %>% dplyr::filter(n_genes > 3) %>% .$gs]

ireGenes <- readRDS("data/gene_sets/zebrafishireGenes.rds") %>% 
  unlist() %>% 
  as.data.frame() %>% 
  rownames_to_column("ire") %>% 
  as_tibble() %>% 
  mutate(ire = str_extract(ire, pattern = "ire[:digit:]_[:alpha:]+")) %>% 
  set_colnames(c("ire", "gene_id")) %>% 
  dplyr::filter(gene_id %in% rownames(x)) %>% 
  split(f = .$ire) %>% 
  lapply(magrittr::extract2,"gene_id") 

chr <- x$genes %>% 
  dplyr::select(gene_id, chromosome) %>% 
  split(f = .$chromosome) %>%
  lapply(extract2, "gene_id") %>% 
  .[c("MT", seq(1:25))] # omit the ALT chromosmes and scaffolds


cell_type_markers <- 
  read_xlsx("data/gene_sets/Suppdata2_jiangetal_2021_fcelldev.xlsx", sheet = "Brain") %>% 
  dplyr::rename("gene_name" = gene) %>% 
  left_join(x$genes) %>% 
  dplyr::filter(gene_id %in% rownames(x)) %>% 
  split(f = .$`cell type`) %>% 
  lapply(extract2, "gene_id")
  

cell_type_markers %<>% 
  lapply(function(y) {
    y[y %in% rownames(x)]
  })

```

## GOSEQ 

I first will look at enrichment of GO terms within the DE genes using goseq. I like goseq as it lets you include a covariate. Since there was still a bit of bias left after cqn for gene length, i will use this as the covariate.  
```{r}
pwfs <- 
  c(toptables_cqn_bygroup, toptables_cqn_simple) %>% 
  lapply(function(y) {
    
    y %>% 
      with(
        nullp(
          DEgenes = structure(DE, names = gene_id), 
          bias.data = length
        )
      )
  })

goseq.results <- 
  pwfs %>% 
  lapply(function(y) {
    goseq(y, gene2cat = GO) %>% 
      as_tibble() %>% 
      mutate(
        FDR = p.adjust(over_represented_pvalue, method = "fdr")
        ) %>% 
      dplyr::select(-under_represented_pvalue) %>% 
      left_join(GOdf %>% dplyr::select(category = gs_name, gs_subcat) %>% unique) %>% 
      dplyr::select(GO = category, GO_subcat = gs_subcat, everything())
  })

goseq.results %<>% 
  map2(names(.), ~.x %>% mutate(coef = .y))  
```

## HMP
Since I'm not detecting any significant gene sets in the EOfAD-like mutants, I need to try another method. I will use the harmonic mean p val method which combines indivisual p vals from 3 gene set enrichmen analysis methods: fry, camera and gsea.
```{r}
# a function to run the HMP calculation. 
runHMP = function(
  geneset, 
  logCPM,
  toptable, 
  design, 
  contrasts
  
) {
 
  # run fry
  fry <- 
    colnames(contrasts) %>% 
  sapply(function(y) {
    logCPM %>%
      limma::fry(
        index = geneset,
        design = design,
        contrast = contrasts[,y],
        sort = "mixed"
      ) %>%
      rownames_to_column("pathway") %>%
      as_tibble() %>% 
      mutate(
        coef = y, 
        coef = str_remove(coef, pattern = " - wt.+")
        ) 
  }, simplify = FALSE)

  # run camera
  camera <- colnames(contrasts) %>% 
  sapply(function(y) {
    logCPM %>%
      limma::camera(
        index = geneset,
        design = design,
        contrast = contrasts[,y]
      ) %>%
      rownames_to_column("pathway") %>%
      as_tibble() %>% 
      mutate(coef = y, 
             coef = str_remove(coef, pattern = " - wt.+")) 
  }, simplify = FALSE)
  
  # create ranks for fgsea
  ranks <-
    sapply(toptable, function(y) {
      y %>%
        mutate(rankstat = sign(logFC) * log10(1/PValue)) %>%
        arrange(rankstat) %>%
        dplyr::select(c("gene_id", "rankstat")) %>% #only want the Pvalue with sign
        with(structure(rankstat, names = gene_id)) %>%
        rev() # reverse so the start of the list is upregulated genes
    }, simplify = FALSE)
  
  # run fgsea
  set.seed(1)
  fgsea <- ranks %>% 
  sapply(function(x){
    fgseaMultilevel(stats = x, 
                    pathways = geneset) %>%
      as_tibble() %>%
      dplyr::rename(FDR = padj) %>%
      mutate(padj = p.adjust(pval, "bonferroni")) %>%
      dplyr::select(pathway, pval, FDR, padj, everything()) %>%
      arrange(pval) 
  }, simplify = F)
  
  fgsea %<>% 
    map2(names(.), ~.x %>% mutate(coef = .y))  

  
  # calculate HMP
  hmp <- 
    fry %>%
    bind_rows() %>%
    dplyr::select(pathway, PValue.Mixed, coef) %>%
    dplyr::rename(fry_p = PValue.Mixed) %>%
    left_join(camera %>%
                bind_rows() %>%
                dplyr::select(pathway, PValue, coef),
              by = c("pathway", "coef")) %>%
    dplyr::rename(camera_p = PValue) %>%
    left_join(fgsea %>%
                bind_rows() %>%
                dplyr::select(pathway, pval, coef),
              by = c("pathway", "coef")) %>%
    dplyr::rename(fgsea_p = pval) %>%
    bind_rows() %>%
    nest(p = one_of(c("fry_p", "camera_p", "fgsea_p"))) %>%
    mutate(harmonic_p = vapply(p, function(x){
      x <- unlist(x)
      x <- x[!is.na(x)]
      p.hmp(x, L = 4)
    }, numeric(1))
    ) %>%
    unnest() %>%
    mutate(harmonic_p_FDR = p.adjust(harmonic_p, "fdr"),
           sig = harmonic_p_FDR < 0.05) %>%
    arrange(harmonic_p)
  
  return( 
    list(
      hmp = hmp,
      fgsea = fgsea
    ))
}
```

### KEGG
```{r}
hmp.adult.kegg.cqn.bygroup <- runHMP(
  geneset = KEGG, logCPM = logCPM, 
  toptable = toptables_cqn_bygroup, 
  design = design.bygroup, contrasts = contrasts
  )

sig.kegg.morethan1 <- hmp.adult.kegg.cqn.bygroup$hmp %>% 
  dplyr::filter(harmonic_p_FDR < 0.05) %>% 
  group_by(pathway) %>% 
  summarise(n = n()) %>% 
  ungroup %>% 
  dplyr::filter(n > 1)

hmp.adult.kegg.cqn.bygroup$hmp %>%   
  dplyr::filter(pathway %in% sig.kegg.morethan1$pathway) %>% 
  left_join(sig.kegg.morethan1) %>% 
  mutate(
    sex = case_when(
      grepl(coef, pattern = "_F") ~ "female", 
      grepl(coef, pattern = "_M") ~ "male"
    )
  ) %>% 
  ggplot(
    aes(
      x = coef, 
      y = reorder(pathway, n-harmonic_p)
      )) +
  geom_tile(
    aes(fill = -log10(harmonic_p), alpha = sig)
  ) +
  geom_label(
    aes(label = signif(harmonic_p_FDR, 2))
  ) +
  scale_fill_viridis_c() +
  facet_wrap(~sex, scales = "free_x")
```

#### oxidative phosphorylation

```{r}
oxphos.adult <- toptables_cqn_bygroup %>% 
  bind_rows() %>% 
  dplyr::filter(gene_id %in% KEGG$KEGG_OXIDATIVE_PHOSPHORYLATION) %>% 
  dplyr::select(gene_name, logFC, coef) %>% 
  dplyr::distinct(gene_name, coef, .keep_all = T) %>% 
  spread(key = "coef", value = "logFC") %>% 
  column_to_rownames("gene_name") %>% 
  as.matrix()
```

```{r}
#png("output/plots/new_KEGG_hms/oxphos.png", width = 8, height = 12, units = "cm", res = 300)
pheatmap(
  oxphos.adult, 
  color = colorRampPalette(rev(brewer.pal(n = 7, name = "RdBu")))(100), 
  breaks = seq(-0.4,0.4, by = 0.01), 
  cellwidth = 20, 
  treeheight_row = 0, 
  show_rownames = FALSE
  )
#dev.off()
```

upset plot to show the overlap of leading edge genes
```{r}
#png("output/plots/new_KEGG_hms/oxphos_upset.png", width = 3.5, height = 3.5, units = "cm", res = 300)
hmp.adult.kegg.cqn.bygroup$fgsea %>% 
  bind_rows() %>% 
  dplyr::filter(coef != "MPSIIIB_M") %>% 
  dplyr::filter(pathway == "KEGG_OXIDATIVE_PHOSPHORYLATION") %>% 
  dplyr::select(coef, leadingEdge) %>% 
  split(f = .$coef) %>% 
  lapply(function(y) {y$leadingEdge[[1]]}) %>% 
  fromList() %>% 
  upset(
    order.by = "freq", 
    #mb.ratio = c(0.5,0.5),
    text.scale = 0.4, 
    point.size = 0.2,
    line.size = 0.2, 
    set_size.show = FALSE
  )
#dev.off()
  
```

shared leading edge genes:
```{r}
leadingedge.oxphos <- hmp.adult.kegg.cqn.bygroup$fgsea %>% 
  bind_rows() %>% 
  dplyr::filter(pathway == "KEGG_OXIDATIVE_PHOSPHORYLATION") %>% 
  dplyr::filter(coef != "MPSIIIB_M") %>% 
  dplyr::select(coef, leadingEdge) %>% 
  split(f = .$coef) %>% 
  lapply(function(y) {y$leadingEdge[[1]]}) 


#png("output/plots/new_KEGG_hms/oxphos_leadingedge.png", width = 8, height = 6, units = "cm", res = 300)
toptables_cqn_bygroup %>% 
  bind_rows() %>% 
  dplyr::filter(gene_id %in% Reduce(intersect, leadingedge.oxphos)) %>% 
  dplyr::select(gene_name, logFC, coef) %>% 
  dplyr::filter(coef != "MPSIIIB_M") %>% 
  dplyr::distinct(gene_name, coef, .keep_all = T) %>% 
  spread(key = "coef", value = "logFC") %>% 
  column_to_rownames("gene_name") %>% 
  as.matrix() %>% 
  pheatmap(
    color = colorRampPalette(rev(brewer.pal(n = 5, name = "RdBu")))(100), 
    breaks = seq(-0.4, 0.4, by = 0.1), 
    cluster_cols = F, 
    cellwidth = 20,
    treeheight_row = 0, 
    fontsize = 6, 
    border_color = "white", 
    height = 20
  )
#dev.off()
```

#### lysosome

```{r}
lyso.adult <- toptables_cqn_bygroup %>% 
  bind_rows() %>% 
  dplyr::filter(gene_id %in% KEGG$KEGG_LYSOSOME) %>% 
  dplyr::select(gene_name, logFC, coef) %>% 
  dplyr::distinct(gene_name, coef, .keep_all = T) %>% 
  spread(key = "coef", value = "logFC") %>% 
  column_to_rownames("gene_name") %>% 
  as.matrix()
```

```{r}
#png("output/plots/new_KEGG_hms/lyso.png", width = 8, height = 12, units = "cm", res = 300)
pheatmap(
  lyso.adult, 
  color = colorRampPalette(rev(brewer.pal(n = 7, name = "RdBu")))(100), 
  breaks = seq(-0.5,0.5, by = 0.01), 
  cellwidth = 20, 
  treeheight_row = 0, 
  show_rownames = FALSE
  )
#dev.off()
```

upset plot to show the overlap of leading edge genes
```{r}
#png("output/plots/new_KEGG_hms/lyso_upset.png", width = 3.5, height = 3.5, units = "cm", res = 300)
hmp.adult.kegg.cqn.bygroup$fgsea %>% 
  bind_rows() %>% 
  dplyr::filter(pathway == "KEGG_LYSOSOME") %>% 
  dplyr::select(coef, leadingEdge) %>% 
  split(f = .$coef) %>% 
  lapply(function(y) {y$leadingEdge[[1]]}) %>%
  fromList() %>% 
  upset(
    order.by = "freq", 
    #mb.ratio = c(0.5,0.5),
    text.scale = 0.4, 
    point.size = 0.2,
    line.size = 0.2, 
    set_size.show = FALSE
  )
#dev.off()
```

leading edge genes shared 
```{r}
leadingedge.lyso <- hmp.adult.kegg.cqn.bygroup$fgsea %>% 
  bind_rows() %>% 
  dplyr::filter(pathway == "KEGG_LYSOSOME") %>% 
  dplyr::select(coef, leadingEdge) %>% 
  split(f = .$coef) %>% 
  lapply(function(y) {y$leadingEdge[[1]]}) 

#png("output/plots/new_KEGG_hms/lyso_leadingedge.png", width = 8, height = 6, units = "cm", res = 300)
toptables_cqn_bygroup %>% 
  bind_rows() %>% 
  dplyr::filter(gene_id %in% Reduce(intersect, leadingedge.lyso)) %>% 
  dplyr::select(gene_name, logFC, coef) %>% 
  dplyr::distinct(gene_name, coef, .keep_all = T) %>% 
  spread(key = "coef", value = "logFC") %>% 
  column_to_rownames("gene_name") %>% 
  as.matrix() %>% 
  pheatmap(
    color = colorRampPalette(rev(brewer.pal(n = 5, name = "RdBu")))(100), 
    breaks = seq(-0.4, 0.4, by = 0.1), 
    cluster_cols = F, 
    cellwidth = 20,
    treeheight_row = 0, 
    fontsize = 6, 
    border_color = "white", 
    height = 20
  )
#dev.off()
```

#### ribosome

```{r}
ribo.adult <- toptables_cqn_bygroup %>% 
  bind_rows() %>% 
  dplyr::filter(gene_id %in% KEGG$KEGG_RIBOSOME) %>% 
  dplyr::select(gene_name, logFC, coef) %>% 
  dplyr::distinct(gene_name, coef, .keep_all = T) %>% 
  spread(key = "coef", value = "logFC") %>% 
  column_to_rownames("gene_name") %>% 
  as.matrix()
```

```{r}
#png("output/plots/new_KEGG_hms/ribo.png", width = 8, height = 12, units = "cm", res = 300)
pheatmap(
  ribo.adult, 
  color = colorRampPalette(rev(brewer.pal(n = 7, name = "RdBu")))(100), 
  breaks = seq(-0.5,0.5, by = 0.01), 
  cellwidth = 20, 
  treeheight_row = 0, 
  show_rownames = FALSE, 
  border_color = NA
  )
#dev.off()
```

upset plot to show the overlap of leading edge genes
```{r}
# png("output/plots/new_KEGG_hms/ribo_upset.png", width = 3.5, height = 3.5, units = "cm", res = 300)
hmp.adult.kegg.cqn.bygroup$fgsea %>% 
  bind_rows() %>% 
  dplyr::filter(pathway == "KEGG_RIBOSOME") %>% 
  dplyr::filter(coef != "EOfADlike_M") %>% 
  dplyr::select(coef, leadingEdge) %>% 
  split(f = .$coef) %>% 
  lapply(function(y) {y$leadingEdge[[1]]}) %>%
  fromList() %>% 
  upset(
    order.by = "freq", 
    #mb.ratio = c(0.5,0.5),
    text.scale = 0.4, 
    point.size = 0.2,
    line.size = 0.2, 
    set_size.show = FALSE
  )
#dev.off()
```


shared leadging edge genes
```{r}
leadingedge.ribo <- hmp.adult.kegg.cqn.bygroup$fgsea %>% 
  bind_rows() %>% 
  dplyr::filter(pathway == "KEGG_RIBOSOME") %>% 
  dplyr::filter(coef != "EOfADlike_M") %>% 
  dplyr::select(coef, leadingEdge) %>% 
  split(f = .$coef) %>% 
  lapply(function(y) {y$leadingEdge[[1]]}) 

#png("output/plots/new_KEGG_hms/ribo_leadingedge.png", width = 8, height = 10, units = "cm", res = 300)
toptables_cqn_bygroup %>% 
  bind_rows() %>% 
  dplyr::filter(coef != "EOfADlike_M") %>% 
  dplyr::filter(gene_id %in% Reduce(intersect, leadingedge.ribo)) %>% 
  dplyr::select(gene_name, logFC, coef) %>% 
  dplyr::distinct(gene_name, coef, .keep_all = T) %>% 
  spread(key = "coef", value = "logFC") %>% 
  column_to_rownames("gene_name") %>% 
  as.matrix() %>% 
  pheatmap(
    color = colorRampPalette(rev(brewer.pal(n = 5, name = "RdBu")))(100), 
    breaks = seq(-0.4, 0.4, by = 0.1), 
    cluster_cols = F, 
    cellwidth = 20,
    treeheight_row = 0, 
    fontsize = 6, 
    border_color = "white", 
    height = 22
  )
#dev.off()
```

#### inflammation gene sets
```{r}
plot.cell.type.hm = function(celltype.gs, plot.title) {

  toptables_cqn_bygroup %>%
    bind_rows() %>%
    mutate(
      exp = "mRNA"
    ) %>%
    dplyr::select(
      gene_name, logFC, PValue, FDR, coef, exp
    ) %>%
    bind_rows(
      toptables %>%
        bind_rows() %>%
        mutate(exp = "protein") %>%
        dplyr::filter(coef != "wt_M - wt_F") %>%
        dplyr::select(
          gene_name, logFC, PValue = pval, FDR, coef, exp
        )
    ) %>%
    dplyr::filter(
      gene_name %in% celltype.gs
    ) %>%
    group_by(exp) %>%

    tidyHeatmap::heatmap(
      # input data
      .column = gene_name,
      .row = coef,
      .value = logFC,

      # colours
      palette_value = circlize::colorRamp2(
        seq(-0.4, 0.4, length.out = 11),
        rev(RColorBrewer::brewer.pal(11, "RdBu"))),
      rect_gp = grid::gpar(col = "#161616", lwd = 0.5),

      # sizes
      row_names_gp = gpar(fontsize = 14),
      column_names_gp = gpar(fontsize = 14),
      heatmap_width = unit(28, "cm"),
      heatmap_height = unit(11, "cm"),


      # dendro
      cluster_rows = FALSE,
      show_column_dend = F,
      show_row_dend = F
    ) %>%
    annotation_tile(exp, show_legend = FALSE) %>%
    wrap_heatmap() +
    ggtitle(label = paste(plot.title))

}
```


### fry: cell types

```{r}
fry.celltype.RNAseq <- 
  colnames(contrasts) %>% 
  sapply(function(y) {
    logCPM %>%
      limma::fry(
        index = cell_type_markers,
        design = design.bygroup,
        contrast = contrasts[,y],
        sort = "directional"
      ) %>%
      rownames_to_column("pathway") %>%
      as_tibble() %>% 
      mutate(
        coef = y, 
        coef = str_remove(coef, pattern = " - wt.+") # cleanup the coef col
      ) 
  }, simplify = FALSE)


fry.celltype.RNAseq %>% 
  bind_rows() %>% 
  mutate(
    logpval = -log10(PValue), 
    FDR = signif(FDR, 2)
    ) %>% 
  tidyHeatmap::heatmap(
    
    .row = pathway, 
    .column = coef, 
    .value = logpval, 
    
    show_column_dend = F,
    show_row_dend = F,
    
    col = viridis::viridis(30)
  
) %>% 
  layer_text(
    .value = FDR, 
    FDR < 0.1
    )

```


## HMP IRE

```{r}
hmp.adult.IREgenes.cqn.bygroup <- runHMP(
  geneset = ireGenes, logCPM = logCPM, 
  toptable = toptables_cqn_bygroup, 
  design = design.bygroup, contrasts = contrasts
  )

hmp.adult.IREgenes.cqn.bygroup$hmp %>% 
  mutate(
    logpval = -log10(harmonic_p), 
    harmonic_p_FDR = signif(harmonic_p_FDR, 2)
    ) %>% 
  tidyHeatmap::heatmap(
    
    .row = pathway, 
    .column = coef, 
    .value = logpval, 
    
    show_column_dend = F,
    show_row_dend = F,
    
    col = colorRamp2(seq(0, 4, length = 10), viridis::viridis(10))
  ) %>% 
  layer_text(
    .value = harmonic_p_FDR,
    )

```

# Post-hoc power calc
```{r}
set.seed(2016)

# convert CQN logCPM to counts. 
mu <- logCPM %>% 
  as.data.frame() %>% 
  rownames_to_column("gene_id") %>% 
  gather(key = "sample", value = "logCPM", colnames(x)) %>%
  dplyr::filter(grepl(sample, pattern = "wt")) %>% 
  left_join(
    x$samples %>% dplyr::select(sample, lib.size)
    ) %>% 
  mutate(
    CPM = 10^logCPM, 
    counts = CPM * lib.size
    ) %>% 
  as_tibble() %>% 
  dplyr::select(-logCPM, - lib.size, -CPM) %>% 
  pivot_wider(
    names_from = "sample", values_from = "counts") %>% 
  column_to_rownames("gene_id") %>% 
  rowMeans()


disp <- logCPM %>% 
  as.data.frame() %>% 
  rownames_to_column("gene_id") %>% 
  gather(key = "sample", value = "logCPM", colnames(x)) %>%
  dplyr::filter(grepl(sample, pattern = "wt")) %>% 
  left_join(
    x$samples %>% dplyr::select(sample, lib.size)
    ) %>% 
  mutate(
    CPM = 10^logCPM, 
    counts = CPM * lib.size
    ) %>% 
  as_tibble() %>% 
  dplyr::select(-logCPM, - lib.size, -CPM) %>% 
  pivot_wider(
    names_from = "sample", values_from = "counts") %>% 
  column_to_rownames("gene_id") %>% 
  DGEList() %>% 
  estimateDisp() %>% 
  .$tagwise.dispersion

fc <- function(y){exp(rnorm(y, log(1), 0.5*log(2)))}

power <- ssizeRNA_vary(nGenes = dim(x)[1], # Num detectable genes
              pi0 = 0.9, # Proportion of non-DE genes
              m = 50, # pseudo sample size
              mu = mu, # Average read count for each gene in control group. Calculated from this current dataset)
              disp = disp, # Dispersion parameter for each gene
              fc = fc, # Fold change per gene
              fdr = 0.05, # FDR level to control
              power = 0.9, # power level
              maxN = 40,
              replace = T
                ) 

power$power %>% 
  as_tibble() %>% 
  set_colnames(c("n", "Power")) %>% 
  ggplot(aes(x = n, y = Power)) +
  geom_point() + 
  geom_line() +
  scale_x_continuous(limits = c(0,20))
```

## Data export
```{r}
x %>% saveRDS("data/R_objects/adult_brain/dge.rds")
logCPM %>% saveRDS("data/R_objects/adult_brain/logcpm.rds")
toptables_cqn_bygroup %>% saveRDS("data/R_objects/adult_brain/toptablescqn_bysex.rds")
fry.celltype.RNAseq %>% saveRDS("data/R_objects/adult_brain/fry_cell_RNAseq.rds")
hmp.adult.IREgenes.cqn.bygroup %>% saveRDS("data/R_objects/adult_brain/hmp_ire.rds")
hmp.adult.kegg.cqn.bygroup %>% saveRDS("data/R_objects/adult_brain/hmp.kegg.rna.rds") 
```




<!-- ## extra plots -->
<!-- ```{r} -->
<!-- hmp.fry %>%  -->
<!--   dplyr::filter(pathway %in% sigpaths) %>%  -->
<!--   arrange(harmonic_p_FDR) %>%  -->
<!--   mutate(order = case_when( -->
<!--     pathway %in% c("KEGG_OXIDATIVE_PHOSPHORYLATION", "KEGG_RIBOSOME") ~ 1, -->
<!--     pathway == "KEGG_PARKINSONS_DISEASE" ~ 2, -->
<!--     pathway == "KEGG_LYSOSOME" ~ 3, -->
<!--     pathway == "KEGG_OTHER_GLYCAN_DEGRADATION" ~ 4, -->
<!--     pathway == "KEGG_AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM" ~ 5, -->
<!--     pathway == "KEGG_GLYCOSAMINOGLYCAN_DEGRADATION" ~ 6, -->
<!--     TRUE ~7 -->
<!--            ) -->
<!--     ) %>%  -->
<!--   ggplot(aes(x = coef, y = reorder(pathway, -order))) +  -->
<!--   geom_tile(aes(fill = -log10(harmonic_p),  -->
<!--                 alpha = sig)) + -->
<!--   geom_label(aes(label = signif(harmonic_p_FDR, digits = 2)),  -->
<!--              fill = NA) + -->
<!--   scale_fill_viridis_c() + -->
<!--   labs(y = "KEGG gene set",  -->
<!--        x = "",  -->
<!--        alpha = "FDR p < 0.05") + -->
<!--   theme_classic() -->
<!-- ``` -->

<!-- # compare with Nhis experiment -->
<!-- ```{r} -->
<!-- x.nhi <- readRDS("data/R_objects/adult_brain/dgeNhi.rds") -->
<!-- logCPM.nhi <- readRDS("data/R_objects/adult_brain/logcpmNhi.rds") -->
<!-- toptable.cqn.nhi <- readRDS("data/R_objects/adult_brain/toptablescqnNhi.rds") -->

<!-- hmp.ire.nhi <- readRDS("data/R_objects/adult_brain/hmp_ireNhi.rds") -->
<!-- fry.go.nhi <- readRDS("data/R_objects/adult_brain/fryGoNhi.rds") -->
<!-- hmp.kegg.nhi <- readRDS("data/R_objects/adult_brain/hmp_keggNhi.rds") -->

<!-- ``` -->

<!-- ```{r} -->
<!-- toptable.cqn.nhi <- toptable.cqn.nhi$`Q96_K97del/+` %<>%  -->
<!--   mutate(coef = "EOfAD-like/+ exp 2") -->
<!-- ``` -->


<!-- ```{r} -->
<!-- png("output/plots/oxphos_kegg_adultBrain2.png", width = 7, height = 10, units = "cm", res = 300) -->
<!-- toptables_cqn %>%  -->
<!--   bind_rows() %>%  -->
<!--   bind_rows(toptable.cqn.nhi) %>%  -->
<!--   dplyr::filter(gene_id %in% KEGG$KEGG_OXIDATIVE_PHOSPHORYLATION) %>%  -->
<!--   dplyr::select(gene_name, logFC, coef) %>%  -->
<!--   dplyr::distinct(gene_name, coef, .keep_all = T) %>%  -->
<!--   spread(key = "coef", value = "logFC") %>%  -->
<!--   column_to_rownames("gene_name") %>%  -->
<!--   na.omit %>%  -->
<!--   pheatmap(color = colorRampPalette(rev(brewer.pal(n = 5, -->
<!--                                                name = "RdBu")))(100), -->
<!--            breaks = seq(-0.4,0.4, length.out = 100),  -->
<!--            #cellheight = 30, cellwidth = 12, -->
<!--            border_color = NA,  -->
<!--            angle_col = 45, -->
<!--            show_rownames = F, -->
<!--            cluster_rows = T, -->
<!--            treeheight_row = 0, treeheight_col = 0 -->
<!--            ) -->
<!-- dev.off() -->
<!-- ``` -->

<!-- ```{r} -->
<!-- png("output/plots/lyso_kegg_adultBrain2.png", width = 7, height = 10, units = "cm", res = 300) -->
<!-- toptables_cqn %>%  -->
<!--   bind_rows() %>%  -->
<!--   bind_rows(toptable.cqn.nhi) %>%  -->
<!--   dplyr::filter(gene_id %in% KEGG$KEGG_LYSOSOME) %>%  -->
<!--   dplyr::select(gene_name, logFC, coef) %>%  -->
<!--   dplyr::distinct(gene_name, coef, .keep_all = T) %>%  -->
<!--   spread(key = "coef", value = "logFC") %>%  -->
<!--   column_to_rownames("gene_name") %>%  -->
<!--   na.omit %>%  -->
<!--   pheatmap(color = colorRampPalette(rev(brewer.pal(n = 5, -->
<!--                                                name = "RdBu")))(100), -->
<!--            breaks = seq(-0.5,0.5, length.out = 100),  -->
<!--            #cellheight = 30, cellwidth = 12, -->
<!--            border_color = NA,  -->
<!--            angle_col = 45, -->
<!--            show_rownames = F, -->
<!--            cluster_rows = T, -->
<!--            treeheight_row = 0, treeheight_col = 0 -->
<!--            ) -->
<!-- dev.off() -->
<!-- ``` -->

<!-- ```{r} -->
<!-- png("output/plots/complement__adultBrain2.png", width = 7, height = 10, units = "cm", res = 300) -->
<!-- toptables_cqn %>%  -->
<!--   bind_rows() %>%  -->
<!--   bind_rows(toptable.cqn.nhi) %>%  -->
<!--   dplyr::filter(gene_id %in% KEGG$KEGG_COMPLEMENT_AND_COAGULATION_CASCADES) %>%  -->
<!--   dplyr::select(gene_name, logFC, coef) %>%  -->
<!--   dplyr::distinct(gene_name, coef, .keep_all = T) %>%  -->
<!--   spread(key = "coef", value = "logFC") %>%  -->
<!--   column_to_rownames("gene_name") %>%  -->
<!--   na.omit %>%  -->
<!--   pheatmap(color = colorRampPalette(rev(brewer.pal(n = 5, -->
<!--                                                name = "RdBu")))(100), -->
<!--            breaks = seq(-0.75,0.75, length.out = 100),  -->
<!--            #cellheight = 30, cellwidth = 12, -->
<!--            border_color = NA,  -->
<!--            angle_col = 45, -->
<!--            show_rownames = F, -->
<!--            cluster_rows = T, -->
<!--            treeheight_row = 0, treeheight_col = 0 -->
<!--            ) -->
<!-- dev.off() -->
<!-- ``` -->

<!-- ```{r} -->
<!-- png("output/plots/complement__adultBrain2.png", width = 7, height = 10, units = "cm", res = 300) -->
<!-- toptables_cqn %>%  -->
<!--   bind_rows() %>%  -->
<!--   bind_rows(toptable.cqn.nhi) %>%  -->
<!--   dplyr::filter(gene_id %in% GO$GO_IRON_ION_BINDING) %>%  -->
<!--   dplyr::select(gene_name, logFC, coef) %>%  -->
<!--   dplyr::distinct(gene_name, coef, .keep_all = T) %>%  -->
<!--   spread(key = "coef", value = "logFC") %>%  -->
<!--   column_to_rownames("gene_name") %>%  -->
<!--   na.omit %>%  -->
<!--   pheatmap(color = colorRampPalette(rev(brewer.pal(n = 5, -->
<!--                                                name = "RdBu")))(100), -->
<!--            breaks = seq(-0.75,0.75, length.out = 100),  -->
<!--            #cellheight = 30, cellwidth = 12, -->
<!--            border_color = NA,  -->
<!--            angle_col = 45, -->
<!--            show_rownames = F, -->
<!--            cluster_rows = T, -->
<!--            treeheight_row = 0, treeheight_col = 0 -->
<!--            ) -->
<!-- dev.off() -->
<!-- ``` -->









