---
title: "explorationRUV"
author: "Karissa Barthelson"
date: "2021-11-29"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction
```{r packages}
library(tidyverse)
library(magrittr)
library(readxl)
library(ngsReports)
library(AnnotationHub)
library(msigdbr)
library(pander)
library(scales)
library(pheatmap)
library(ggpubr)
library(ggrepel)
library(ggfortify)
library(RColorBrewer)
library(edgeR)
library(RUVSeq)

theme_set(theme_bw())
```


```{r import}

x <- readRDS("data/R_objects/dge.rds")
```


## RUVSeq to remove unwanted variation ~
I did not detect many DE genes in the original analysis. Therefore, here I will explore whether removal of unwanted variation (RUV) using the RUVSeq package makes a difference. RUVSeq has 3 methods of estimating the unwanted variation:

• RUVg uses negative control genes, assumed to have constant expression across samples.

• RUVs uses centered (technical) replicate/negative control samples for which the covariates of interest are constant.

• RUVr uses residuals, e.g., from a first-pass GLM regression of the counts on the covariates of interest.

Here, I will use the RUVg method using negative control genes, defining the 5000 least DE genes from an ANOVA-like test for differential expression using edgeR. 5000 genes are what was used in the RUVSeq vignette.Thereofre, this is how many I will use

```{r}
design <- model.matrix(~Genotype, data = x$samples) %>% 
  set_colnames(gsub(colnames(.), pattern = "Genotype", replacement = ""))

RUVneg <- 
  x %>% 
  estimateDisp(design) %>%
  glmFit(design) %>%
  glmLRT(coef = 2:3) %>% # use both genotype coefs
  topTags(n = Inf, adjust.method = "fdr", sort.by = "p") %>% 
  .$table %>% 
  arrange(desc(PValue)) %>% 
  .[10000,] %>% 
  .$gene_id

### Perform the RUVg method with 1 factor of unwanted variation removed, 
RUV_k1 <- x$counts %>% 
  round %>% 
  RUVg(RUVneg, 1)
```

```{r}
pca_RUV_k1 <- RUV_k1$normalizedCounts %>% 
  as.data.frame() %>% 
    cpm(log = T) %>% 
    t() %>% 
    prcomp()

pca_RUV_k1 %>% 
  autoplot(
    data = tibble(sample = rownames(.$x)) %>%
      left_join(x$samples),
    colour = "Genotype",
    size = 4
  ) 
```

```{r}
x$samples %<>% 
  cbind(RUV_k1$W)

design_RUVk1 <- 
  model.matrix(~Genotype + W_1 , data = x$samples) %>% 
  set_colnames(gsub("Genotype", "", colnames(.)))

fit_RUVk1 <- x %>% 
  estimateDisp(design_RUVk1) %>% 
  glmFit(design_RUVk1)

topTables_RUVk1 <- design_RUVk1 %>% colnames() %>% .[2:3] %>% 
   sapply(function(y){
    glmLRT(fit_RUVk1, coef = y) %>%
      topTags(n = Inf) %>%
      .[["table"]] %>%
      as_tibble() %>%
      arrange(PValue) %>%
       mutate(
        coef = y,
        DE = FDR < 0.05
      ) %>% 
      dplyr::select(
        gene_name, logFC, logCPM, PValue, FDR, DE, everything()  
      )
      
  }, simplify = FALSE)

KEGG <- msigdbr("Danio rerio", category = "C2", subcategory = "CP:KEGG") %>% 
  left_join(x$genes, by = c("gene_symbol" = "gene_name")) %>% 
  dplyr::filter(gene_id %in% rownames(x)) %>% 
  distinct(gs_name, gene_id, .keep_all = TRUE) %>% 
  split(f = .$gs_name) %>%
  lapply(extract2, "gene_id")

sizes <- KEGG %>% 
  lapply(length) %>% 
  unlist %>% 
  as.data.frame() %>% 
  set_colnames( "n_genes") %>% 
  rownames_to_column("gs")


# retain gene sets with at least 10 genes in it
KEGG <- KEGG[sizes %>% dplyr::filter(n_genes > 10) %>% .$gs]

design_RUVk1 %>% colnames() %>% .[2:3] %>% 
  sapply(function(y) {
    x %>% 
      cpm(log = TRUE) %>% 
      fry(
        index = KEGG,
        design = design_RUVk1, 
        contrast = y, 
        sort = "mixed"
      ) %>% 
      rownames_to_column("pathway") %>% 
      as_tibble() %>% 
      mutate(coef = y, 
             sig = FDR.Mixed < 0.05
      )
  }, simplify = FALSE)

```

```{r}
topTables_RUVk1 %>% 
  bind_rows() %>% 
  dplyr::filter(gene_id %in% KEGG$KEGG_GLYCOSAMINOGLYCAN_DEGRADATION) %>% 
  dplyr::select(gene_name, logFC, coef) %>% 
  spread(key = "coef", value = "logFC") %>% 
  ggplot(aes(x =  `EOfAD-like`, y = `MPS-IIIB`)) +
  geom_point() +
  geom_smooth(method = "lm") +
  stat_cor(method = "pearson", aes(label = paste(..rr.label.., ..p.label.., sep = "~`,`~"))) +
  geom_abline(slope = 1)
                
```

## Exploration with HMP

```{r}
fry <- 
  design %>% colnames() %>% .[2:3] %>% 
  sapply(function(y) {
    cpm(x$counts, log = TRUE) %>% 
      fry(
        index = KEGG,
        design = design_RUVk1, 
        contrast = y, 
        sort = "mixed"
      ) %>% 
      rownames_to_column("pathway") %>% 
      as_tibble() %>% 
      mutate(coef = y)
  }, simplify = FALSE)

camera <- 
  design %>% colnames() %>% .[2:3] %>% 
  sapply(function(y) {
    cpm(x$counts, log = TRUE) %>% 
      camera(
        index = KEGG,
        design = design_RUVk1, 
        contrast = y
      ) %>% 
      rownames_to_column("pathway") %>% 
      as_tibble() %>% 
      mutate(coef = y, 
             sig = FDR < 0.05)
  }, simplify = FALSE)

ranks <- 
   sapply(topTables_RUVk1, function(y) {
     y %>% 
       mutate(rankstat = sign(logFC) * log10(1/PValue)) %>% 
       arrange(rankstat) %>% 
       dplyr::select(c("gene_id", "rankstat")) %>% #only want the Pvalue with sign
       with(structure(rankstat, names = gene_id)) %>% 
       rev() # reverse so the start of the list is upregulated genes
   }, simplify = FALSE)

# Run fgsea

# set a seed for a reproducible result
set.seed(33)
fgsea <- ranks %>%
  sapply(function(x){
    fgseaMultilevel(stats = x, pathways = KEGG) %>%
      as_tibble() %>%
      dplyr::rename(FDR = padj) %>%
      mutate(padj = p.adjust(pval, "bonferroni")) %>%
      dplyr::select(pathway, pval, FDR, padj, everything()) %>%
      arrange(pval) %>%
      mutate(sig = padj < 0.05)
  }, simplify = F)

fgsea$`MPS-IIIB` %<>% 
  mutate(coef = "MPS-IIIB")
fgsea$`EOfAD-like` %<>% 
  mutate(coef = "EOfAD-like")


HMP <- fry %>% 
  bind_rows() %>% 
  dplyr::select(pathway, PValue.Mixed, coef) %>% 
  dplyr::rename(fry_p = PValue.Mixed) %>% 
  left_join(camera %>% 
              bind_rows() %>% 
              dplyr::select(pathway, PValue, coef), 
            by = c("pathway", "coef")) %>% 
  dplyr::rename(camera_p = PValue) %>% 
  left_join(fgsea %>% 
              bind_rows() %>% 
              dplyr::select(pathway, pval, coef), 
            by = c("pathway", "coef")) %>% 
  dplyr::rename(fgsea_p = pval) %>% 
  bind_rows() %>% 
  nest(p = one_of(c("fry_p", "camera_p", "fgsea_p"))) %>% 
 mutate(harmonic_p = vapply(p, function(x){
        x <- unlist(x)
        x <- x[!is.na(x)]
        p.hmp(x, L = 4)
      }, numeric(1))
 ) %>% 
  unnest() %>% 
  mutate(harmonic_p_FDR = p.adjust(harmonic_p, "BY"),
         sig = harmonic_p_FDR < 0.05) %>% 
  arrange(harmonic_p) 


sigpaths <- HMP %>% 
  dplyr::filter(sig == TRUE) %>% 
  .$pathway

HMP %>% 
  dplyr::filter(pathway %in% sigpaths) %>% 
  ggplot(aes(y = pathway, x = -log10(harmonic_p))) +
  geom_col(aes(fill = coef), position = "dodge") +
  geom_vline(xintercept = -log10(2.279347e-03))


RUV_k1$normalizedCounts %>% 
  cpm(log=TRUE) %>% 
  as.data.frame() %>% 
  .[KEGG$KEGG_GLYCOSAMINOGLYCAN_DEGRADATION,] %>% 
  t %>% 
  prcomp() %>% 
  autoplot(data = tibble(sample = rownames(.$x)) %>%
      left_join(x$samples),
    colour = "Genotype", 
    size = 6
  ) +
  theme_bw() + 
  theme(aspect.ratio = 1) +
  labs(colour = "Genotype")

```






