---
title: "QC_adult"
author: "Karissa Barthelson"
date: "2024-11-01"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction
```{r setup, include=FALSE}
knitr::opts_chunk$set(
  autodep = TRUE,
  echo = TRUE,
  warning = FALSE,
  message = FALSE,
  fig.align = "center", 
  out.width ="75%", 
  out.height = "75%"
)
```

```{r loadLibs}
library(tidyverse)
library(magrittr)
library(readxl)
library(ngsReports)
library(plotly)
library(AnnotationHub)
library(pander)
library(scales)
library(pheatmap)
library(ggpubr)

theme_set(theme_bw())
```

```{r anno}
ah <- AnnotationHub() %>%
	subset(species == "Danio rerio") %>%
	subset(rdataclass == "EnsDb")

ensDb <- ah[["AH83189"]] # for release 101, latest version and the alignment
grTrans <- transcripts(ensDb)
trLengths <- exonsBy(ensDb, "tx") %>%
	width() %>%
	vapply(sum, integer(1))
mcols(grTrans)$length <- trLengths[names(grTrans)]
gcGene <- grTrans %>%
  mcols() %>%
  as.data.frame() %>%
  dplyr::select(gene_id, tx_id, gc_content, length) %>%
  as_tibble() %>%
  group_by(gene_id) %>%
  summarise(
    gc_content = sum(gc_content*length) / sum(length),
    length = ceiling(median(length))
  )
grGenes <- genes(ensDb)
mcols(grGenes) %<>%
  as.data.frame() %>%
  left_join(gcGene) %>%
  as.data.frame() %>%
  DataFrame()
```

```{r meta}
# readin meta and tidy up columns
meta <- read_excel("data/adult_brain/karissas_metadata.xlsx", sheet = "onlyseq") %>% 
  mutate(genotype = case_when(
    `usable genotype?` == "wt" ~ "wt", 
    `usable genotype?` == "EOfAD" ~ "EOfADlike", 
    `usable genotype?` == "MPS-III" ~ "MPSIIIB"
  ) %>% 
    factor(levels = c("wt",  "EOfADlike", "MPSIIIB")), 
  tank = as.factor(tank), 
  sex  = as.factor(sex),
  sample = paste0(fish, "_", genotype), 
  group = paste0(genotype, "_", sex) %>% as.factor(),
  age = "6 m adult brain",
  RIN = as.numeric(`RIN/DIN`)
  ) %>% 
  dplyr::select(
    # reorder the metadata nicely
    fish,genotype,sex,group,batch,tank,RIN,everything()
    ) %>% 
  as_tibble()
```


Here, I will assess the quality of the RNA-seq data for the *psen1* Q96_K97del/+ vs *naglu* A603fs/A603fs experiment on zebrafish brains at 6 mo old. 

Total RNA was purified from the brains of individual fish, while the tail end was used for gDNA extraction and PCR genotyping. The total RNA was DNase treated (to remove any genomic DNA which was carried over from the RNA extraction), then delivered to SAGC for polyA+ library preparation and sequencing using the MGI DNBSEQ technology. 

The sequencing was performed over four lanes which were subsequently merged. This was done using the `merge.sh` script shown below. 

```{r meregFilesBashScript}
## Insert the merge files script here
readLines("code/mergeFiles.sh") %>% 
  cat(sep = "\n")
```

## fastqc: raw data

Here, I will use the `ngsReports` package to combine and visualise the fastqc results. 
```{r fastqcRawObejct}
fastqc_raw <- list.files(
  path = "data/adult_brain/fastqc_raw",
  pattern = "zip", 
  recursive = TRUE,
  full.names = TRUE) %>% 
  FastqcDataList()
```

The total number of reads ranged between `r range(readTotals(fastqc_raw)$Total_Sequences) %>% comma %>% pander` reads. Note that the number of reads in the `R1` file indeed equals to the number of reads in the `R2` file. 

```{r}
readTotals(fastqc_raw) %>% 
  mutate(Read = case_when(
    grepl(Filename, pattern = "_R1") ~ "R1", 
    grepl(Filename, pattern = "_R2") ~ "R2"
  ), 
  ULN = str_remove(Filename, "_S.+_merged.+")
  ) %>% 
  left_join(meta) %>% 
  ggplot(aes(x = ULN, y = Total_Sequences, fill = Read)) + 
           geom_col(position = "dodge") +
  coord_flip() +
  scale_fill_viridis_d(end = 0.8) +
  facet_wrap(~genotype, scales = "free_y", ncol = 1, strip.position = "right")
```

The base quality of all the reads also looked good.

```{r}
plotBaseQuals(fastqc_raw)
```

### GC Content

All samples have similar GC content. No issues are present. 

```{r}
plotGcContent(
  x = fastqc_raw, 
  plotType = "line",
  gcType = "Transcriptome", 
  species = "Drerio", 
  usePlotly = TRUE
)
```

### Over-repreented seq

No over-represented sequences are present in this dataset. 

```{r}
getModule(fastqc_raw, "Overrep") 
```


## trimmed data fastQC

The raw fastq. files were then processed with `fastp`. In this step, the adaptor sequeces were trimmed from the reads. Then all length and quality filters were left as default values. Less than 1% of the reads was discarded, and no observed changes are apparent in the %GC in the reads, and quality all looks good. 

```{r}
fastqc_trim <- list.files(path = "data/adult_brain/fastqc_trim",
  pattern = "zip", 
  recursive = TRUE,
  full.names = TRUE) %>% 
  FastqcDataList()
```

```{r}
trimStats <- readTotals(fastqc_raw) %>%
  dplyr::rename(Raw = Total_Sequences) %>%
  left_join(readTotals(fastqc_trim), by = "Filename") %>%
  dplyr::rename(Trimmed = Total_Sequences) %>%
  mutate(
    Discarded = 1 - Trimmed / Raw,
    Retained = Trimmed / Raw
  )

trimStats %>% 
  mutate(ULN = str_remove(Filename, "_S.+_merged.+")
  ) %>% 
  left_join(meta) %>% 
  unique() %>% 
  ggplot(aes(y = ULN)) +
  geom_col(aes(x = Discarded*100)) +
  facet_wrap(~genotype, scales = "free_y", ncol = 1, strip.position = "right") +
  labs(x = "Percentage reads discarded by fastp")
```

```{r}
plotBaseQuals(fastqc_trim)
```


```{r}
ggarrange(
  plotGcContent(
    x = fastqc_raw, 
    plotType = "line",
    gcType = "Transcriptome", 
    species = "Drerio"
  ) +
    theme(legend.position = "none") +
    ggtitle("Before fastp"), 
  plotGcContent(
  x = fastqc_trim, 
  plotType = "line",
  gcType = "Transcriptome", 
  species = "Drerio"
) +
  theme(legend.position = "none")+
  ggtitle("After fastp")
) 

```

## Aligned QC
The reads were aligned to the GRCz11 genome. The majority of reads were aligned uniquely. 

```{r}
fastqc_align <- list.files(
  path = "data/adult_brain/fastqc_align",
  pattern = "zip", 
  recursive = TRUE,
  full.names = TRUE) %>% 
  FastqcDataList()
```

```{r}
list.files("data/adult_brain/starlog", full.names = TRUE) %>% 
  .[grepl(x = ., pattern = "Log.final.out")] %>% 
  ngsReports::plotAlignmentSummary(type = "star") +
  scale_fill_viridis_d(end = 0.8) +
  theme(legend.position = "right") +
  ggtitle("Summary of alignment (STAR)", 
          subtitle = "In all samples, the majority of reads mapped uniquely to the zebrafish genome.")
```

```{r}
plotBaseQuals(fastqc_align)
```


```{r}
plotGcContent(x = fastqc_align, 
    plotType = "line",
    gcType = "Transcriptome", 
    species = "Drerio"
  ) +
  theme(legend.position = "none") 
```

## Dedup align QC

This dataset was processed with UMIs, which allow PCR duplicates to be removed. I did this using `umi-tools`. After de-duplciation ** reads were retained. 

```{r}
fastqc_align_dedup <- list.files(
  path = "data/adult_brain/fastqc_dedup",
  pattern = "zip", 
  recursive = TRUE,
  full.names = TRUE) %>% 
  FastqcDataList()
```

```{r}
readTotals(fastqc_align) %>% 
  mutate(align = "raw") %>% 
  bind_rows(readTotals(fastqc_align_dedup) %>% 
              mutate(align = "dedup")) %>% 
  mutate(ULN = str_remove(Filename, "_S.+_merged.+")) %>% 
  left_join(meta) %>% 
  ggplot(aes(x = ULN, y = Total_Sequences, fill = align)) + 
           geom_col(position = "dodge") +
  coord_flip() +
  scale_fill_viridis_d(end = 0.8) +
  scale_y_continuous(labels = comma) +
  facet_wrap(~genotype, scales = "free_y", ncol = 1, strip.position = "right")
```

## FeatureCounts summary

The majority of reads are counted unqiely. 
```{r}
FC_summary <- 
  read.delim("data/adult_brain/05_featureCounts/counts.out.summary") %>% 
  set_colnames(colnames(.) %>% 
                 str_remove(pattern = "_S.+_merged.Aligned.sortedByCoord.dedup.out.bam") %>% 
                 str_remove(pattern = "X04_dedup.bam.") %>% 
                 str_replace(pattern = "\\.", replacement = "\\-")
  )

FC_summary %>% 
 gather(key = "ULN", value = "NumReads", starts_with("22")) %>% 
  left_join(meta) %>% 
  as_tibble() %>% 
  dplyr::filter(NumReads > 0) %>%   
  ggplot(aes(y = ULN, x = NumReads, fill = Status)) +
  geom_col() +
  scale_fill_viridis_d(end = 0.8) +
  scale_x_continuous(labels = comma) +
  facet_wrap(~genotype, scales = "free_y", ncol = 1, strip.position = "right")
```

# Conclusion

Data looks of sufficient quality. Proceed to analysis. 











